<!DOCTYPE html>
  <head>

 <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="">
<meta name="keywords" content=" ">
<title>Programming in R  | GEN242</title>
<link rel="stylesheet" type="text/css" href="../css/syntax.css">
<link rel="stylesheet" type="text/css" href="../css/font-awesome.min.css">
<!--<link rel="stylesheet" type="text/css" href="../css/bootstrap.min.css">-->
<link rel="stylesheet" type="text/css" href="../css/modern-business.css">
<link rel="stylesheet" type="text/css" href="../css/lavish-bootstrap.css">
<link rel="stylesheet" type="text/css" href="../css/customstyles.css">
<link rel="stylesheet" type="text/css" href="../css/theme-blue.css">
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-cookie/1.4.1/jquery.cookie.min.js"></script>
<script src="../js/jquery.navgoco.min.js"></script>
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/js/bootstrap.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/anchor-js/2.0.0/anchor.min.js"></script>
<script src="../js/toc.js"></script>
<script src="../js/customscripts.js"></script>
<link rel="shortcut icon" href="../common_images/favicon.ico" type="image/x-icon">
<!-- HTML5 Shim and Respond.js IE8 support of HTML5 elements and media queries -->
<!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
<script src="https://oss.maxcdn.com/libs/respond.js/1.4.2/respond.min.js"></script>
<![endif]-->







 

<script>
  $(function () {
      $('[data-toggle="tooltip"]').tooltip()
  })
</script>



  </head>

<body>

   <!-- Navigation -->
<nav class="navbar navbar-inverse navbar-fixed-top" role="navigation">
    <div class="container topnavlinks">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <a class="fa fa-home fa-lg navbar-brand" href="home.html">&nbsp;<span class="projectTitle"> Data Analysis in Genome Biology</span></a>

        </div>
        <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
            <ul class="nav navbar-nav navbar-right">
                <!-- entries without drop-downs appear here -->
                <!-- conditional logic to control which topnav appears for the audience defined in the configuration file.-->
                



















                
                
                
                
                <li><a href="https://piazza.com/class/ilbslqb3j2ovy" target="_blank">Piazza</a></li>
                
                
                
                
                
                <li><a href="https://github.com/tgirke/GEN242" target="_blank">Github Repo</a></li>
                
                
                
                


                <!-- entries with drop-downs appear here -->
                <!-- conditional logic to control which topnav appears for the audience defined in the configuration file.-->

                <li class="dropdown">
                    
                    
                    
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Resources<b class="caret"></b></a>
                    <ul class="dropdown-menu">
                        
                        
                        
                        <li><a href="https://cran.r-project.org/" target="_blank">CRAN</a></li>
                        
                        
                        
                        
                        
                        <li><a href="http://bioconductor.org/" target="_blank">Bioconductor</a></li>
                        
                        
                        
                        
                        
                        <li><a href="http://manuals.bioinformatics.ucr.edu/home/R_BioCondManual" target="_blank">R & Bioconductor Manual</a></li>
                        
                        
                        
                        
                        
                        <li><a href="https://guides.github.com/activities/hello-world/" target="_blank">GitHub</a></li>
                        
                        
                        
                        
                        
                        <li><a href="http://jekyllrb.com/" target="_blank">Jekyll</a></li>
                        
                        
                        
                        
                        
                        <li><a href="http://girke.bioinformatics.ucr.edu/" target="_blank">Girke Lab</a></li>
                        
                        
                        
                        
                        
                        <li><a href="http://idratherbewriting.com/documentation-theme-jekyll" target="_blank">Jekyll Doc Theme</a></li>
                        
                        
                        
                        

                    </ul>
                </li>
                
                


                <!-- special insertion -->

               
                
                
                <!-- Send feedback function -->
<script>
function SendLinkByMail(href) {
var subject= "GEN242 feedback";
var body = "I have some feedback about the Programming in R page: ";
body += window.location.href;
body += "";
var uri = "mailto:?subject=";
uri += encodeURIComponent(subject);
uri += "&body=";
uri += encodeURIComponent(body);
window.location.href = uri;
}
</script>

<li><a href="javascript:(function()%7BSendLinkByMail()%3B%7D)()%3B" target="_blank"><i class="fa fa-envelope-o"></i> Feedback</a></li>


                <!--uncomment this block if you want simple search instead of algolia-->
                <li>
                     <!--start search-->
                    <div id="search-demo-container">
                        <input type="text" id="search-input" placeholder="search...">
                        <ul id="results-container"></ul>
                    </div>
                    <script src="../js/jekyll-search.js" type="text/javascript"></script>
                    <script type="text/javascript">
                        SimpleJekyllSearch.init({
                            searchInput: document.getElementById('search-input'),
                            resultsContainer: document.getElementById('results-container'),
                            dataSource: '../search.json',
                            searchResultTemplate: '<li><a href="{url}" title="Programming in R">{title}</a></li>',
                        noResultsText: 'No results found.',
                                limit: 10,
                                fuzzy: true,
                        })
                    </script>
                     <!--end search-->
                </li>

            
        </div>
        <!-- /.container -->
</nav>


    <!-- Page Content -->
    <div class="container">
        <div class="col-lg-12">&nbsp;</div>


<!-- Content Row -->
<div class="row">
    <!-- Sidebar Column -->
    <div class="col-md-3">

        <script>

            $(document).ready(function() {
                // Initialize navgoco with default options
                $("#mysidebar").navgoco({
                    caretHtml: '',
                    accordion: true,
                    openClass: 'active', // open
                    save: false, // leave false or nav highlighting doesn't work right
                    cookie: {
                        name: 'navgoco',
                        expires: false,
                        path: '/'
                    },
                    slide: {
                        duration: 400,
                        easing: 'swing'
                    }
                });

                $("#collapseAll").click(function(e) {
                    e.preventDefault();
                    $("#mysidebar").navgoco('toggle', false);
                });

                $("#expandAll").click(function(e) {
                    e.preventDefault();
                    $("#mysidebar").navgoco('toggle', true);
                });

            });

        </script>


        



















        <ul id="mysidebar" class="nav">

            <span class="siteTagline">GEN242</span>
            <span class="versionTagline">Spring 2016</span>

            
            
            
                
            
            <li><a href="#">Home</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/home.html">Introduction</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_syllabus.html">Syllabus</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Internal</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_internal.html">Internal Resources</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Schedule</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_schedule.html">Course Schedule</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Slides</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_01.html">Course Introduction</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_02.html">Molbiol for Bioinformatics</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_03.html">Databases and Software</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_04.html">DNA Sequencing</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_05.html">Introduction to R</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_06.html">Sequence Alignments</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_07.html">Programming in R</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_08.html">Multiple Alignments</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_09.html">Short Read Alignments</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_10.html">NGS Analysis Basics</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_11.html">Gene Expression Analysis</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_12.html">NGS Workflows</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Homework</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_homework_01.html">HW1 - Online Tools</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_homework_02.html">HW2 - Biocluster/Linux</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_homework_03.html">HW3 - Introduction to R</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_homework_04.html">HW4 - Pairwise Alignments</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_09.html">HW5 - Programming in R</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_08.html">HW6 - Sequence Analysis</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">GitHub</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_tutorial_01.html">GitHub</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Biocluster & Linux</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_tutorial_02.html">Biocluster</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Introduction to R</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_01.html">1. Overview</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_02.html">2. R Package Repositories</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_03.html">3. Installation of R and Add-on Packages</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_04.html">4. Getting Around</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_05.html">5. Basic Syntax</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_06.html">6. Data Types</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_07.html">7. Data objects</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_08.html">8. Important Utilities</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_09.html">9. Operators and Calculations</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_10.html">10. Reading and Writing External Data</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_11.html">11. Useful R Functions</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_12.html">12. Graphics in R</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_13.html">13. Analysis Routine</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_14.html">14. Session Info</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_15.html">15. References</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Programming in R</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_01.html">1. Overview</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_02.html">2. Control Structures</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_03.html">3. Loops</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_04.html">4. Functions</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_05.html">5. Useful Utilities</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_06.html">6. Running R Scripts</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_07.html">7. Building R Packages</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_08.html">8. Programming Exercises</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_09.html">9. Homework 5</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_10.html">10. Session Info</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_11.html">11. References</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">NGS Analysis Basics</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_01.html">1. Overview</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_02.html">2. Package Requirements</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_03.html">3. Strings in R Base</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_04.html">4. Sequences in Bioconductor</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_05.html">5. NGS Sequences</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_06.html">6. Range Operations</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_07.html">7. Transcript Ranges</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_08.html">8. Homework 6</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_09.html">9. Session Info</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_10.html">10. References</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">NGS Workflow Overview</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeR_1.html">1. Introduction</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeR_2.html">2. Getting Started</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeR_3.html">3. Workflow overview</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeR_4.html">4. Workflow templates</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeR_5.html">5. Version information</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeR_6.html">6. References</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">RNA-Seq Workflow</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeRNAseq_01.html">1. Introduction</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeRNAseq_02.html">2. Sample definitions and environment settings</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeRNAseq_03.html">3. Read preprocessing</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeRNAseq_04.html">4. Alignments</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeRNAseq_05.html">5. Read quantification per annotation range</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeRNAseq_06.html">6. Analysis of differentially expressed genes with edgeR</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeRNAseq_07.html">7. Clustering and heat maps</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeRNAseq_08.html">8. Version Information</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeRNAseq_09.html">9. Funding</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeRNAseq_10.html">10. References</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Projects</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_project_01.html">Project 1</a></li>
                    

                    

                    
                    
                </ul>
                
                
                


                <!-- if you aren't using the accordion, uncomment this block:

                     <p class="external">
                         <a href="#" id="collapseAll">Collapse All</a> | <a href="#" id="expandAll">Expand All</a>
                     </p>
                 -->

    </div>
    <!-- this highlights the active parent class in the navgoco sidebar. this is critical so that the parent expands when you're viewing a page. This must appear below the sidebar code above. Otherwise, if placed inside customscripts.js, the script runs before the sidebar code runs and the class never gets inserted.-->
    <script>$("li.active").parents('li').toggleClass("active");</script>


            <!-- Content Column -->
            <div class="col-md-9">
                
                <div class="post-header">
   <h1 class="post-title-main">Programming in R</h1>
</div>

<div class="post-content">

   

<!-- this handles the automatic toc. use ## for subheads to auto-generate the on-page minitoc. if you use html tags, you must supply an ID for the heading element in order for it to appear in the minitoc. -->
<script>
$( document ).ready(function() {
  // Handler for .ready() called.

$('#toc').toc({ minimumHeaders: 0, listType: 'ul', showSpeed: 0, headers: 'h2,h3,h4' });

/* this offset helps account for the space taken up by the floating toolbar. */
$('#toc').on('click', 'a', function() {
  var target = $(this.getAttribute('href'))
    , scroll_target = target.offset().top

  $(window).scrollTop(scroll_target - 10);
  return false
})
  
});
</script>


<div id="toc"></div>


<!--
    
    <a target="_blank" href="https://github.com/tgirke/manuals/tree/master/mydoc/" class="btn btn-default githubEditButton" role="button"><i class="fa fa-github fa-lg"></i> Edit me</a>
    
-->
<!--
%% \VignetteEngine{knitr::rmarkdown}
%\VignetteIndexEntry{Overview Vignette}
%% \VignetteDepends{methods}
%% \VignetteKeywords{compute cluster, pipeline, reports}
%% \VignettePackage{longevityTools}
-->

<!---
- Compile from command-line
echo "rmarkdown::render('Programming_in_R.Rmd', clean=F)" | R -slave; R CMD Stangle Programming_in_R.Rmd; Rscript ../md2jekyll.R Programming_in_R.knit.md 9

- Commit to github
git commit -am "some edits"; git push -u origin master

- To customize font size and other style features, add this line to output section in preamble:  
    css: style.css
-->

<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script>

<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script>

<h1 id="overview">Overview</h1>

<p>One of the main attractions of using the R
(<a href="http://cran.at.r-project.org">http://cran.at.r-project.org</a>) environment is
the ease with which users can write their own programs and custom functions.
The R programming syntax is extremely easy to learn, even for users with no
previous programming experience. Once the basic R programming control
structures are understood, users can use the R language as a powerful
environment to perform complex custom analyses of almost any type of data [@Gentleman2008-xo].</p>

<h2 id="why-programming-in-r">Why Programming in R?</h2>

<ul>
  <li>Powerful statistical environment and programming language</li>
  <li>Facilitates reproducible research</li>
  <li>Efficient data structures make programming very easy</li>
  <li>Ease of implementing custom functions</li>
  <li>Powerful graphics</li>
  <li>Access to fast growing number of analysis packages</li>
  <li>Most widely used language in bioinformatics</li>
  <li>Is standard for data mining and biostatistical analysis</li>
  <li>Technical advantages: free, open-source, available for all OSs</li>
</ul>

<h2 id="r-basics">R Basics</h2>

<p>The previous <a href="http://girke.bioinformatics.ucr.edu/GEN242/mydoc/mydoc_Rbasics_01.html">Rbasics</a> tutorial provides a general introduction to the usage of the R environment and its basic command syntax.
More details can be found in the R &amp; BioConductor manual <a href="http://manuals.bioinformatics.ucr.edu/home/R_BioCondManual">here</a>.</p>

<h2 id="code-editors-for-r">Code Editors for R</h2>

<p>Several excellent code editors are available that provide functionalities like R syntax highlighting, auto code indenting and utilities to send code/functions to the R console.</p>

<ul>
  <li><a href="https://www.rstudio.com/products/rstudio/features/">RStudio</a>: GUI-based IDE for R</li>
  <li><a href="http://manuals.bioinformatics.ucr.edu/home/programming-in-r/vim-r">Vim-R-Tmux</a>: R working environment based on vim and tmux</li>
  <li><a href="http://www.xemacs.org/Download/index.html">Emacs</a> (<a href="http://ess.r-project.org/">ESS add-on package</a>)</li>
  <li><a href="https://wiki.gnome.org/Apps/Gedit">gedit</a> and <a href="https://wiki.gnome.org/Apps/Gedit">Rgedit</a></li>
  <li><a href="https://rkward.kde.org/">RKWard</a></li>
  <li><a href="http://www.walware.de/goto/statet">Eclipse</a></li>
  <li><a href="http://jekyll.math.byuh.edu/other/howto/tinnr/install.shtml">Tinn-R</a></li>
  <li><a href="https://sourceforge.net/projects/npptor/">Notepad++ (NppToR)</a></li>
</ul>

<center>Programming in R using RStudio</center>
<center><img title="R_Interfaces" src="images/rstudio.png" /></center>

<center> Programming in R using Vim or Emacs</center>
<center><img title="vim-r" src="images/vimR.png" /></center>

<h2 id="finding-help">Finding Help</h2>

<p>Reference list on R programming (selection)</p>

<ul>
  <li><a href="http://adv-r.had.co.nz/">Advanced R</a>, by Hadley Wickham</li>
  <li><a href="http://master.bioconductor.org/help/publications/books/r-programming-for-bioinformatics/">R Programming for Bioinformatics</a>, by Robert Gentleman</li>
  <li><a href="http://www.stats.ox.ac.uk/pub/MASS3/Sprog/">S Programming</a>, by W. N. Venables and B. D. Ripley</li>
  <li><a href="http://www.amazon.com/Programming-Data-Language-Lecture-Economics/dp/0387985034">Programming with Data</a>, by John M. Chambers</li>
  <li><a href="http://www1.maths.lth.se/help/R/">R Help</a> &amp; <a href="http://www1.maths.lth.se/help/R/RCC/">R Coding Conventions</a>, Henrik Bengtsson, Lund University</li>
  <li><a href="http://zoonek2.free.fr/UNIX/48_R/02.html">Programming in R</a> (Vincent Zoonekynd)</li>
  <li><a href="http://www2.warwick.ac.uk/fac/sci/moac/people/students/peter_cock/r">Peter’s R Programming Pages</a>, University of Warwick</li>
  <li><a href="http://pj.freefaculty.org/R/statsRus.html">Rtips</a>, Paul Johnsson, University of Kansas</li>
  <li><a href="http://heather.cs.ucdavis.edu/~matloff/r.html">R for Programmers</a>, Norm Matloff, UC Davis</li>
  <li><a href="http://www.statistik.uni-dortmund.de/useR-2008/tutorials/useR2008introhighperfR.pdf">High-Performance R</a>, Dirk Eddelbuettel tutorial presented at useR-2008</li>
  <li><a href="http://www.stat.harvard.edu/ccr2005/index.html">C/C++ level programming for R</a>, Gopi Goswami</li>
</ul>

<h1 id="control-structures">Control Structures</h1>

<h2 id="important-operators">Important Operators</h2>

<h3 id="comparison-operators">Comparison operators</h3>

<ul>
  <li><code>==</code> (equal)</li>
  <li><code>!=</code> (not equal)</li>
  <li><code>&gt;</code> (greater than)</li>
  <li><code>&gt;=</code> (greater than or equal)</li>
  <li><code>&lt;</code> (less than)</li>
  <li><code>&lt;=</code> (less than or equal)</li>
</ul>

<h3 id="logical-operators">Logical operators</h3>

<ul>
  <li><code>&amp;</code> (and)</li>
  <li><code>|</code> (or)</li>
  <li><code>!</code> (not)</li>
</ul>

<h2 id="conditional-executions-if-statements">Conditional Executions: <code>if</code> Statements</h2>

<p>An <code>if</code> statement operates on length-one logical vectors.</p>

<p><strong>Syntax</strong></p>

<p><code>r
if(TRUE) { 
	statements_1 
} else { 
	statements_2 
}
</code></p>

<p><strong>Example</strong></p>

<p><code>r
if(1==0) { 
	print(1) 
} else { 
	print(2) 
}
</code></p>

<p><code>
## [1] 2
</code></p>

<h2 id="conditional-executions-ifelse-statements">Conditional Executions: <code>ifelse</code> Statements</h2>

<p>The <code>ifelse</code> statement operates on vectors.</p>

<p><strong>Syntax</strong></p>

<p><code>r
ifelse(test, true_value, false_value)
</code>
<strong>Example</strong></p>

<p><code>r
x &lt;- 1:10 
ifelse(x&lt;5, x, 0)
</code></p>

<p><code>
##  [1] 1 2 3 4 0 0 0 0 0 0
</code></p>

<h1 id="loops">Loops</h1>

<h2 id="for-loop"><code>for</code> loop</h2>

<p><code>for</code> loops iterate over elements of a looping vector.</p>

<p><strong>Syntax</strong></p>

<p><code>r
for(variable in sequence) { 
	statements 
}
</code>
<strong>Example</strong></p>

<p><code>r
mydf &lt;- iris
myve &lt;- NULL
for(i in seq(along=mydf[,1])) {
	myve &lt;- c(myve, mean(as.numeric(mydf[i,1:3])))
}
myve[1:8]
</code></p>

<p><code>
## [1] 3.333333 3.100000 3.066667 3.066667 3.333333 3.666667 3.133333 3.300000
</code></p>

<p><strong>Note:</strong> Inject into objecs is much faster than append approach with <code>c</code>, <code>cbind</code>, etc.</p>

<p><strong>Example</strong></p>

<p><code>r
myve &lt;- numeric(length(mydf[,1]))
for(i in seq(along=myve)) {
	myve[i] &lt;- mean(as.numeric(mydf[i,1:3]))
}
myve[1:8]
</code></p>

<p><code>
## [1] 3.333333 3.100000 3.066667 3.066667 3.333333 3.666667 3.133333 3.300000
</code></p>

<h3 id="conditional-stop-of-loops">Conditional Stop of Loops</h3>

<p>The <code>stop</code> function can be used to break out of a loop (or a function) when a condition becomes <code>TRUE</code>. In addition, an error message will be printed.</p>

<p><strong>Example</strong></p>

<p><code>r
x &lt;- 1:10
z &lt;- NULL
for(i in seq(along=x)) { 
	if(x[i] &lt; 5) { 
		z &lt;- c(z, x[i]-1)  
	} else { 
		stop("values need to be &lt; 5") 
	}
}
</code></p>

<h2 id="while-loop"><code>while</code> loop</h2>

<p>Iterates as long as a condition is true.</p>

<p><strong>Syntax</strong></p>

<p><code>r
while(condition) {
	statements
}
</code></p>

<p><strong>Example</strong></p>

<p><code>r
z &lt;- 0
while(z&lt;5) { 
	z &lt;- z + 2
	print(z)  
}
</code></p>

<p><code>
## [1] 2
## [1] 4
## [1] 6
</code></p>

<h2 id="the-apply-function-family">The <code>apply</code> Function Family</h2>

<h3 id="apply"><code>apply</code></h3>

<p><strong>Syntax</strong></p>

<p><code>r
apply(X, MARGIN, FUN, ARGs)
</code></p>

<p><strong>Arguments</strong></p>

<ul>
  <li><code>X</code>: <code>array</code>, <code>matrix</code> or <code>data.frame</code></li>
  <li><code>MARGIN</code>: <code>1</code> for rows, <code>2</code> for columns</li>
  <li><code>FUN</code>: one or more functions</li>
  <li><code>ARGs</code>: possible arguments for functions</li>
</ul>

<p><strong>Example</strong></p>

<p><code>r
apply(iris[1:8,1:3], 1, mean)
</code></p>

<p><code>
##        1        2        3        4        5        6        7        8 
## 3.333333 3.100000 3.066667 3.066667 3.333333 3.666667 3.133333 3.300000
</code></p>

<h3 id="tapply"><code>tapply</code></h3>

<p>Applies a function to vector components that are defined by a factor.</p>

<p><strong>Syntax</strong></p>

<p><code>r
tapply(vector, factor, FUN)
</code></p>

<p><strong>Example</strong></p>

<p><code>r
iris[1:2,]
</code></p>

<p><code>
##   Sepal.Length Sepal.Width Petal.Length Petal.Width Species
## 1          5.1         3.5          1.4         0.2  setosa
## 2          4.9         3.0          1.4         0.2  setosa
</code></p>

<p><code>r
tapply(iris$Sepal.Length, iris$Species, mean)
</code></p>

<p><code>
##     setosa versicolor  virginica 
##      5.006      5.936      6.588
</code></p>

<h3 id="sapply-and-lapply"><code>sapply</code> and <code>lapply</code></h3>

<p>Both apply a function to vector or list objects. The <code>lapply</code> function always returns a list object, while <code>sapply</code> returns <code>vector</code> or <code>matrix</code> objects when it is possible.</p>

<p><strong>Examples</strong></p>

<p><code>r
x &lt;- list(a = 1:10, beta = exp(-3:3), logic = c(TRUE,FALSE,FALSE,TRUE))
lapply(x, mean)
</code></p>

<p><code>
## $a
## [1] 5.5
## 
## $beta
## [1] 4.535125
## 
## $logic
## [1] 0.5
</code></p>

<p><code>r
sapply(x, mean)
</code></p>

<p><code>
##        a     beta    logic 
## 5.500000 4.535125 0.500000
</code></p>

<p>Often used in combination with a function definition:</p>

<p><code>r
lapply(names(x), function(x) mean(x))
sapply(names(x), function(x) mean(x))
</code></p>

<h1 id="functions">Functions</h1>

<h2 id="function-overview">Function Overview</h2>

<p>A very useful feature of the R environment is the possibility to expand existing functions and to easily write custom functions. In fact, most of the R software can be viewed as a series of R functions.</p>

<p><strong>Syntax</strong> to define function</p>

<p><code>r
myfct &lt;- function(arg1, arg2, ...) { 
	function_body 
}
</code>
<strong>Syntax</strong> to call functions</p>

<p><code>r
myfct(arg1=..., arg2=...)
</code>
The value returned by a function is the value of the function body, which is usually an unassigned final expression, <em>e.g.</em>: <code>return()</code></p>

<h2 id="function-syntax-rules">Function Syntax Rules</h2>

<p><strong>General</strong></p>

<ul>
  <li>Functions are defined by
    <ol>
      <li>The assignment with the keyword <code>function</code></li>
      <li>The declaration of arguments/variables (<code>arg1, arg2, ...</code>)</li>
      <li>The definition of operations (<code>function_body</code>) that perform computations on the provided arguments. A function name needs to be assigned to call the function.</li>
    </ol>
  </li>
</ul>

<p><strong>Naming</strong></p>

<ul>
  <li>Function names can be almost anything. However, the usage of names of existing functions should be avoided.</li>
</ul>

<p><strong>Arguments</strong></p>

<ul>
  <li>It is often useful to provide default values for arguments (<em>e.g.</em>: <code>arg1=1:10</code>). This way they don’t need to be provided in a function call. The argument list can also be left empty (<code>myfct &lt;- function() { fct_body }</code>) if a function is expected to return always the same value(s). The argument <code>...</code> can be used to allow one function to pass on argument settings to another.</li>
</ul>

<p><strong>Body</strong></p>

<ul>
  <li>The actual expressions (commands/operations) are defined in the function body which should be enclosed by braces. The individual commands are separated by semicolons or new lines (preferred).</li>
</ul>

<p><strong>Usage</strong></p>

<ul>
  <li>Functions are called by their name followed by parentheses containing possible argument names. Empty parenthesis after the function name will result in an error message when a function requires certain arguments to be provided by the user. The function name alone will print the definition of a function.</li>
</ul>

<p><strong>Scope</strong></p>

<ul>
  <li>Variables created inside a function exist only for the life time of a function. Thus, they are not accessible outside of the function. To force variables in functions to exist globally, one can use the double assignment operator: <code>&lt;&lt;-</code></li>
</ul>

<h2 id="examples">Examples</h2>

<p><strong>Define sample function</strong></p>

<p><code>r
myfct &lt;- function(x1, x2=5) { 
	z1 &lt;- x1 / x1
	z2 &lt;- x2 * x2
        myvec &lt;- c(z1, z2) 
        return(myvec)
} 
</code></p>

<p><strong>Function usage</strong></p>

<p>Apply function to values <code>2</code> and <code>5</code></p>

<p><code>r
myfct(x1=2, x2=5) 
</code></p>

<p><code>
## [1]  1 25
</code></p>

<p>Run without argument names</p>

<p><code>r
myfct(2, 5) 
</code></p>

<p><code>
## [1]  1 25
</code></p>

<p>Makes use of default value <code>5</code></p>

<p><code>r
myfct(x1=2) 
</code></p>

<p><code>
## [1]  1 25
</code>
Print function definition (often unintended)</p>

<p><code>r
myfct 
</code></p>

<p><code>
## function(x1, x2=5) { 
## 	z1 &lt;- x1 / x1
## 	z2 &lt;- x2 * x2
##         myvec &lt;- c(z1, z2) 
##         return(myvec)
## }
</code></p>

<h1 id="useful-utilities">Useful Utilities</h1>

<h2 id="debugging-utilities">Debugging Utilities</h2>

<p>Several debugging utilities are available for R. They include:</p>

<ul>
  <li><code>traceback</code></li>
  <li><code>browser</code></li>
  <li><code>options(error=recover)</code></li>
  <li><code>options(error=NULL)</code></li>
  <li><code>debug</code></li>
</ul>

<p>The <a href="http://www.stats.uwo.ca/faculty/murdoch/software/debuggingR/">Debugging in R page</a> provides an overview of the available resources.</p>

<h2 id="regular-expressions">Regular Expressions</h2>

<p>R’s regular expression utilities work similar as in other languages. To learn how to use them in R, one can consult the main help page on this topic with <code>?regexp</code>.</p>

<h3 id="string-matching-with-grep">String matching with <code>grep</code></h3>

<p>The grep function can be used for finding patterns in strings, here letter <code>A</code> in vector <code>month.name</code>.</p>

<p><code>r
month.name[grep("A", month.name)] 
</code></p>

<p><code>
## [1] "April"  "August"
</code></p>

<h3 id="string-substitution-with-gsub">String substitution with <code>gsub</code></h3>

<p>Example for using regular expressions to substitute a pattern by another one using a back reference. Remember: single escapes <code>\</code> need to be double escaped <code>\\</code> in R.</p>

<p><code>r
gsub('(i.*a)', 'xxx_\\1', "virginica", perl = TRUE) 
</code></p>

<p><code>
## [1] "vxxx_irginica"
</code></p>

<h2 id="interpreting-a-character-string-as-expression">Interpreting a Character String as Expression</h2>

<p>Some useful examples</p>

<p>Generates vector of object names in session</p>

<p><code>r
mylist &lt;- ls()
mylist[1] 
</code></p>

<p><code>
## [1] "i"
</code></p>

<p>Executes 1st entry as expression</p>

<p><code>r
get(mylist[1])
</code></p>

<p><code>
## [1] 150
</code></p>

<p>Alternative approach</p>

<p><code>r
eval(parse(text=mylist[1])) 
</code></p>

<p><code>
## [1] 150
</code></p>

<h2 id="replacement-split-and-paste-functions-for-strings">Replacement, Split and Paste Functions for Strings</h2>

<p><strong>Selected examples</strong></p>

<p>Substitution with back reference which inserts in this example <code>_</code> character</p>

<p><code>r
x &lt;- gsub("(a)","\\1_", month.name[1], perl=T) 
x
</code></p>

<p><code>
## [1] "Ja_nua_ry"
</code></p>

<p>Split string on inserted character from above</p>

<p><code>r
strsplit(x,"_")
</code></p>

<p><code>
## [[1]]
## [1] "Ja"  "nua" "ry"
</code></p>

<p>Reverse a character string by splitting first all characters into vector fields</p>

<p><code>r
paste(rev(unlist(strsplit(x, NULL))), collapse="") 
</code></p>

<p><code>
## [1] "yr_aun_aJ"
</code></p>

<h2 id="time-date-and-sleep">Time, Date and Sleep</h2>

<p><strong>Selected examples</strong></p>

<p>Return CPU (and other) times that an expression used (here ls)</p>

<p><code>r
system.time(ls()) 
</code></p>

<p><code>
##    user  system elapsed 
##       0       0       0
</code></p>

<p>Return the current system date and time</p>

<p><code>r
date() 
</code></p>

<p><code>
## [1] "Wed Apr 13 16:06:06 2016"
</code></p>

<p>Pause execution of R expressions for a given number of seconds (e.g. in loop)</p>

<p><code>r
Sys.sleep(1) 
</code></p>

<h3 id="example">Example</h3>

<h4 id="import-of-specific-file-lines-with-regular-expression">Import of Specific File Lines with Regular Expression</h4>

<p>The following example demonstrates the retrieval of specific lines from an external file with a regular expression. First, an external file is created with the <code>cat</code> function, all lines of this file are imported into a vector with <code>readLines</code>, the specific elements (lines) are then retieved with the <code>grep</code> function, and the resulting lines are split into vector fields with <code>strsplit</code>.</p>

<p><code>r
cat(month.name, file="zzz.txt", sep="\n")
x &lt;- readLines("zzz.txt")
x[1:6] 
</code></p>

<p><code>
## [1] "January"  "February" "March"    "April"    "May"      "June"
</code></p>

<p><code>r
x &lt;- x[c(grep("^J", as.character(x), perl = TRUE))]
t(as.data.frame(strsplit(x, "u")))
</code></p>

<p><code>
##                 [,1]  [,2] 
## c..Jan....ary.. "Jan" "ary"
## c..J....ne..    "J"   "ne" 
## c..J....ly..    "J"   "ly"
</code>
## Calling External Software</p>

<p>External command-line software can be called with <code>system</code>. The following example calls <code>blastall</code> from R</p>

<p><code>r
system("blastall -p blastp -i seq.fasta -d uniprot -o seq.blastp")
</code></p>

<h1 id="running-r-scripts">Running R Scripts</h1>

<h2 id="possibilities-for-executing-r-scripts">Possibilities for Executing R Scripts</h2>

<h3 id="r-console">R console</h3>

<p><code>r
source("my_script.R")
</code></p>

<h3 id="command-line">Command-line</h3>

<p><code>sh
Rscript my_script.R # or just ./myscript.R after making it executable
R CMD BATCH my_script.R # Alternative way 1 
R --slave &lt; my_script.R # Alternative way 2
</code>
### Passing arguments from command-line to R</p>

<p>Create an R script named <code>test.R</code> with the following content:</p>

<p><code>sh
myarg &lt;- commandArgs()
print(iris[1:myarg[6], ])
</code></p>

<p>Then run it from the command-line like this:</p>

<p><code>sh
Rscript test.R 10
</code></p>

<p>In the given example the number <code>10</code> is passed on from the command-line as an argument to the R script which is used to return to <code>STDOUT</code> the first 10 rows of the <code>iris</code> sample data. If several arguments are provided, they will be interpreted as one string and need to be split in R with the strsplit function. A more detailed example can be found <a href="http://manuals.bioinformatics.ucr.edu/home/ht-seq\#TOC-Quality-Reports-of-FASTQ-Files-">here</a>.</p>

<h1 id="building-r-packages">Building R Packages</h1>

<h2 id="short-overview-of-package-building-process">Short Overview of Package Building Process</h2>

<p>R packages can be built with the <code>package.skeleton</code> function. The given example will create a directory named <code>mypackage</code> containing the skeleton of the package for all functions, methods and classes defined in the R script(s) passed on to the <code>code_files</code> argument. The basic structure of the package directory is described <a href="http://manuals.bioinformatics.ucr.edu/home/programming-in-r#Progr_pack">here</a>. The package directory will also contain a file named <code>Read-and-delete-me</code> with instructions for completing the package:</p>

<p><code>r
package.skeleton(name="mypackage", code_files=c("script1.R", "script2.R"))
</code></p>

<p>Once a package skeleton is available one can build the package from the command-line (Linux/OS X). This will create a tarball of the package with its version number encoded in the file name. Subequently, the package tarball needs to be checked for errors with:</p>

<p><code>r
R CMD build mypackage
R CMD check mypackage_1.0.tar.gz
</code></p>

<p>Install package from source</p>

<p><code>r
install.packages("mypackage_1.0.tar.gz", repos=NULL) 
</code></p>

<p>For more details see <a href="http://manuals.bioinformatics.ucr.edu/home/programming-in-r#TOC-Building-R-Packages">here</a></p>

<h1 id="programming-exercises">Programming Exercises</h1>

<h2 id="exercise-1">Exercise 1</h2>

<h3 id="for-loop-1"><code>for</code> loop</h3>

<p><strong>Task 1.1</strong>: Compute the mean of each row in <code>myMA</code> by applying the mean function in a <code>for</code> loop.</p>

<p><code>r
myMA &lt;- matrix(rnorm(500), 100, 5, dimnames=list(1:100, paste("C", 1:5, sep="")))
myve_for &lt;- NULL
for(i in seq(along=myMA[,1])) {
	myve_for &lt;- c(myve_for, mean(as.numeric(myMA[i, ])))
}
myResult &lt;- cbind(myMA, mean_for=myve_for)
myResult[1:4, ]
</code></p>

<p><code>
##           C1         C2         C3          C4         C5   mean_for
## 1 -0.9832766  0.8446066  0.4196481  0.23814667  0.3493797  0.1737009
## 2  1.2980835  0.6924483  0.5996748 -0.51642965  0.6701380  0.5487830
## 3 -1.1466949  0.6752775 -0.9384848  0.07464206 -0.7651351 -0.4200791
## 4  0.8122978 -1.3107710  0.6664631  0.12316103 -0.1726270  0.0237048
</code></p>

<h3 id="while-loop-1"><code>while</code> loop</h3>

<p><strong>Task 1.2</strong>: Compute the mean of each row in <code>myMA</code> by applying the mean function in a <code>while</code> loop.</p>

<p><code>r
z &lt;- 1
myve_while &lt;- NULL
while(z &lt;= length(myMA[,1])) {
	myve_while &lt;- c(myve_while, mean(as.numeric(myMA[z, ])))
	z &lt;- z + 1
}
myResult &lt;- cbind(myMA, mean_for=myve_for, mean_while=myve_while)
myResult[1:4, -c(1,2)]
</code></p>

<p><code>
##           C3          C4         C5   mean_for mean_while
## 1  0.4196481  0.23814667  0.3493797  0.1737009  0.1737009
## 2  0.5996748 -0.51642965  0.6701380  0.5487830  0.5487830
## 3 -0.9384848  0.07464206 -0.7651351 -0.4200791 -0.4200791
## 4  0.6664631  0.12316103 -0.1726270  0.0237048  0.0237048
</code>
<strong>Task 1.3</strong>: Confirm that the results from both mean calculations are identical</p>

<p><code>r
all(myResult[,6] == myResult[,7])
</code></p>

<p><code>
## [1] TRUE
</code></p>

<h3 id="apply-loop"><code>apply</code> loop</h3>

<p><strong>Task 1.4</strong>: Compute the mean of each row in myMA by applying the mean function in an <code>apply</code> loop</p>

<p><code>r
myve_apply &lt;- apply(myMA, 1, mean)
myResult &lt;- cbind(myMA, mean_for=myve_for, mean_while=myve_while, mean_apply=myve_apply)
myResult[1:4, -c(1,2)]
</code></p>

<p><code>
##           C3          C4         C5   mean_for mean_while mean_apply
## 1  0.4196481  0.23814667  0.3493797  0.1737009  0.1737009  0.1737009
## 2  0.5996748 -0.51642965  0.6701380  0.5487830  0.5487830  0.5487830
## 3 -0.9384848  0.07464206 -0.7651351 -0.4200791 -0.4200791 -0.4200791
## 4  0.6664631  0.12316103 -0.1726270  0.0237048  0.0237048  0.0237048
</code></p>

<h3 id="avoiding-loops">Avoiding loops</h3>

<p><strong>Task 1.5</strong>: When operating on large data sets it is much faster to use the rowMeans function</p>

<p><code>r
mymean &lt;- rowMeans(myMA)
myResult &lt;- cbind(myMA, mean_for=myve_for, mean_while=myve_while, mean_apply=myve_apply, mean_int=mymean)
myResult[1:4, -c(1,2,3)]
</code></p>

<p><code>
##            C4         C5   mean_for mean_while mean_apply   mean_int
## 1  0.23814667  0.3493797  0.1737009  0.1737009  0.1737009  0.1737009
## 2 -0.51642965  0.6701380  0.5487830  0.5487830  0.5487830  0.5487830
## 3  0.07464206 -0.7651351 -0.4200791 -0.4200791 -0.4200791 -0.4200791
## 4  0.12316103 -0.1726270  0.0237048  0.0237048  0.0237048  0.0237048
</code></p>

<h2 id="exercise-2">Exercise 2</h2>

<h3 id="custom-functions">Custom functions</h3>

<p><strong>Task 2.1</strong>: Use the following code as basis to implement a function that allows the user to compute the mean for any combination of columns in a matrix or data frame. The first argument of this function should specify the input data set, the second the mathematical function to be passed on (<em>e.g.</em> <code>mean</code>, <code>sd</code>, <code>max</code>) and the third one should allow the selection of the columns by providing a grouping vector.</p>

<p><code>r
myMA &lt;- matrix(rnorm(100000), 10000, 10, dimnames=list(1:10000, paste("C", 1:10, sep="")))
myMA[1:2,]
</code></p>

<p><code>
##           C1         C2         C3        C4         C5        C6         C7         C8         C9
## 1 0.17958477 -0.5714262 -0.8866647  2.463907 -0.8126814 -0.933438 -0.8118745 -0.5546591 -0.9026947
## 2 0.04191538  1.2456075  0.4953736 -2.703100  0.2055566 -1.945671 -2.8696620 -1.5043589 -1.8783809
##          C10
## 1 -0.5698361
## 2 -1.6321152
</code></p>

<p><code>r
myList &lt;- tapply(colnames(myMA), c(1,1,1,2,2,2,3,3,4,4), list) 
names(myList) &lt;- sapply(myList, paste, collapse="_")
myMAmean &lt;- sapply(myList, function(x) apply(myMA[,x], 1, mean))
myMAmean[1:4,] 
</code></p>

<p><code>
##     C1_C2_C3   C4_C5_C6      C7_C8     C9_C10
## 1 -0.4261687  0.2392626 -0.6832668 -0.7362654
## 2  0.5942988 -1.4810715 -2.1870104 -1.7552480
## 3  0.1217488 -0.7225502 -0.6295343  0.4990018
## 4 -0.9118941 -0.3107419  0.3284317 -0.5693107
</code>
&lt;!—
Solution</p>

<p>–&gt;</p>

<h2 id="exercise-3">Exercise 3</h2>

<h3 id="nested-loops-to-generate-similarity-matrices">Nested loops to generate similarity matrices</h3>

<p><strong>Task 3.1</strong>: Create a sample list populated with character vectors of different lengths</p>

<p><code>r
setlist &lt;- lapply(11:30, function(x) sample(letters, x, replace=TRUE))
names(setlist) &lt;- paste("S", seq(along=setlist), sep="") 
setlist[1:6]
</code></p>

<p><code>
## $S1
##  [1] "x" "r" "j" "n" "l" "z" "b" "o" "v" "j" "i"
## 
## $S2
##  [1] "k" "b" "p" "c" "z" "f" "v" "u" "e" "d" "c" "f"
## 
## $S3
##  [1] "l" "e" "p" "j" "i" "k" "y" "i" "w" "l" "w" "x" "p"
## 
## $S4
##  [1] "d" "e" "v" "o" "h" "q" "i" "e" "d" "y" "o" "m" "q" "y"
## 
## $S5
##  [1] "s" "q" "r" "j" "o" "z" "q" "g" "s" "v" "w" "j" "l" "r" "d"
## 
## $S6
##  [1] "c" "l" "h" "v" "e" "a" "i" "u" "g" "h" "s" "f" "u" "b" "e" "y"
</code></p>

<p><strong>Task 3.2</strong>: Compute the length for all pairwise intersects of the vectors stored in <code>setlist</code>. The intersects can be determined with the <code>%in%</code> function like this: <code>sum(setlist[[1]] %in% setlist[[2]])</code></p>

<p><code>r
setlist &lt;- sapply(setlist, unique)
olMA &lt;- sapply(names(setlist), function(x) sapply(names(setlist), 
               function(y) sum(setlist[[x]] %in% setlist[[y]])))
olMA[1:12,] 
</code></p>

<p><code>
##     S1 S2 S3 S4 S5 S6 S7 S8 S9 S10 S11 S12 S13 S14 S15 S16 S17 S18 S19 S20
## S1  10  3  4  3  6  4  6  6  7   8   7   2   5   8   8   8   8   7   8   7
## S2   3 10  3  3  3  6  7  4  7   2   5   6   4   8   7   6   8   6   7   7
## S3   4  3  9  3  3  4  4  5  4   5   5   3   4   5   7   7   8   5   5   5
## S4   3  3  3  9  4  5  3  4  5   5   5   7   7   5   8   7   7   5   6   5
## S5   6  3  3  4 11  4  3  5  6   6   7   7   8  10   9   8   8   7   8   8
## S6   4  6  4  5  4 13  4  5  7   5   7   8   6   9   9   8   9   7   7   9
## S7   6  7  4  3  3  4 12  6  7   5   9   4   6   9   9   8  10   8  10   8
## S8   6  4  5  4  5  5  6 13  9   9  10   7   7   9  12   9  11   9   9   7
## S9   7  7  4  5  6  7  7  9 14   7   8   7   7  11  11  10  11   8   9  11
## S10  8  2  5  5  6  5  5  9  7  13   8   6   6   6  10  11   9   9   8   9
## S11  7  5  5  5  7  7  9 10  8   8  15   7   9  11  11  10  12  11  12  10
## S12  2  6  3  7  7  8  4  7  7   6   7  14   8  10  11   9  11   9   9   8
</code>
<strong>Task 3.3</strong> Plot the resulting intersect matrix as heat map. The <code>image</code> or the <code>heatmap.2</code> function from the <code>gplots</code> library can be used for this.</p>

<p><code>r
image(olMA)
</code></p>

<p><img src="Programming_in_R_files/figure-html/nested_loops3-1.png" alt="" />\</p>

<h2 id="exercise-4">Exercise 4</h2>

<h3 id="build-your-own-r-package">Build your own R package</h3>

<p><strong>Task 4.1</strong>: Save one or more of your functions to a file called <code>script.R</code> and build the package with the <code>package.skeleton</code> function.</p>

<p><code>r
package.skeleton(name="mypackage", code_files=c("script1.R"), namespace=TRUE)
</code></p>

<p><strong>Task 4.2</strong>: Build tarball of the package</p>

<p><code>r
system("R CMD build mypackage")
</code></p>

<p><strong>Task 4.3</strong>: Install and use package</p>

<p><code>r
install.packages("mypackage_1.0.tar.gz", repos=NULL, type="source")
library(mypackage)
?myMAcomp # Opens help for function defined by mypackage
</code></p>

<h1 id="homework-5">Homework 5</h1>

<h2 id="reverse-and-complement-of-dna">Reverse and complement of DNA</h2>

<p><strong>Task 1</strong>: Write a <code>RevComp</code> function that returns the reverse and complement of a DNA sequence string. Include an argument that will allow to return only the reversed sequence, the complemented sequence or the reversed and complemented sequence. The following R functions will be useful for the implementation:</p>

<p><code>r
x &lt;- c("ATGCATTGGACGTTAG")  
x
</code></p>

<p><code>
## [1] "ATGCATTGGACGTTAG"
</code></p>

<p><code>r
x &lt;- substring(x, 1:nchar(x), 1:nchar(x)) 
x
</code></p>

<p><code>
##  [1] "A" "T" "G" "C" "A" "T" "T" "G" "G" "A" "C" "G" "T" "T" "A" "G"
</code></p>

<p><code>r
x &lt;- rev(x) 
x
</code></p>

<p><code>
##  [1] "G" "A" "T" "T" "G" "C" "A" "G" "G" "T" "T" "A" "C" "G" "T" "A"
</code></p>

<p><code>r
x &lt;- paste(x, collapse="")
x
</code></p>

<p><code>
## [1] "GATTGCAGGTTACGTA"
</code></p>

<p><code>r
chartr("ATGC", "TACG", x) 
</code></p>

<p><code>
## [1] "CTAACGTCCAATGCAT"
</code></p>

<p><strong>Task 2</strong>: Write a function that applies the <code>RevComp</code> function to many sequences stored in a vector.</p>

<h2 id="translate-dna-into-protein">Translate DNA into Protein</h2>

<p><strong>Task 3</strong>: Write a function that will translate one or many DNA sequences in all three reading frames into proteins. The following commands will simplify this task:</p>

<p><code>r
AAdf &lt;- read.table(file="http://faculty.ucr.edu/~tgirke/Documents/R_BioCond/My_R_Scripts/AA.txt", header=TRUE, sep="\t") 
AAdf[1:4,]
</code></p>

<p><code>
##   Codon AA_1 AA_3 AA_Full AntiCodon
## 1   TCA    S  Ser  Serine       TGA
## 2   TCG    S  Ser  Serine       CGA
## 3   TCC    S  Ser  Serine       GGA
## 4   TCT    S  Ser  Serine       AGA
</code></p>

<p><code>r
AAv &lt;- as.character(AAdf[,2]) 
names(AAv) &lt;- AAdf[,1] 
AAv
</code></p>

<p><code>
## TCA TCG TCC TCT TTT TTC TTA TTG TAT TAC TAA TAG TGT TGC TGA TGG CTA CTG CTC CTT CCA CCG CCC CCT CAT 
## "S" "S" "S" "S" "F" "F" "L" "L" "Y" "Y" "*" "*" "C" "C" "*" "W" "L" "L" "L" "L" "P" "P" "P" "P" "H" 
## CAC CAA CAG CGA CGG CGC CGT ATT ATC ATA ATG ACA ACG ACC ACT AAT AAC AAA AAG AGT AGC AGA AGG GTA GTG 
## "H" "Q" "Q" "R" "R" "R" "R" "I" "I" "I" "M" "T" "T" "T" "T" "N" "N" "K" "K" "S" "S" "R" "R" "V" "V" 
## GTC GTT GCA GCG GCC GCT GAT GAC GAA GAG GGA GGG GGC GGT 
## "V" "V" "A" "A" "A" "A" "D" "D" "E" "E" "G" "G" "G" "G"
</code></p>

<p><code>r
y &lt;- gsub("(...)", "\\1_", x) 
y &lt;- unlist(strsplit(y, "_")) 
y &lt;- y[grep("^...$", y)] 
AAv[y] 
</code></p>

<p><code>
## GAT TGC AGG TTA CGT 
## "D" "C" "R" "L" "R"
</code></p>

<h2 id="homework-submission">Homework submission</h2>
<p>Submit the 3 functions in one well structured and annotated R script to the instructor. The script should include instructions on how to use the functions.</p>

<h2 id="due-date">Due date</h2>

<p>This homework is due on Thu, April 21th at 6:00 PM.</p>

<h2 id="homework-solutions">Homework Solutions</h2>

<p>See <a href="https://drive.google.com/file/d/0B-lLYVUOliJFWlBhb2xNOWdfS0U/view?usp=sharing">here</a></p>

<h1 id="session-info">Session Info</h1>

<p><code>r
sessionInfo()
</code></p>

<p><code>
## R version 3.2.4 Revised (2016-03-16 r70336)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 14.04.4 LTS
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8       
##  [4] LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
## [10] LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  utils     datasets  grDevices methods   base     
## 
## other attached packages:
## [1] ggplot2_2.0.0   limma_3.26.3    BiocStyle_1.8.0
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_0.12.3      codetools_0.2-14 digest_0.6.9     plyr_1.8.3       grid_3.2.4      
##  [6] gtable_0.1.2     formatR_1.2.1    magrittr_1.5     evaluate_0.8     scales_0.3.0    
## [11] stringi_1.0-1    rmarkdown_0.9.2  tools_3.2.4      stringr_1.0.0    munsell_0.4.2   
## [16] yaml_2.1.13      colorspace_1.2-6 htmltools_0.3    knitr_1.12
</code></p>

<h1 id="references">References</h1>



<div class="tags">
    
</div>



</div>



<footer>
            <div class="row">
                <div class="col-lg-12 footer">
               &copy;2016 UC Riverside. All rights reserved. <br />
 Site last generated: Apr 28, 2016 <br />
<p><img src="../common_images/company_logo.png/></p>
                </div>
            </div>
</footer>

            </div><!-- /.row -->

    </div>    <!-- /.container -->

</body>


</html>

