<!DOCTYPE html>
  <head>

 <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="">
<meta name="keywords" content=" ">
<title>Programming in R  | GEN242</title>
<link rel="stylesheet" type="text/css" href="../css/syntax.css">
<link rel="stylesheet" type="text/css" href="../css/font-awesome.min.css">
<!--<link rel="stylesheet" type="text/css" href="../css/bootstrap.min.css">-->
<link rel="stylesheet" type="text/css" href="../css/modern-business.css">
<link rel="stylesheet" type="text/css" href="../css/lavish-bootstrap.css">
<link rel="stylesheet" type="text/css" href="../css/customstyles.css">
<link rel="stylesheet" type="text/css" href="../css/theme-blue.css">
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-cookie/1.4.1/jquery.cookie.min.js"></script>
<script src="../js/jquery.navgoco.min.js"></script>
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/js/bootstrap.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/anchor-js/2.0.0/anchor.min.js"></script>
<script src="../js/toc.js"></script>
<script src="../js/customscripts.js"></script>
<link rel="shortcut icon" href="../common_images/favicon.ico" type="image/x-icon">
<!-- HTML5 Shim and Respond.js IE8 support of HTML5 elements and media queries -->
<!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
<script src="https://oss.maxcdn.com/libs/respond.js/1.4.2/respond.min.js"></script>
<![endif]-->







 

<script>
  $(function () {
      $('[data-toggle="tooltip"]').tooltip()
  })
</script>



  </head>

<body>

   <!-- Navigation -->
<nav class="navbar navbar-inverse navbar-fixed-top" role="navigation">
    <div class="container topnavlinks">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <a class="fa fa-home fa-lg navbar-brand" href="home.html">&nbsp;<span class="projectTitle"> Data Analysis in Genome Biology</span></a>

        </div>
        <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
            <ul class="nav navbar-nav navbar-right">
                <!-- entries without drop-downs appear here -->
                <!-- conditional logic to control which topnav appears for the audience defined in the configuration file.-->
                



















                
                
                
                
                <li><a href="https://piazza.com/class/ilbslqb3j2ovy" target="_blank">Piazza</a></li>
                
                
                
                
                
                <li><a href="https://github.com/tgirke/GEN242" target="_blank">Github Repo</a></li>
                
                
                
                


                <!-- entries with drop-downs appear here -->
                <!-- conditional logic to control which topnav appears for the audience defined in the configuration file.-->

                <li class="dropdown">
                    
                    
                    
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Resources<b class="caret"></b></a>
                    <ul class="dropdown-menu">
                        
                        
                        
                        <li><a href="https://cran.r-project.org/" target="_blank">CRAN</a></li>
                        
                        
                        
                        
                        
                        <li><a href="http://bioconductor.org/" target="_blank">Bioconductor</a></li>
                        
                        
                        
                        
                        
                        <li><a href="https://guides.github.com/activities/hello-world/" target="_blank">GitHub</a></li>
                        
                        
                        
                        
                        
                        <li><a href="http://jekyllrb.com/" target="_blank">Jekyll</a></li>
                        
                        
                        
                        
                        
                        <li><a href="http://girke.bioinformatics.ucr.edu/" target="_blank">Girke Lab</a></li>
                        
                        
                        
                        
                        
                        <li><a href="http://idratherbewriting.com/documentation-theme-jekyll" target="_blank">Jekyll Doc Theme</a></li>
                        
                        
                        
                        

                    </ul>
                </li>
                
                


                <!-- special insertion -->

               
                
                
                <!-- Send feedback function -->
<script>
function SendLinkByMail(href) {
var subject= "GEN242 feedback";
var body = "I have some feedback about the Programming in R page: ";
body += window.location.href;
body += "";
var uri = "mailto:?subject=";
uri += encodeURIComponent(subject);
uri += "&body=";
uri += encodeURIComponent(body);
window.location.href = uri;
}
</script>

<li><a href="javascript:(function()%7BSendLinkByMail()%3B%7D)()%3B" target="_blank"><i class="fa fa-envelope-o"></i> Feedback</a></li>


                <!--uncomment this block if you want simple search instead of algolia-->
                <li>
                     <!--start search-->
                    <div id="search-demo-container">
                        <input type="text" id="search-input" placeholder="search...">
                        <ul id="results-container"></ul>
                    </div>
                    <script src="../js/jekyll-search.js" type="text/javascript"></script>
                    <script type="text/javascript">
                        SimpleJekyllSearch.init({
                            searchInput: document.getElementById('search-input'),
                            resultsContainer: document.getElementById('results-container'),
                            dataSource: '../search.json',
                            searchResultTemplate: '<li><a href="{url}" title="Programming in R">{title}</a></li>',
                        noResultsText: 'No results found.',
                                limit: 10,
                                fuzzy: true,
                        })
                    </script>
                     <!--end search-->
                </li>

            
        </div>
        <!-- /.container -->
</nav>


    <!-- Page Content -->
    <div class="container">
        <div class="col-lg-12">&nbsp;</div>


<!-- Content Row -->
<div class="row">
    <!-- Sidebar Column -->
    <div class="col-md-3">

        <script>

            $(document).ready(function() {
                // Initialize navgoco with default options
                $("#mysidebar").navgoco({
                    caretHtml: '',
                    accordion: true,
                    openClass: 'active', // open
                    save: false, // leave false or nav highlighting doesn't work right
                    cookie: {
                        name: 'navgoco',
                        expires: false,
                        path: '/'
                    },
                    slide: {
                        duration: 400,
                        easing: 'swing'
                    }
                });

                $("#collapseAll").click(function(e) {
                    e.preventDefault();
                    $("#mysidebar").navgoco('toggle', false);
                });

                $("#expandAll").click(function(e) {
                    e.preventDefault();
                    $("#mysidebar").navgoco('toggle', true);
                });

            });

        </script>


        



















        <ul id="mysidebar" class="nav">

            <span class="siteTagline">GEN242</span>
            <span class="versionTagline">Spring 2016</span>

            
            
            
                
            
            <li><a href="#">Home</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/home.html">Introduction</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_syllabus.html">Syllabus</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Internal</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_internal.html">Internal Resources</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Schedule</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_schedule.html">Course Schedule</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Slides</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_01.html">Course Introduction</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_02.html">Molbiol for Bioinformatics</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_03.html">Databases and Software</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_04.html">DNA Sequencing</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_05.html">Introduction to R</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_06.html">Sequence Alignments</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_07.html">Programming in R</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Homework</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_homework_01.html">HW1 - Online Tools</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_homework_02.html">HW2 - Biocluster/Linux</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_homework_03.html">HW3 - Introduction to R</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_homework_04.html">HW4 - Pairwise Alignments</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">GitHub</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_tutorial_01.html">GitHub</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Biocluster</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_tutorial_02.html">Biocluster</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Introduction to R</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_01.html">1. Overview</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_02.html">2. R Package Repositories</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_03.html">3. Installation of R and Add-on Packages</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_04.html">4. Getting Around</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_05.html">5. Basic Syntax</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_06.html">6. Data Types</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_07.html">7. Data objects</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_08.html">8. Important Utilities</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_09.html">9. Operators and Calculations</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_10.html">10. Reading and Writing External Data</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_11.html">11. Useful R Functions</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_12.html">12. Graphics in R</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_13.html">13. Analysis Routine</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_14.html">14. Session Info</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_15.html">15. References</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Programming</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_1.html">1. Overview</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_2.html">2. Control Structures</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_3.html">3. Loops</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_4.html">4. Session Info</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_5.html">5. References</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Programming in R</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_1.html">1. Overview</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_2.html">2. Control Structures</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_3.html">3. Loops</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_4.html">4. Session Info</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_5.html">5. References</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Projects</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_project_01.html">Project 1</a></li>
                    

                    

                    
                    
                </ul>
                
                
                


                <!-- if you aren't using the accordion, uncomment this block:

                     <p class="external">
                         <a href="#" id="collapseAll">Collapse All</a> | <a href="#" id="expandAll">Expand All</a>
                     </p>
                 -->

    </div>
    <!-- this highlights the active parent class in the navgoco sidebar. this is critical so that the parent expands when you're viewing a page. This must appear below the sidebar code above. Otherwise, if placed inside customscripts.js, the script runs before the sidebar code runs and the class never gets inserted.-->
    <script>$("li.active").parents('li').toggleClass("active");</script>


            <!-- Content Column -->
            <div class="col-md-9">
                
                <div class="post-header">
   <h1 class="post-title-main">Programming in R</h1>
</div>

<div class="post-content">

   

<!-- this handles the automatic toc. use ## for subheads to auto-generate the on-page minitoc. if you use html tags, you must supply an ID for the heading element in order for it to appear in the minitoc. -->
<script>
$( document ).ready(function() {
  // Handler for .ready() called.

$('#toc').toc({ minimumHeaders: 0, listType: 'ul', showSpeed: 0, headers: 'h2,h3,h4' });

/* this offset helps account for the space taken up by the floating toolbar. */
$('#toc').on('click', 'a', function() {
  var target = $(this.getAttribute('href'))
    , scroll_target = target.offset().top

  $(window).scrollTop(scroll_target - 10);
  return false
})
  
});
</script>


<div id="toc"></div>


<!--
    
    <a target="_blank" href="https://github.com/tgirke/manuals/tree/master/mydoc//vignettes/09_Programming_in_R/Programming_in_R.Rmd" class="btn btn-default githubEditButton" role="button"><i class="fa fa-github fa-lg"></i> Edit me</a>
    
-->
<!--
%% \VignetteEngine{knitr::rmarkdown}
%\VignetteIndexEntry{Overview Vignette}
%% \VignetteDepends{methods}
%% \VignetteKeywords{compute cluster, pipeline, reports}
%% \VignettePackage{longevityTools}
-->

<!---
- Compile from command-line
echo "rmarkdown::render('Programming_in_R.Rmd', clean=F)" | R -slave; R CMD Stangle Programming_in_R.Rmd; Rscript ../md2jekyll.R Programming_in_R.knit.md 9

- Commit to github
git commit -am "some edits"; git push -u origin master

- To customize font size and other style features, add this line to output section in preamble:  
    css: style.css
-->

```{r style, echo = FALSE, results = 'asis'}
BiocStyle::markdown()
options(width=100, max.print=1000)
knitr::opts_chunk$set(
    eval=as.logical(Sys.getenv("KNITR_EVAL", "TRUE")),
    cache=as.logical(Sys.getenv("KNITR_CACHE", "TRUE")))
```

```{r setup, echo=FALSE, messages=FALSE, warnings=FALSE}
suppressPackageStartupMessages({
    library(limma) 
    library(ggplot2) }) 
```

# Overview

One of the main attractions of using the R
([http://cran.at.r-project.org](http://cran.at.r-project.org)) environment is
the ease with which users can write their own programs and custom functions.
The R programming syntax is extremely easy to learn, even for users with no
previous programming experience. Once the basic R programming control
structures are understood, users can use the R language as a powerful
environment to perform complex custom analyses of almost any type of data [@Gentleman2008-xo].


## Why Programmin in R?

* Powerful statistical environment and programming language
* Facilitates reproducible research
* Efficient data structures make programming very easy
* Ease of implementing custom functions
* Powerful graphics
* Access to fast growing number of analysis packages
* Most widely used language in bioinformatics
* Is standard for data mining and biostatistical analysis
* Technical advantages: free, open-source, available for all OSs


## R Basics 

The previous [Rbasics](http://girke.bioinformatics.ucr.edu/GEN242/mydoc/mydoc_Rbasics_01.html) provides a general introduction to the usage of the R environment and its basic command syntax.
More details can be found R & BioConductor manual [here](http://manuals.bioinformatics.ucr.edu/home/R_BioCondManual).

## Code Editors for R

Several excellent code editors are available that provide functionalities like R syntax highlighting, auto code indenting and utilities to send code/functions to the R console.

* [RStudio](https://www.rstudio.com/products/rstudio/features/): GUI-based IDE for R
* [Vim-R-Tmux](http://manuals.bioinformatics.ucr.edu/home/programming-in-r/vim-r): R working environment based on vim and tmux
* [Emacs](http://www.xemacs.org/Download/index.html) ([ESS add-on package](http://ess.r-project.org/))
* [gedit](https://wiki.gnome.org/Apps/Gedit) and [Rgedit](https://wiki.gnome.org/Apps/Gedit)
* [RKWard](https://rkward.kde.org/)
* [Eclipse](http://www.walware.de/goto/statet)
* [Tinn-R](http://jekyll.math.byuh.edu/other/howto/tinnr/install.shtml)
* [Notepad++ (NppToR)](https://sourceforge.net/projects/npptor/)

<center> Programming in R using Vim or Emacs</center>
<center><img title="vim-r" src="images/vimR.png"/></center>

<center>Programming in R using RStudio</center>
<center><img title="R_Interfaces" src="images/rstudio.png"/></center>

## Finding Help

Reference list on R programming (selection)

* [Advanced R](http://adv-r.had.co.nz/), by Hadley Wickham
* [R Programming for Bioinformatics](http://master.bioconductor.org/help/publications/books/r-programming-for-bioinformatics/), by Robert Gentleman
* [S Programming](http://www.stats.ox.ac.uk/pub/MASS3/Sprog/), by W. N. Venables and B. D. Ripley
* [Programming with Data](http://www.amazon.com/Programming-Data-Language-Lecture-Economics/dp/0387985034), by John M. Chambers
* [R Help](http://www1.maths.lth.se/help/R/) & [R Coding Conventions](http://www1.maths.lth.se/help/R/RCC/), Henrik Bengtsson, Lund University
* [Programming in R](http://zoonek2.free.fr/UNIX/48_R/02.html) (Vincent Zoonekynd)
* [Peter's R Programming Pages](http://www2.warwick.ac.uk/fac/sci/moac/people/students/peter_cock/r), University of Warwick
* [Rtips](http://pj.freefaculty.org/R/statsRus.html), Paul Johnsson, University of Kansas
* [R for Programmers](http://heather.cs.ucdavis.edu/~matloff/r.html), Norm Matloff, UC Davis
* [High-Performance R](http://www.statistik.uni-dortmund.de/useR-2008/tutorials/useR2008introhighperfR.pdf), Dirk Eddelbuettel tutorial presented at useR-2008
* [C/C++ level programming for R](http://www.stat.harvard.edu/ccr2005/index.html), Gopi Goswami


# Control Structures

## Important Operators

### Comparison operators

* `==` (equal)
* `!=` (not equal)
* `>` (greater than)
* `>=` (greater than or equal)
* `<` (less than)
* `<=` (less than or equal)

### Logical operators
		
* `&` (and)
* `|` (or) 
* `!` (not)

## Conditional Executions: `if` Statements

An `if` statement operates on length-one logical vectors.

__Syntax__
```{r if_statement, eval=FALSE}
if(TRUE) { 
	statements_1 
} else { 
	statements_2 
}
```

__Example__
```{r if_statement_example, eval=TRUE}
if(1==0) { 
	print(1) 
} else { 
	print(2) 
}
```

## Conditional Executions: `ifelse` Statements`

The `ifelse` statement operates on vectors.

__Syntax__
```{r ifelse_statement, eval=FALSE}
ifelse(test, true_value, false_value)
```
__Example__
```{r ifelse_statement_example, eval=TRUE}
x <- 1:10 
ifelse(x<5, x, 0)
```

# Loops

## `for` loop

`for` loops iterate over elements of a looping vector.

__Syntax__
```{r for_loop, eval=FALSE}
for(variable in sequence) { 
	statements 
}
```
__Example__
```{r for_loop_example, eval=TRUE}
mydf <- iris
myve <- NULL
for(i in seq(along=mydf[,1])) {
	myve <- c(myve, mean(as.numeric(mydf[i,1:3])))
}
myve[1:8]
```

__Note:__ Inject into objecs is much faster than append approach with `c`, `cbind`, etc.

__Example__
```{r for_loop_inject_example, eval=TRUE}
myve <- numeric(length(mydf[,1]))
for(i in seq(along=myve)) {
	myve[i] <- mean(as.numeric(mydf[i,1:3]))
}
myve[1:8]
```

### Conditional Stop of Loops

The `stop` function can be used to break out of a loop (or a function) when a condition becomes `TRUE`. In addition, an error message will be printed.

__Example__
```{r for_loop_stop_example, eval=FALSE}
x <- 1:10
z <- NULL
for(i in seq(along=x)) { 
	if(x[i] < 5) { 
		z <- c(z, x[i]-1)  
	} else { 
		stop("values need to be < 5") 
	}
}
```

## `while` loop

Iterates as long as a condition is true.

__Syntax__
```{r while_loop, eval=FALSE}
while(condition) {
	statements
}
```

__Example__
```{r while_loop_example, eval=TRUE}
z <- 0
while(z<5) { 
	z <- z + 2
	print(z)  
}
```

## The `apply` Function Family

### `apply`

__Syntax__
```{r apply_loop, eval=FALSE}
apply(X, MARGIN, FUN, ARGs)
```

__Arguments__

* `X`: `array`, `matrix` or `data.frame`
* `MARGIN`: `1` for rows, `2` for columns
* `FUN`: one or more functions
* `ARGs`: possible arguments for functions

__Example__
```{r apply_loop_example, eval=TRUE}
apply(iris[1:8,1:3], 1, mean)
```

### `tapply`

Applies a function to vector components that are defined by a factor.

__Syntax__
```{r tapply_loop, eval=FALSE}
tapply(vector, factor, FUN)
```

__Example__
```{r tapply_loop_example, eval=TRUE}
iris[1:2,]
tapply(iris$Sepal.Length, iris$Species, mean)
```

### `sapply` and `lapply`

Both apply a function to vector or list objects. The `lapply` function always returns a list object, while `sapply` returns `vector` or `matrix` objects when it is possible. 

__Examples__
```{r lapply_loop_example, eval=TRUE}
x <- list(a = 1:10, beta = exp(-3:3), logic = c(TRUE,FALSE,FALSE,TRUE))
lapply(x, mean)
sapply(x, mean)
```

Often used in combination with a function definition:
```{r lapply_loop_fct_example, eval=TRUE}
lapply(names(x), function(x) mean(x))
sapply(names(x), function(x) mean(x))
```

# Session Info

```{r sessionInfo}
sessionInfo()
```

# References



<div class="tags">
    
</div>



</div>



<footer>
            <div class="row">
                <div class="col-lg-12 footer">
               &copy;2016 UC Riverside. All rights reserved. <br />
 Site last generated: Apr 13, 2016 <br />
<p><img src="../common_images/company_logo.png/></p>
                </div>
            </div>
</footer>

            </div><!-- /.row -->

    </div>    <!-- /.container -->

</body>


</html>

