<!DOCTYPE html>
  <head>

 <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="">
<meta name="keywords" content=" ">
<title>Analysis Routine  | GEN242</title>
<link rel="stylesheet" type="text/css" href="../css/syntax.css">
<link rel="stylesheet" type="text/css" href="../css/font-awesome.min.css">
<!--<link rel="stylesheet" type="text/css" href="../css/bootstrap.min.css">-->
<link rel="stylesheet" type="text/css" href="../css/modern-business.css">
<link rel="stylesheet" type="text/css" href="../css/lavish-bootstrap.css">
<link rel="stylesheet" type="text/css" href="../css/customstyles.css">
<link rel="stylesheet" type="text/css" href="../css/theme-blue.css">
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-cookie/1.4.1/jquery.cookie.min.js"></script>
<script src="../js/jquery.navgoco.min.js"></script>
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/js/bootstrap.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/anchor-js/2.0.0/anchor.min.js"></script>
<script src="../js/toc.js"></script>
<script src="../js/customscripts.js"></script>
<link rel="shortcut icon" href="../common_images/favicon.ico" type="image/x-icon">
<!-- HTML5 Shim and Respond.js IE8 support of HTML5 elements and media queries -->
<!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
<script src="https://oss.maxcdn.com/libs/respond.js/1.4.2/respond.min.js"></script>
<![endif]-->







 

<script>
  $(function () {
      $('[data-toggle="tooltip"]').tooltip()
  })
</script>



  </head>

<body>

   <!-- Navigation -->
<nav class="navbar navbar-inverse navbar-fixed-top" role="navigation">
    <div class="container topnavlinks">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <a class="fa fa-home fa-lg navbar-brand" href="home.html">&nbsp;<span class="projectTitle"> Data Analysis in Genome Biology</span></a>

        </div>
        <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
            <ul class="nav navbar-nav navbar-right">
                <!-- entries without drop-downs appear here -->
                <!-- conditional logic to control which topnav appears for the audience defined in the configuration file.-->
                



















                
                
                
                
                <li><a href="https://piazza.com/class/ilbslqb3j2ovy" target="_blank">Piazza</a></li>
                
                
                
                
                
                <li><a href="https://github.com/tgirke/GEN242" target="_blank">Github Repo</a></li>
                
                
                
                


                <!-- entries with drop-downs appear here -->
                <!-- conditional logic to control which topnav appears for the audience defined in the configuration file.-->

                <li class="dropdown">
                    
                    
                    
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Resources<b class="caret"></b></a>
                    <ul class="dropdown-menu">
                        
                        
                        
                        <li><a href="https://cran.r-project.org/" target="_blank">CRAN</a></li>
                        
                        
                        
                        
                        
                        <li><a href="http://bioconductor.org/" target="_blank">Bioconductor</a></li>
                        
                        
                        
                        
                        
                        <li><a href="https://guides.github.com/activities/hello-world/" target="_blank">GitHub</a></li>
                        
                        
                        
                        
                        
                        <li><a href="http://jekyllrb.com/" target="_blank">Jekyll</a></li>
                        
                        
                        
                        
                        
                        <li><a href="http://girke.bioinformatics.ucr.edu/" target="_blank">Girke Lab</a></li>
                        
                        
                        
                        
                        
                        <li><a href="http://idratherbewriting.com/documentation-theme-jekyll" target="_blank">Jekyll Doc Theme</a></li>
                        
                        
                        
                        

                    </ul>
                </li>
                
                


                <!-- special insertion -->

               
                
                
                <!-- Send feedback function -->
<script>
function SendLinkByMail(href) {
var subject= "GEN242 feedback";
var body = "I have some feedback about the Analysis Routine page: ";
body += window.location.href;
body += "";
var uri = "mailto:?subject=";
uri += encodeURIComponent(subject);
uri += "&body=";
uri += encodeURIComponent(body);
window.location.href = uri;
}
</script>

<li><a href="javascript:(function()%7BSendLinkByMail()%3B%7D)()%3B" target="_blank"><i class="fa fa-envelope-o"></i> Feedback</a></li>


                <!--uncomment this block if you want simple search instead of algolia-->
                <li>
                     <!--start search-->
                    <div id="search-demo-container">
                        <input type="text" id="search-input" placeholder="search...">
                        <ul id="results-container"></ul>
                    </div>
                    <script src="../js/jekyll-search.js" type="text/javascript"></script>
                    <script type="text/javascript">
                        SimpleJekyllSearch.init({
                            searchInput: document.getElementById('search-input'),
                            resultsContainer: document.getElementById('results-container'),
                            dataSource: '../search.json',
                            searchResultTemplate: '<li><a href="{url}" title="Analysis Routine">{title}</a></li>',
                        noResultsText: 'No results found.',
                                limit: 10,
                                fuzzy: true,
                        })
                    </script>
                     <!--end search-->
                </li>

            
        </div>
        <!-- /.container -->
</nav>


    <!-- Page Content -->
    <div class="container">
        <div class="col-lg-12">&nbsp;</div>


<!-- Content Row -->
<div class="row">
    <!-- Sidebar Column -->
    <div class="col-md-3">

        <script>

            $(document).ready(function() {
                // Initialize navgoco with default options
                $("#mysidebar").navgoco({
                    caretHtml: '',
                    accordion: true,
                    openClass: 'active', // open
                    save: false, // leave false or nav highlighting doesn't work right
                    cookie: {
                        name: 'navgoco',
                        expires: false,
                        path: '/'
                    },
                    slide: {
                        duration: 400,
                        easing: 'swing'
                    }
                });

                $("#collapseAll").click(function(e) {
                    e.preventDefault();
                    $("#mysidebar").navgoco('toggle', false);
                });

                $("#expandAll").click(function(e) {
                    e.preventDefault();
                    $("#mysidebar").navgoco('toggle', true);
                });

            });

        </script>


        



















        <ul id="mysidebar" class="nav">

            <span class="siteTagline">GEN242</span>
            <span class="versionTagline">Spring 2016</span>

            
            
            
                
            
            <li><a href="#">Home</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/home.html">Introduction</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_syllabus.html">Syllabus</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Internal</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_internal.html">Internal Resources</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Schedule</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_schedule.html">Course Schedule</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Slides</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_01.html">Course Introduction</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_02.html">Molbiol for Bioinformatics</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_03.html">Databases and Software</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_04.html">DNA Sequencing</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_05.html">Introduction to R</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_06.html">Sequence Alignments</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_07.html">Programming in R</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Homework</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_homework_01.html">HW1 - Online Tools</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_homework_02.html">HW2 - Biocluster/Linux</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_homework_03.html">HW3 - Introduction to R</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_homework_04.html">HW4 - Pairwise Alignments</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_09.html">HW5 - Programming in R</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">GitHub</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_tutorial_01.html">GitHub</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Biocluster</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_tutorial_02.html">Biocluster</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Introduction to R</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_01.html">1. Overview</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_02.html">2. R Package Repositories</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_03.html">3. Installation of R and Add-on Packages</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_04.html">4. Getting Around</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_05.html">5. Basic Syntax</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_06.html">6. Data Types</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_07.html">7. Data objects</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_08.html">8. Important Utilities</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_09.html">9. Operators and Calculations</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_10.html">10. Reading and Writing External Data</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_11.html">11. Useful R Functions</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_12.html">12. Graphics in R</a></li>
                    

                    

                    
                    
                    
                    
                    <li class="active"><a href="../mydoc/mydoc_Rbasics_13.html">13. Analysis Routine</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_14.html">14. Session Info</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_15.html">15. References</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Programming in R</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_01.html">1. Overview</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_02.html">2. Control Structures</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_03.html">3. Loops</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_04.html">4. Functions</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_05.html">5. Useful Utilities</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_06.html">6. Running R Scripts</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_07.html">7. Building R Packages</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_08.html">8. Programming Exercises</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_09.html">9. Homework 5</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_10.html">10. Session Info</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_11.html">11. References</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Projects</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_project_01.html">Project 1</a></li>
                    

                    

                    
                    
                </ul>
                
                
                


                <!-- if you aren't using the accordion, uncomment this block:

                     <p class="external">
                         <a href="#" id="collapseAll">Collapse All</a> | <a href="#" id="expandAll">Expand All</a>
                     </p>
                 -->

    </div>
    <!-- this highlights the active parent class in the navgoco sidebar. this is critical so that the parent expands when you're viewing a page. This must appear below the sidebar code above. Otherwise, if placed inside customscripts.js, the script runs before the sidebar code runs and the class never gets inserted.-->
    <script>$("li.active").parents('li').toggleClass("active");</script>


            <!-- Content Column -->
            <div class="col-md-9">
                
                <div class="post-header">
   <h1 class="post-title-main">Analysis Routine</h1>
</div>

<div class="post-content">

   

<!-- this handles the automatic toc. use ## for subheads to auto-generate the on-page minitoc. if you use html tags, you must supply an ID for the heading element in order for it to appear in the minitoc. -->
<script>
$( document ).ready(function() {
  // Handler for .ready() called.

$('#toc').toc({ minimumHeaders: 0, listType: 'ul', showSpeed: 0, headers: 'h2,h3,h4' });

/* this offset helps account for the space taken up by the floating toolbar. */
$('#toc').on('click', 'a', function() {
  var target = $(this.getAttribute('href'))
    , scroll_target = target.offset().top

  $(window).scrollTop(scroll_target - 10);
  return false
})
  
});
</script>


<div id="toc"></div>


<!--
    
    <a target="_blank" href="https://github.com/tgirke/manuals/tree/master/mydoc/" class="btn btn-default githubEditButton" role="button"><i class="fa fa-github fa-lg"></i> Edit me</a>
    
-->
<h2 id="overview">Overview</h2>

<p>The following exercise introduces a variety of useful data analysis utilities in R.</p>

<h2 id="analysis-routine-data-import">Analysis Routine: Data Import</h2>

<ul>
  <li>
    <p><strong>Step 1</strong>: To get started with this exercise, direct your R session to a dedicated workshop directory and download into this directory the following sample tables. Then import the files into Excel and save them as tab delimited text files.</p>

    <ul>
      <li><a href="http://faculty.ucr.edu/~tgirke/Documents/R_BioCond/Samples/MolecularWeight_tair7.xls">MolecularWeight_tair7.xls</a></li>
      <li><a href="http://faculty.ucr.edu/~tgirke/Documents/R_BioCond/Samples/TargetP_analysis_tair7.xls">TargetP_analysis_tair7.xls</a></li>
    </ul>
  </li>
</ul>

<p><strong>Import the tables into R</strong></p>

<p>Import molecular weight table</p>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">my_mw</span> <span class="o">&lt;-</span> <span class="n">read.delim</span><span class="p">(</span><span class="n">file</span><span class="o">=</span><span class="s2">"MolecularWeight_tair7.xls"</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="n">T</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"\t"</span><span class="p">)</span> 
<span class="n">my_mw</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">,]</span></code></pre></div>

<p>Import subcelluar targeting table</p>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">my_target</span> <span class="o">&lt;-</span> <span class="n">read.delim</span><span class="p">(</span><span class="n">file</span><span class="o">=</span><span class="s2">"TargetP_analysis_tair7.xls"</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="n">T</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"\t"</span><span class="p">)</span> 
<span class="n">my_target</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">,]</span></code></pre></div>

<p>Online import of molecular weight table</p>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">my_mw</span> <span class="o">&lt;-</span> <span class="n">read.delim</span><span class="p">(</span><span class="n">file</span><span class="o">=</span><span class="s2">"http://faculty.ucr.edu/~tgirke/Documents/R_BioCond/Samples/MolecularWeight_tair7.xls"</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="n">T</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"\t"</span><span class="p">)</span> 
<span class="n">my_mw</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">,]</span></code></pre></div>

<div class="highlight"><pre><code class="language-txt" data-lang="txt">##   Sequence.id Molecular.Weight.Da. Residues
## 1 AT1G08520.1                83285      760
## 2 AT1G08530.1                27015      257</code></pre></div>

<p>Online import of subcelluar targeting table</p>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">my_target</span> <span class="o">&lt;-</span> <span class="n">read.delim</span><span class="p">(</span><span class="n">file</span><span class="o">=</span><span class="s2">"http://faculty.ucr.edu/~tgirke/Documents/R_BioCond/Samples/TargetP_analysis_tair7.xls"</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="n">T</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"\t"</span><span class="p">)</span> 
<span class="n">my_target</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">,]</span></code></pre></div>

<div class="highlight"><pre><code class="language-txt" data-lang="txt">##      GeneName Loc   cTP   mTP    SP other
## 1 AT1G08520.1   C 0.822 0.137 0.029 0.039
## 2 AT1G08530.1   C 0.817 0.058 0.010 0.100</code></pre></div>

<h2 id="merging-data-frames">Merging Data Frames</h2>

<ul>
  <li><strong>Step 2</strong>: Assign uniform gene ID column titles</li>
</ul>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">colnames</span><span class="p">(</span><span class="n">my_target</span><span class="p">)[</span><span class="m">1</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="s2">"ID"</span>
<span class="n">colnames</span><span class="p">(</span><span class="n">my_mw</span><span class="p">)[</span><span class="m">1</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="s2">"ID"</span></code></pre></div>

<ul>
  <li><strong>Step 3</strong>: Merge the two tables based on common ID field</li>
</ul>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">my_mw_target</span> <span class="o">&lt;-</span> <span class="n">merge</span><span class="p">(</span><span class="n">my_mw</span><span class="p">,</span> <span class="n">my_target</span><span class="p">,</span> <span class="n">by.x</span><span class="o">=</span><span class="s2">"ID"</span><span class="p">,</span> <span class="n">by.y</span><span class="o">=</span><span class="s2">"ID"</span><span class="p">,</span> <span class="n">all.x</span><span class="o">=</span><span class="n">T</span><span class="p">)</span></code></pre></div>

<ul>
  <li><strong>Step 4</strong>: Shorten one table before the merge and then remove the non-matching rows (NAs) in the merged file</li>
</ul>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">my_mw_target2a</span> <span class="o">&lt;-</span> <span class="n">merge</span><span class="p">(</span><span class="n">my_mw</span><span class="p">,</span> <span class="n">my_target</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">40</span><span class="p">,],</span> <span class="n">by.x</span><span class="o">=</span><span class="s2">"ID"</span><span class="p">,</span> <span class="n">by.y</span><span class="o">=</span><span class="s2">"ID"</span><span class="p">,</span> <span class="n">all.x</span><span class="o">=</span><span class="n">T</span><span class="p">)</span>  <span class="c1"># To remove non-matching rows, use the argument setting 'all=F'.
</span><span class="n">my_mw_target2</span> <span class="o">&lt;-</span> <span class="n">na.omit</span><span class="p">(</span><span class="n">my_mw_target2a</span><span class="p">)</span> <span class="err">#</span> <span class="n">Removes</span> <span class="n">rows</span> <span class="n">containing</span> <span class="s2">"NAs"</span> <span class="p">(</span><span class="n">non</span><span class="o">-</span><span class="n">matching</span> <span class="n">rows</span><span class="p">)</span><span class="err">.</span></code></pre></div>

<ul>
  <li><strong>Homework 3D</strong>: How can the merge function in the previous step be executed so that only the common rows among the two data frames are returned? Prove that both methods - the two step version with <code>na.omit</code> and your method - return identical results.</li>
  <li><strong>Homework 3E</strong>: Replace all <code>NAs</code> in the data frame <code>my_mw_target2a</code> with zeros.</li>
</ul>

<h2 id="filtering-data">Filtering Data</h2>

<ul>
  <li><strong>Step 5</strong>: Retrieve all records with a value of greater than 100,000 in ‘MW’ column and ‘C’ value in ‘Loc’ column (targeted to chloroplast).</li>
</ul>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">query</span> <span class="o">&lt;-</span> <span class="n">my_mw_target</span><span class="p">[</span><span class="n">my_mw_target</span><span class="p">[,</span> <span class="m">2</span><span class="p">]</span> <span class="o">&gt;</span> <span class="m">100000</span> <span class="o">&amp;</span> <span class="n">my_mw_target</span><span class="p">[,</span> <span class="m">4</span><span class="p">]</span> <span class="o">==</span> <span class="s2">"C"</span><span class="p">,</span> <span class="p">]</span> 
<span class="n">query</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">,</span> <span class="p">]</span></code></pre></div>

<div class="highlight"><pre><code class="language-txt" data-lang="txt">##              ID Molecular.Weight.Da. Residues Loc   cTP   mTP    SP other
## 219 AT1G02730.1               132588     1181   C 0.972 0.038 0.008 0.045
## 243 AT1G02890.1               136825     1252   C 0.748 0.529 0.011 0.013
## 281 AT1G03160.1               100732      912   C 0.871 0.235 0.011 0.007
## 547 AT1G05380.1               126360     1138   C 0.740 0.099 0.016 0.358</code></pre></div>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">dim</span><span class="p">(</span><span class="n">query</span><span class="p">)</span></code></pre></div>

<div class="highlight"><pre><code class="language-txt" data-lang="txt">## [1] 170   8</code></pre></div>

<ul>
  <li><strong>Homework 3F</strong>: How many protein entries in the <code>my</code>_mw<code>_target</code> data frame have a MW of greater then 4,000 and less then 5,000. Subset the data frame accordingly and sort it by MW to check that your result is correct.</li>
</ul>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">query2</span> <span class="o">&lt;-</span> <span class="n">my_mw_target</span><span class="p">[</span><span class="n">my_mw_target</span><span class="p">[,</span> <span class="m">2</span><span class="p">]</span> <span class="o">&gt;</span> <span class="m">4000</span> <span class="o">&amp;</span> <span class="n">my_mw_target</span><span class="p">[,</span> <span class="m">2</span><span class="p">]</span> <span class="o">&lt;</span> <span class="m">5000</span><span class="p">,</span> <span class="p">]</span> 
<span class="n">dim</span><span class="p">(</span><span class="n">query2</span><span class="p">)</span></code></pre></div>

<div class="highlight"><pre><code class="language-txt" data-lang="txt">## [1] 38  8</code></pre></div>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">query2</span><span class="p">[</span><span class="n">order</span><span class="p">(</span><span class="n">query2</span><span class="p">[,</span><span class="m">2</span><span class="p">]),]</span></code></pre></div>

<div class="highlight"><pre><code class="language-txt" data-lang="txt">##                ID Molecular.Weight.Da. Residues Loc   cTP   mTP    SP other
## 13211 AT2G47660.1                 4043       35   _ 0.153 0.099 0.140 0.909
## 13808 AT3G05080.1                 4068       35   * 0.091 0.200 0.231 0.704
## 17007 AT3G42090.1                 4113       37   * 0.089 0.071 0.345 0.825
## 31743 ATCG00550.1                 4117       40   S 0.003 0.065 0.953 0.366
## 24539 AT5G02502.1                 4133       35   S 0.006 0.013 0.995 0.106
## 31739 ATCG00510.1                 4134       37   S 0.005 0.025 0.992 0.066
## 31713 ATCG00080.1                 4168       36   S 0.012 0.030 0.991 0.044
## 31748 ATCG00600.1                 4204       37   S 0.010 0.054 0.981 0.058
## 27714 AT5G35480.1                 4298       38   * 0.049 0.157 0.398 0.672
## 192   AT1G02490.1                 4337       38   * 0.094 0.085 0.438 0.695
## 30410 AT5G57567.1                 4356       41   S 0.066 0.044 0.843 0.331
## 14816 AT3G13240.1                 4366       40   * 0.216 0.075 0.261 0.784
## 24276 AT4G39403.1                 4390       36   * 0.009 0.713 0.363 0.101
## 5668  AT1G59535.1                 4404       39   _ 0.122 0.132 0.135 0.895
## 31745 ATCG00570.1                 4424       39   S 0.005 0.309 0.718 0.213
## 31762 ATCG00760.1                 4461       37   * 0.110 0.542 0.074 0.334
## 31744 ATCG00560.1                 4470       38   S 0.016 0.148 0.737 0.321
## 823   AT1G07610.1                 4495       45   * 0.656 0.044 0.248 0.589
## 288   AT1G03200.1                 4573       41   * 0.089 0.358 0.119 0.647
## 822   AT1G07600.1                 4580       45   * 0.545 0.046 0.234 0.713
## 16351 AT3G25717.1                 4620       40   M 0.022 0.890 0.039 0.208
## 292   AT1G03240.1                 4644       42   * 0.114 0.317 0.132 0.597
## 25923 AT5G14560.1                 4681       40   * 0.091 0.268 0.122 0.779
## 1568  AT1G13245.1                 4719       41   * 0.024 0.746 0.067 0.306
## 31756 ATCG00700.1                 4722       43   S 0.011 0.048 0.972 0.099
## 6552  AT1G67148.1                 4734       45   _ 0.203 0.096 0.126 0.897
## 6566  AT1G67265.1                 4738       40   * 0.059 0.283 0.098 0.738
## 26710 AT5G21020.2                 4744       42   S 0.032 0.069 0.951 0.079
## 30432 AT5G57730.1                 4763       42   * 0.103 0.308 0.126 0.511
## 14906 AT3G13857.1                 4778       42   * 0.244 0.476 0.303 0.018
## 27114 AT5G24980.1                 4839       42   S 0.004 0.030 0.991 0.076
## 13050 AT2G46390.1                 4884       46   S 0.003 0.155 0.957 0.075
## 16657 AT3G28120.1                 4897       41   S 0.006 0.075 0.970 0.145
## 3797  AT1G32670.1                 4926       44   * 0.233 0.607 0.129 0.103
## 13470 AT3G02390.1                 4933       43   _ 0.135 0.118 0.147 0.877
## 9128  AT2G14460.1                 4950       41   * 0.143 0.165 0.139 0.784
## 1389  AT1G11785.1                 4983       46   _ 0.142 0.110 0.146 0.901
## 23211 AT4G31030.1                 4983       44   * 0.041 0.133 0.352 0.795</code></pre></div>

<h2 id="string-substitutions">String Substitutions</h2>

<ul>
  <li><strong>Step 6</strong>: Use a regular expression in a substitute function to generate a separate ID column that lacks the gene model extensions.
«label=Exercise 4.7, eval=TRUE, echo=TRUE, keep.source=TRUE»=</li>
</ul>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">my_mw_target3</span> <span class="o">&lt;-</span> <span class="n">data.frame</span><span class="p">(</span><span class="n">loci</span><span class="o">=</span><span class="n">gsub</span><span class="p">(</span><span class="s2">"\\..*"</span><span class="p">,</span> <span class="s2">""</span><span class="p">,</span> <span class="n">as.character</span><span class="p">(</span><span class="n">my_mw_target</span><span class="p">[,</span><span class="m">1</span><span class="p">]),</span> <span class="n">perl</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">),</span> <span class="n">my_mw_target</span><span class="p">)</span>
<span class="n">my_mw_target3</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">3</span><span class="p">,</span><span class="m">1</span><span class="o">:</span><span class="m">8</span><span class="p">]</span></code></pre></div>

<div class="highlight"><pre><code class="language-txt" data-lang="txt">##        loci          ID Molecular.Weight.Da. Residues Loc  cTP   mTP    SP
## 1 AT1G01010 AT1G01010.1                49426      429   _ 0.10 0.090 0.075
## 2 AT1G01020 AT1G01020.1                28092      245   * 0.01 0.636 0.158
## 3 AT1G01020 AT1G01020.2                21711      191   * 0.01 0.636 0.158</code></pre></div>

<ul>
  <li><strong>Homework 3G</strong>: Retrieve those rows in <code>my_mw_target3</code> where the second column contains the following identifiers: <code>c("AT5G52930.1", "AT4G18950.1", "AT1G15385.1", "AT4G36500.1", "AT1G67530.1")</code>. Use the <code>%in%</code> function for this query. As an alternative approach, assign the second column to the row index of the data frame and then perform the same query again using the row index. Explain the difference of the two methods.</li>
</ul>

<h2 id="calculations-on-data-frames">Calculations on Data Frames</h2>

<ul>
  <li><strong>Step 7</strong>: Count the number of duplicates in the loci column with the <code>table</code> function and append the result to the data frame with the <code>cbind</code> function.</li>
</ul>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">mycounts</span> <span class="o">&lt;-</span> <span class="n">table</span><span class="p">(</span><span class="n">my_mw_target3</span><span class="p">[,</span><span class="m">1</span><span class="p">])[</span><span class="n">my_mw_target3</span><span class="p">[,</span><span class="m">1</span><span class="p">]]</span>
<span class="n">my_mw_target4</span> <span class="o">&lt;-</span> <span class="n">cbind</span><span class="p">(</span><span class="n">my_mw_target3</span><span class="p">,</span> <span class="n">Freq</span><span class="o">=</span><span class="n">mycounts</span><span class="p">[</span><span class="n">as.character</span><span class="p">(</span><span class="n">my_mw_target3</span><span class="p">[,</span><span class="m">1</span><span class="p">])])</span></code></pre></div>

<ul>
  <li><strong>Step 8</strong>: Perform a vectorized devision of columns 3 and 4 (average AA weight per protein)</li>
</ul>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">data.frame</span><span class="p">(</span><span class="n">my_mw_target4</span><span class="p">,</span> <span class="n">avg_AA_WT</span><span class="o">=</span><span class="p">(</span><span class="n">my_mw_target4</span><span class="p">[,</span><span class="m">3</span><span class="p">]</span> <span class="o">/</span> <span class="n">my_mw_target4</span><span class="p">[,</span><span class="m">4</span><span class="p">]))[</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">,</span><span class="m">5</span><span class="o">:</span><span class="m">11</span><span class="p">]</span></code></pre></div>

<div class="highlight"><pre><code class="language-txt" data-lang="txt">##   Loc  cTP   mTP    SP other Freq avg_AA_WT
## 1   _ 0.10 0.090 0.075 0.925    1  115.2121
## 2   * 0.01 0.636 0.158 0.448    2  114.6612</code></pre></div>

<ul>
  <li><strong>Step 9</strong>: Calculate for each row the mean and standard deviation across several columns</li>
</ul>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">mymean</span> <span class="o">&lt;-</span> <span class="n">apply</span><span class="p">(</span><span class="n">my_mw_target4</span><span class="p">[,</span><span class="m">6</span><span class="o">:</span><span class="m">9</span><span class="p">],</span> <span class="m">1</span><span class="p">,</span> <span class="n">mean</span><span class="p">)</span>
<span class="n">mystdev</span> <span class="o">&lt;-</span> <span class="n">apply</span><span class="p">(</span><span class="n">my_mw_target4</span><span class="p">[,</span><span class="m">6</span><span class="o">:</span><span class="m">9</span><span class="p">],</span> <span class="m">1</span><span class="p">,</span> <span class="n">sd</span><span class="p">,</span> <span class="n">na.rm</span><span class="o">=</span><span class="n">TRUE</span><span class="p">)</span>
<span class="n">data.frame</span><span class="p">(</span><span class="n">my_mw_target4</span><span class="p">,</span> <span class="n">mean</span><span class="o">=</span><span class="n">mymean</span><span class="p">,</span> <span class="n">stdev</span><span class="o">=</span><span class="n">mystdev</span><span class="p">)[</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">,</span><span class="m">5</span><span class="o">:</span><span class="m">12</span><span class="p">]</span></code></pre></div>

<div class="highlight"><pre><code class="language-txt" data-lang="txt">##   Loc  cTP   mTP    SP other Freq   mean     stdev
## 1   _ 0.10 0.090 0.075 0.925    1 0.2975 0.4184595
## 2   * 0.01 0.636 0.158 0.448    2 0.3130 0.2818912</code></pre></div>

<h2 id="plotting-example">Plotting Example</h2>

<ul>
  <li><strong>Step 10</strong>: Generate scatter plot columns: ‘MW’ and ‘Residues’</li>
</ul>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">plot</span><span class="p">(</span><span class="n">my_mw_target4</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">500</span><span class="p">,</span><span class="m">3</span><span class="o">:</span><span class="m">4</span><span class="p">],</span> <span class="n">col</span><span class="o">=</span><span class="s2">"red"</span><span class="p">)</span></code></pre></div>

<p><img src="Rbasics_files/plot_example-1.png" alt="" /></p>

<h2 id="export-results-and-run-entire-exercise-as-script">Export Results and Run Entire Exercise as Script</h2>

<ul>
  <li><strong>Step 11</strong>: Write the data frame <code>my_mw_target4</code> into a tab-delimited text file and inspect it in Excel.</li>
</ul>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">write.table</span><span class="p">(</span><span class="n">my_mw_target4</span><span class="p">,</span> <span class="n">file</span><span class="o">=</span><span class="s2">"my_file.xls"</span><span class="p">,</span> <span class="n">quote</span><span class="o">=</span><span class="n">F</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"\t"</span><span class="p">,</span> <span class="n">col.names</span> <span class="o">=</span> <span class="n">NA</span><span class="p">)</span></code></pre></div>

<ul>
  <li><strong>Homework 3H</strong>: Write all commands from this exercise into an R script named <code>exerciseRbasics.R</code>, or download it from <a href="http://faculty.ucr.edu/~tgirke/Documents/R_BioCond/My_R_Scripts/exerciseRbasics.R">here</a>. Then execute the script with the <code>source</code> function like this: <code>source("exerciseRbasics.R")</code>. This will run all commands of this exercise and generate the corresponding output files in the current working directory.</li>
</ul>

<div class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">source</span><span class="p">(</span><span class="s2">"exerciseRbasics.R"</span><span class="p">)</span></code></pre></div>



<div class="tags">
    
</div>



</div>



<footer>
            <div class="row">
                <div class="col-lg-12 footer">
               &copy;2016 UC Riverside. All rights reserved. <br />
<p>Page last updated:</span> Thu Apr  7 17:52:19 2016<br/> Site last generated: Apr 13, 2016 <br />
<p><img src="../common_images/company_logo.png/></p>
                </div>
            </div>
</footer>

            </div><!-- /.row -->

    </div>    <!-- /.container -->

</body>


</html>

