<!DOCTYPE html>
  <head>

 <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="">
<meta name="keywords" content=" ">
<title>Graphics and Data Visualization in R  | GEN242</title>
<link rel="stylesheet" type="text/css" href="../css/syntax.css">
<link rel="stylesheet" type="text/css" href="../css/font-awesome.min.css">
<!--<link rel="stylesheet" type="text/css" href="../css/bootstrap.min.css">-->
<link rel="stylesheet" type="text/css" href="../css/modern-business.css">
<link rel="stylesheet" type="text/css" href="../css/lavish-bootstrap.css">
<link rel="stylesheet" type="text/css" href="../css/customstyles.css">
<link rel="stylesheet" type="text/css" href="../css/theme-blue.css">
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-cookie/1.4.1/jquery.cookie.min.js"></script>
<script src="../js/jquery.navgoco.min.js"></script>
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/js/bootstrap.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/anchor-js/2.0.0/anchor.min.js"></script>
<script src="../js/toc.js"></script>
<script src="../js/customscripts.js"></script>
<script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
<link rel="shortcut icon" href="../common_images/favicon.ico" type="image/x-icon">
<!-- HTML5 Shim and Respond.js IE8 support of HTML5 elements and media queries -->
<!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
<script src="https://oss.maxcdn.com/libs/respond.js/1.4.2/respond.min.js"></script>
<![endif]-->







 


<script>
  $(function () {
      $('[data-toggle="tooltip"]').tooltip()
  })
</script>



  </head>

<body>

   <!-- Navigation -->
<nav class="navbar navbar-inverse navbar-fixed-top" role="navigation">
    <div class="container topnavlinks">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <a class="fa fa-home fa-lg navbar-brand" href="home.html">&nbsp;<span class="projectTitle"> Data Analysis in Genome Biology</span></a>

        </div>
        <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
            <ul class="nav navbar-nav navbar-right">
                <!-- entries without drop-downs appear here -->
                <!-- conditional logic to control which topnav appears for the audience defined in the configuration file.-->
                



















                
                
                
                
                <li><a href="https://piazza.com/class/ilbslqb3j2ovy" target="_blank">Piazza</a></li>
                
                
                
                
                
                <li><a href="https://github.com/tgirke/GEN242" target="_blank">Github Repo</a></li>
                
                
                
                


                <!-- entries with drop-downs appear here -->
                <!-- conditional logic to control which topnav appears for the audience defined in the configuration file.-->

                <li class="dropdown">
                    
                    
                    
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Resources<b class="caret"></b></a>
                    <ul class="dropdown-menu">
                        
                        
                        
                        <li><a href="https://cran.r-project.org/" target="_blank">CRAN</a></li>
                        
                        
                        
                        
                        
                        <li><a href="http://bioconductor.org/" target="_blank">Bioconductor</a></li>
                        
                        
                        
                        
                        
                        <li><a href="http://manuals.bioinformatics.ucr.edu/home/R_BioCondManual" target="_blank">R & Bioconductor Manual</a></li>
                        
                        
                        
                        
                        
                        <li><a href="https://guides.github.com/activities/hello-world/" target="_blank">GitHub</a></li>
                        
                        
                        
                        
                        
                        <li><a href="http://jekyllrb.com/" target="_blank">Jekyll</a></li>
                        
                        
                        
                        
                        
                        <li><a href="http://girke.bioinformatics.ucr.edu/" target="_blank">Girke Lab</a></li>
                        
                        
                        
                        
                        
                        <li><a href="http://idratherbewriting.com/documentation-theme-jekyll" target="_blank">Jekyll Doc Theme</a></li>
                        
                        
                        
                        

                    </ul>
                </li>
                
                


                <!-- special insertion -->

               
                
                
                <!-- Send feedback function -->
<script>
function SendLinkByMail(href) {
var subject= "GEN242 feedback";
var body = "I have some feedback about the Graphics and Data Visualization in R page: ";
body += window.location.href;
body += "";
var uri = "mailto:?subject=";
uri += encodeURIComponent(subject);
uri += "&body=";
uri += encodeURIComponent(body);
window.location.href = uri;
}
</script>

<li><a href="javascript:(function()%7BSendLinkByMail()%3B%7D)()%3B" target="_blank"><i class="fa fa-envelope-o"></i> Feedback</a></li>


                <!--uncomment this block if you want simple search instead of algolia-->
                <li>
                     <!--start search-->
                    <div id="search-demo-container">
                        <input type="text" id="search-input" placeholder="search...">
                        <ul id="results-container"></ul>
                    </div>
                    <script src="../js/jekyll-search.js" type="text/javascript"></script>
                    <script type="text/javascript">
                        SimpleJekyllSearch.init({
                            searchInput: document.getElementById('search-input'),
                            resultsContainer: document.getElementById('results-container'),
                            dataSource: '../search.json',
                            searchResultTemplate: '<li><a href="{url}" title="Graphics and Data Visualization in R">{title}</a></li>',
                        noResultsText: 'No results found.',
                                limit: 10,
                                fuzzy: true,
                        })
                    </script>
                     <!--end search-->
                </li>

            
        </div>
        <!-- /.container -->
</nav>


    <!-- Page Content -->
    <div class="container">
        <div class="col-lg-12">&nbsp;</div>


<!-- Content Row -->
<div class="row">
    <!-- Sidebar Column -->
    <div class="col-md-3">

        <script>

            $(document).ready(function() {
                // Initialize navgoco with default options
                $("#mysidebar").navgoco({
                    caretHtml: '',
                    accordion: true,
                    openClass: 'active', // open
                    save: false, // leave false or nav highlighting doesn't work right
                    cookie: {
                        name: 'navgoco',
                        expires: false,
                        path: '/'
                    },
                    slide: {
                        duration: 400,
                        easing: 'swing'
                    }
                });

                $("#collapseAll").click(function(e) {
                    e.preventDefault();
                    $("#mysidebar").navgoco('toggle', false);
                });

                $("#expandAll").click(function(e) {
                    e.preventDefault();
                    $("#mysidebar").navgoco('toggle', true);
                });

            });

        </script>


        



















        <ul id="mysidebar" class="nav">

            <span class="siteTagline">GEN242</span>
            <span class="versionTagline">Spring 2016</span>

            
            
            
                
            
            <li><a href="#">Home</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/home.html">Introduction</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_syllabus.html">Syllabus</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Internal</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_internal.html">Internal Resources</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Schedule</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_schedule.html">Course Schedule</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Lecture Slides</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_01.html">Course Introduction</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_02.html">Molbiol for Bioinformatics</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_03.html">Databases and Software</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_04.html">DNA Sequencing</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_05.html">Introduction to R</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_06.html">Sequence Alignments</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_07.html">Programming in R</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_08.html">Multiple Alignments</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_09.html">Short Read Alignments</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_10.html">NGS Analysis Basics</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_11.html">Gene Expression Analysis</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_12.html">NGS Workflows</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_13.html">ChIP-Seq Overview</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_14.html">VAR-Seq Overview</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_15.html">Sequence Assembly</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_16.html">Gene Annotation and Ontologies</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_17.html">Cluster Analysis</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Homework</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_homework_01.html">HW1 - Online Tools</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_homework_02.html">HW2 - Biocluster/Linux</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_homework_03.html">HW3 - Introduction to R</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_homework_04.html">HW4 - Pairwise Alignments</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_09.html">HW5 - Programming in R</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_08.html">HW6 - Sequence Analysis</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">GitHub</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_tutorial_01.html">GitHub</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Biocluster & Linux</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_tutorial_02.html">Basic Tutorial</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Introduction to R</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_01.html">1. Overview</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_02.html">2. R Package Repositories</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_03.html">3. Installation of R and Add-on Packages</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_04.html">4. Getting Around</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_05.html">5. Basic Syntax</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_06.html">6. Data Types</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_07.html">7. Data objects</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_08.html">8. Important Utilities</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_09.html">9. Operators and Calculations</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_10.html">10. Reading and Writing External Data</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_11.html">11. Useful R Functions</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_12.html">12. SQLite Databases</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_13.html">13. Graphics in R</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_14.html">14. Analysis Routine</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_15.html">15. R Markdown</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_16.html">16. Session Info</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_17.html">17. References</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Programming in R</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_01.html">1. Overview</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_02.html">2. Control Structures</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_03.html">3. Loops</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_04.html">4. Functions</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_05.html">5. Useful Utilities</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_06.html">6. Running R Scripts</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_07.html">7. Building R Packages</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_08.html">8. Programming Exercises</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_09.html">9. Homework 5</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_10.html">10. Session Info</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_11.html">11. References</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">NGS Analysis Basics</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_01.html">1. Overview</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_02.html">2. Package Requirements</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_03.html">3. Strings in R Base</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_04.html">4. Sequences in Bioconductor</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_05.html">5. NGS Sequences</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_06.html">6. Range Operations</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_07.html">7. Transcript Ranges</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_08.html">8. Homework 6</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_09.html">9. Session Info</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_10.html">10. References</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">NGS Workflow Overview</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeR_1.html">1. Introduction</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeR_2.html">2. Getting Started</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeR_3.html">3. Workflow overview</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeR_4.html">4. Workflow templates</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeR_5.html">5. Version information</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeR_6.html">6. References</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">RNA-Seq Workflow</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeRNAseq_01.html">1. Introduction</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeRNAseq_02.html">2. Sample definitions and environment settings</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeRNAseq_03.html">3. Read preprocessing</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeRNAseq_04.html">4. Alignments</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeRNAseq_05.html">5. Read quantification per annotation range</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeRNAseq_06.html">6. Analysis of differentially expressed genes with edgeR</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeRNAseq_07.html">7. GO term enrichment analysis of DEGs</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeRNAseq_08.html">8. Clustering and heat maps</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeRNAseq_09.html">9. Version Information</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeRNAseq_10.html">10. Funding</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeRNAseq_11.html">11. References</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">ChIP-Seq Workflow</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeChIPseq_01.html">1. Introduction</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeChIPseq_02.html">2. Load workflow environment</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeChIPseq_03.html">3. Read preprocessing</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeChIPseq_04.html">4. Alignments</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeChIPseq_05.html">5. Utilities for coverage data</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeChIPseq_06.html">6. Peak calling with MACS2</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeChIPseq_07.html">7. Annotate peaks with genomic context</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeChIPseq_08.html">8. Count reads overlapping peaks</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeChIPseq_09.html">9. Differential binding analysis</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeChIPseq_10.html">10. GO term enrichment analysis</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeChIPseq_11.html">11. Motif analysis</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeChIPseq_12.html">12. Version Information</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeChIPseq_13.html">13. Funding</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeChIPseq_14.html">14. References</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">VAR-Seq Workflow</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeVARseq_01.html">1. Introduction</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeVARseq_02.html">2. Load workflow environment</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeVARseq_03.html">3. Read preprocessing</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeVARseq_04.html">4. Alignments</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeVARseq_05.html">5. Variant calling</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeVARseq_06.html">6. Filter variants</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeVARseq_07.html">7. Annotate filtered variants</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeVARseq_08.html">8. Combine annotation results among samples</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeVARseq_09.html">9. Summary statistics of variants</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeVARseq_10.html">10. Venn diagram of variants</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeVARseq_11.html">11. Plot variants programmatically</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeVARseq_12.html">12. Version Information</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeVARseq_13.html">13. Funding</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_systemPipeVARseq_14.html">14. References</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Graphics and Visualization</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rgraphics_1.html">1. Overview</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rgraphics_2.html">2. Base Graphics</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rgraphics_3.html">3. Grid Graphics</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rgraphics_4.html">4. lattice Graphics</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rgraphics_5.html">5. ggplot2 Graphics</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rgraphics_6.html">6. Specialty Graphics</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rgraphics_7.html">7. Genome Graphics</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rgraphics_8.html">8. References</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Cluster Analysis</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rclustering_1.html">1. Introduction</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rclustering_2.html">2. Data Preprocessing</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rclustering_3.html">3. Clustering Algorithms</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rclustering_4.html">4. Clustering Exercises</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rclustering_5.html">5. Version Information</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rclustering_6.html">6. References</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Paper Presentations</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_paper_presentations.html">Paper Presentations</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Course Projects</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_project_00.html">Overview</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_project_01.html">ChIP-Seq1 - Motif Prediction</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_project_02.html">ChIP-Seq2 - Peak Callers</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_project_03.html">RNA-Seq1 - RNA-Seq Aligners</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_project_04.html">RNA-Seq2 - DEG Methods</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_project_05.html">VAR-Seq1 - Variant Callers</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_project_06.html">VAR-Seq2 - Functional Prediction</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_project_07.html">Project Data</a></li>
                    

                    

                    
                    
                </ul>
                
                
                


                <!-- if you aren't using the accordion, uncomment this block:

                     <p class="external">
                         <a href="#" id="collapseAll">Collapse All</a> | <a href="#" id="expandAll">Expand All</a>
                     </p>
                 -->

    </div>
    <!-- this highlights the active parent class in the navgoco sidebar. this is critical so that the parent expands when you're viewing a page. This must appear below the sidebar code above. Otherwise, if placed inside customscripts.js, the script runs before the sidebar code runs and the class never gets inserted.-->
    <script>$("li.active").parents('li').toggleClass("active");</script>


            <!-- Content Column -->
            <div class="col-md-9">
                
                <div class="post-header">
   <h1 class="post-title-main">Graphics and Data Visualization in R</h1>
</div>

<div class="post-content">

   

<!-- this handles the automatic toc. use ## for subheads to auto-generate the on-page minitoc. if you use html tags, you must supply an ID for the heading element in order for it to appear in the minitoc. -->
<script>
$( document ).ready(function() {
  // Handler for .ready() called.

$('#toc').toc({ minimumHeaders: 0, listType: 'ul', showSpeed: 0, headers: 'h2,h3,h4' });

/* this offset helps account for the space taken up by the floating toolbar. */
$('#toc').on('click', 'a', function() {
  var target = $(this.getAttribute('href'))
    , scroll_target = target.offset().top

  $(window).scrollTop(scroll_target - 10);
  return false
})
  
});
</script>


<div id="toc"></div>


<!--
    
    <a target="_blank" href="https://github.com/tgirke/manuals/tree/master/mydoc/" class="btn btn-default githubEditButton" role="button"><i class="fa fa-github fa-lg"></i> Edit me</a>
    
-->
<!--
%% \VignetteEngine{knitr::rmarkdown}
%\VignetteIndexEntry{Graphics in R}
%% \VignetteDepends{methods}
%% \VignetteKeywords{compute cluster, pipeline, reports}
-->

<!---
- Conversion from Rnw/tex to Rmd
    - Build vignette from Rnw 
    - Then run: pandoc -s Rgraphics.tex -o Rgraphics.text
    - Then fix things, usually a lot...

- Compile from command-line
echo "rmarkdown::render('Rgraphics.Rmd', clean=F)" | R -slave; R CMD Stangle Rgraphics.Rmd; Rscript ../md2jekyll.R Rgraphics.knit.md 15

- Commit to github
git commit -am "some edits"; git push -u origin master

- To customize font size and other style features, add this line to output section in preamble:  
    css: style.css
-->

<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script>

<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script>

<p>Note: the most recent version of this tutorial can be found <a href="https://htmlpreview.github.io/?https://github.com/tgirke/GEN242/blob/master/vignettes/15_Rgraphics/Rgraphics.html">here</a> and a slide show <a href="https://drive.google.com/file/d/0B-lLYVUOliJFUmFVa0N3Y0kxOVE/view?usp=sharing">here</a>.</p>

<h1 id="overview">Overview</h1>

<h2 id="graphics-in-r">Graphics in R</h2>

<ul>
  <li>Powerful environment for visualizing scientific data</li>
  <li>Integrated graphics and statistics infrastructure</li>
  <li>Publication quality graphics</li>
  <li>Fully programmable</li>
  <li>Highly reproducible</li>
  <li>Full <a href="http://www.latex-project.org/">LaTeX</a>, <a href="http://www.stat.auckland.ac.nz/~dscott/782/Sweave-manual-20060104.pdf">Sweave</a>, <a href="http://yihui.name/knitr/">knitr</a> and <a href="http://rmarkdown.rstudio.com/">R Markdown</a> support.
support</li>
  <li>Vast number of R packages with graphics utilities</li>
</ul>

<h2 id="documentation-on-graphics-in-r">Documentation on Graphics in R</h2>

<ul>
  <li>General
    <ul>
      <li><a href="http://cran.r-project.org/web/views/Graphics.html">Graphics Task Page</a></li>
      <li><a href="http://www.r-graph-gallery.com/">R Graph Gallery</a></li>
      <li><a href="http://bm2.genes.nig.ac.jp/RGM2/index.php">R Graphical Manual</a></li>
      <li><a href="http://www.stat.auckland.ac.nz/~paul/RGraphics/rgraphics.html">Paul Murrell’s book R (Grid) Graphics</a></li>
    </ul>
  </li>
  <li>Interactive graphics
    <ul>
      <li><a href="http://www.ggobi.org/"><code>rggobi</code> (GGobi)</a></li>
      <li><a href="http://www.iplots.org/"><code>iplots</code></a></li>
      <li><a href="http://rgl.neoscientists.org/gallery.shtml">Open GL (<code>rgl</code>)</a></li>
    </ul>
  </li>
</ul>

<h2 id="graphics-environments">Graphics Environments</h2>

<ul>
  <li>Viewing and savings graphics in R
    <ul>
      <li>On-screen graphics</li>
      <li>postscript, pdf, svg</li>
      <li>jpeg/png/wmf/tiff/…</li>
    </ul>
  </li>
  <li>Four major graphics environments
    <ul>
      <li>Low-level infrastructure
        <ul>
          <li>R Base Graphics (low- and high-level)</li>
          <li><code>grid</code>: <a href="http://www.stat.auckland.ac.nz/~paul/grid/grid.html">Manual</a>, <a href="http://www.stat.auckland.ac.nz/~paul/RGraphics/rgraphics.html">Book</a></li>
        </ul>
      </li>
      <li>High-level infrastructure
        <ul>
          <li><code>lattice</code>: <a href="http://lmdvr.r-forge.r-project.org">Manual</a>, <a href="http://www.his.sunderland.ac.uk/~cs0her/Statistics/UsingLatticeGraphicsInR.htm">Intro</a>, <a href="http://www.amazon.com/Lattice-Multivariate-Data-Visualization-Use/dp/0387759689">Book</a></li>
          <li><code>ggplot2</code>: <a href="http://docs.ggplot2.org/current/">Manual</a>, <a href="http://www.ling.upenn.edu/~joseff/rstudy/summer2010_ggplot2_intro.html">Intro</a>, <a href="http://had.co.nz/ggplot2/book/">Book</a></li>
        </ul>
      </li>
    </ul>
  </li>
</ul>

<h1 id="base-graphics">Base Graphics</h1>

<h2 id="overview-1">Overview</h2>

<ul>
  <li>Important high-level plotting functions
    <ul>
      <li><code>plot</code>: generic x-y plotting</li>
      <li><code>barplot</code>: bar plots</li>
      <li><code>boxplot</code>: box-and-whisker plot</li>
      <li><code>hist</code>: histograms</li>
      <li><code>pie</code>: pie charts</li>
      <li><code>dotchart</code>: cleveland dot plots</li>
      <li><code>image, heatmap, contour, persp</code>: functions to generate image-like plots</li>
      <li><code>qqnorm, qqline, qqplot</code>: distribution comparison plots</li>
      <li><code>pairs, coplot</code>: display of multivariant data</li>
    </ul>
  </li>
  <li>Help on these functions
    <ul>
      <li><code>?myfct</code>&lt;/span&gt;&lt;/span&gt;</li>
      <li><code>?plot</code>&lt;/span&gt;&lt;/span&gt;</li>
      <li><code>?par</code>&lt;/span&gt;&lt;/span&gt;</li>
    </ul>
  </li>
</ul>

<h2 id="preferred-input-data-objects">Preferred Input Data Objects</h2>

<ul>
  <li>Matrices and data frames</li>
  <li>Vectors</li>
  <li>Named vectors</li>
</ul>

<h2 id="scatter-plots">Scatter Plots</h2>

<h3 id="basic-scatter-plots">Basic scatter plots</h3>

<p>Sample data set for subsequent plots</p>

<p><code>r
set.seed(1410)
y &lt;- matrix(runif(30), ncol=3, dimnames=list(letters[1:10], LETTERS[1:3]))
plot(y[,1], y[,2]) 
</code></p>

<p><img src="Rgraphics_files/figure-html/scatter_plot_basic-1.png" alt="" /><!-- --></p>

<h3 id="all-pairs">All pairs</h3>

<p><code>r
pairs(y) 
</code></p>

<p><img src="Rgraphics_files/figure-html/scatter_plot_allpairs-1.png" alt="" /><!-- --></p>

<h3 id="plot-labels">Plot labels</h3>

<p><code>r
plot(y[,1], y[,2], pch=20, col="red", main="Symbols and Labels")
text(y[,1]+0.03, y[,2], rownames(y))
</code></p>

<p><img src="Rgraphics_files/figure-html/scatter_plot_labels-1.png" alt="" /><!-- --></p>

<h3 id="more-examples">More examples</h3>

<p>Print instead of symbols the row names</p>

<p><code>r
plot(y[,1], y[,2], type="n", main="Plot of Labels")
text(y[,1], y[,2], rownames(y)) 
</code></p>

<p>Usage of important plotting parameters</p>

<p><code>r
grid(5, 5, lwd = 2) 
op &lt;- par(mar=c(8,8,8,8), bg="lightblue")
plot(y[,1], y[,2], type="p", col="red", cex.lab=1.2, cex.axis=1.2, 
     cex.main=1.2, cex.sub=1, lwd=4, pch=20, xlab="x label", 
     ylab="y label", main="My Main", sub="My Sub")
par(op)
</code></p>

<p>Important arguments}
    - <code>mar</code>: specifies the margin sizes around the plotting area in order: <code>c(bottom, left, top, right)</code> 
    - <code>col</code>: color of symbols
    - <code>pch</code>: type of symbols, samples: <code>example(points)</code>
    - <code>lwd</code>: size of symbols
    - <code>cex.*</code>: control font sizes
    - For details see <code>?par</code></p>

<p>Add a regression line to a plot</p>

<p><code>r
plot(y[,1], y[,2])
myline &lt;- lm(y[,2]~y[,1]); abline(myline, lwd=2) 
</code></p>

<p><img src="Rgraphics_files/figure-html/scatter_plot_regress-1.png" alt="" /><!-- --></p>

<p><code>r
summary(myline) 
</code></p>

<p><code>
## 
## Call:
## lm(formula = y[, 2] ~ y[, 1])
## 
## Residuals:
##      Min       1Q   Median       3Q      Max 
## -0.40357 -0.17912 -0.04299  0.22147  0.46623 
## 
## Coefficients:
##             Estimate Std. Error t value Pr(&gt;|t|)  
## (Intercept)   0.5764     0.2110   2.732   0.0258 *
## y[, 1]       -0.3647     0.3959  -0.921   0.3839  
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## Residual standard error: 0.3095 on 8 degrees of freedom
## Multiple R-squared:  0.09589,	Adjusted R-squared:  -0.01712 
## F-statistic: 0.8485 on 1 and 8 DF,  p-value: 0.3839
</code></p>

<p>Same plot as above, but on log scale</p>

<p><code>r
plot(y[,1], y[,2], log="xy") 
</code></p>

<p><img src="Rgraphics_files/figure-html/scatter_plot_log_scale-1.png" alt="" /><!-- --></p>

<p>Add a mathematical expression to a plot</p>

<p><code>r
plot(y[,1], y[,2]); text(y[1,1], y[1,2], 
     expression(sum(frac(1,sqrt(x^2*pi)))), cex=1.3) 
</code></p>

<p><img src="Rgraphics_files/figure-html/scatter_plot_math-1.png" alt="" /><!-- --></p>

<h3 id="exercise-1">Exercise 1</h3>

<ul>
  <li><strong>Task 1</strong>: Generate scatter plot for first two columns in <code>iris</code> data frame and color dots by its <code>Species</code> column.</li>
  <li><strong>Task 2</strong>: Use the <code>xlim/ylim</code> arguments to set limits on the x- and y-axes so that all data points are restricted to the left bottom quadrant of the plot.</li>
</ul>

<p>Structure of iris data set:</p>

<p><code>r
class(iris)
</code></p>

<p><code>
## [1] "data.frame"
</code></p>

<p><code>r
iris[1:4,]
</code></p>

<p><code>
##   Sepal.Length Sepal.Width Petal.Length Petal.Width Species
## 1          5.1         3.5          1.4         0.2  setosa
## 2          4.9         3.0          1.4         0.2  setosa
## 3          4.7         3.2          1.3         0.2  setosa
## 4          4.6         3.1          1.5         0.2  setosa
</code></p>

<p><code>r
table(iris$Species)
</code></p>

<p><code>
## 
##     setosa versicolor  virginica 
##         50         50         50
</code></p>

<h2 id="line-plots">Line Plots</h2>

<h3 id="single-data-set">Single Data Set</h3>

<p><code>r
plot(y[,1], type="l", lwd=2, col="blue") 
</code></p>

<p><img src="Rgraphics_files/figure-html/line_plot_single-1.png" alt="" /><!-- --></p>

<h3 id="many-data-sets">Many Data Sets</h3>

<p>Plots line graph for all columns in data frame <code>y</code>. The <code>split.screen</code> function is used in this example in a for loop to overlay several line graphs in the same plot.</p>

<p><code>r
split.screen(c(1,1)) 
</code></p>

<p><code>
## [1] 1
</code></p>

<p><code>r
plot(y[,1], ylim=c(0,1), xlab="Measurement", ylab="Intensity", type="l", lwd=2, col=1)
for(i in 2:length(y[1,])) { 
	screen(1, new=FALSE)
	plot(y[,i], ylim=c(0,1), type="l", lwd=2, col=i, xaxt="n", yaxt="n", ylab="", 
             xlab="", main="", bty="n") 
}
</code></p>

<p><img src="Rgraphics_files/figure-html/line_plot_many-1.png" alt="" /><!-- --></p>

<p><code>r
close.screen(all=TRUE) 
</code></p>

<h2 id="bar-plots">Bar Plots</h2>

<h3 id="basics">Basics</h3>

<p><code>r
barplot(y[1:4,], ylim=c(0, max(y[1:4,])+0.3), beside=TRUE, 
        legend=letters[1:4]) 
text(labels=round(as.vector(as.matrix(y[1:4,])),2), x=seq(1.5, 13, by=1)
     +sort(rep(c(0,1,2), 4)), y=as.vector(as.matrix(y[1:4,]))+0.04) 
</code></p>

<p><img src="Rgraphics_files/figure-html/bar_plot_basic-1.png" alt="" /><!-- --></p>

<h3 id="error-bars">Error bars</h3>

<p><code>r
bar &lt;- barplot(m &lt;- rowMeans(y) * 10, ylim=c(0, 10))
stdev &lt;- sd(t(y))
arrows(bar, m, bar, m + stdev, length=0.15, angle = 90)
</code></p>

<p><img src="Rgraphics_files/figure-html/bar_plot_error_bar-1.png" alt="" /><!-- --></p>

<h3 id="mirrored-bar-plot">Mirrored bar plot</h3>

<p><code>r
df &lt;- data.frame(group = rep(c("Above", "Below"), each=10), x = rep(1:10, 2), y = c(runif(10, 0, 1), runif(10, -1, 0)))
plot(c(0,12),range(df$y),type = "n")
barplot(height = df$y[df$group == "Above"], add = TRUE,axes = FALSE)
barplot(height = df$y[df$group == "Below"], add = TRUE,axes = FALSE)
</code></p>

<p><img src="Rgraphics_files/figure-html/bar_plot_mirrored-1.png" alt="" /><!-- --></p>

<h2 id="histograms">Histograms</h2>

<p><code>r
hist(y, freq=TRUE, breaks=10)
</code></p>

<p><img src="Rgraphics_files/figure-html/hist_plot-1.png" alt="" /><!-- --></p>

<h2 id="density-plots">Density Plots}</h2>

<p><code>r
plot(density(y), col="red")
</code></p>

<p><img src="Rgraphics_files/figure-html/density_plot-1.png" alt="" /><!-- --></p>

<h2 id="pie-charts">Pie Charts</h2>

<p><code>r
pie(y[,1], col=rainbow(length(y[,1]), start=0.1, end=0.8), clockwise=TRUE)
legend("topright", legend=row.names(y), cex=1.3, bty="n", pch=15, pt.cex=1.8, 
col=rainbow(length(y[,1]), start=0.1, end=0.8), ncol=1) 
</code></p>

<p><img src="Rgraphics_files/figure-html/pie_chart-1.png" alt="" /><!-- --></p>

<h2 id="color-selection-utilities">Color Selection Utilities</h2>

<p>Default color palette and how to change it</p>

<p><code>r
palette()
</code></p>

<p><code>
## [1] "black"   "red"     "green3"  "blue"    "cyan"    "magenta" "yellow"  "gray"
</code></p>

<p><code>r
palette(rainbow(5, start=0.1, end=0.2))
palette()
</code></p>

<p><code>
## [1] "#FF9900" "#FFBF00" "#FFE600" "#F2FF00" "#CCFF00"
</code></p>

<p><code>r
palette("default")
</code></p>

<p>The <code>gray</code> function allows to select any type of gray shades by providing values from 0 to 1</p>

<p><code>r
gray(seq(0.1, 1, by= 0.2))
</code></p>

<p><code>
## [1] "#1A1A1A" "#4D4D4D" "#808080" "#B3B3B3" "#E6E6E6"
</code></p>

<p>Color gradients with <code>colorpanel</code> function from <code>gplots</code> library</p>

<p><code>r
library(gplots)
colorpanel(5, "darkblue", "yellow", "white")
</code></p>

<p>Much more on colors in R see Earl Glynn’s <a href="http://research.stowers-institute.org/efg/R/Color/Chart/">color chart</a></p>

<h2 id="arranging-several-plots-on-single-page">Arranging Several Plots on Single Page</h2>

<p>With <code>par(mfrow=c(nrow,ncol))</code> one can define how several plots are arranged next to each other.</p>

<p><code>r
par(mfrow=c(2,3)); for(i in 1:6) { plot(1:10) } 
</code></p>

<p><img src="Rgraphics_files/figure-html/par_mfrow-1.png" alt="" /><!-- --></p>

<h3 id="arranging-plots-with-variable-width">Arranging Plots with Variable Width</h3>

<p>The <code>layout</code> function allows to divide the plotting device into variable numbers of rows and columns with the column-widths and the row-heights specified in the respective arguments.</p>

<p><code>r
nf &lt;- layout(matrix(c(1,2,3,3), 2, 2, byrow=TRUE), c(3,7), c(5,5), 
             respect=TRUE)
# layout.show(nf)
for(i in 1:3) { barplot(1:10) } 
</code></p>

<p><img src="Rgraphics_files/figure-html/layout_plot-1.png" alt="" /><!-- --></p>

<h2 id="saving-graphics-to-files">Saving Graphics to Files</h2>

<p>After the <code>pdf()</code> command all graphs are redirected to file <code>test.pdf</code>. Works for all common formats similarly: jpeg, png, ps, tiff, …</p>

<p><code>r
pdf("test.pdf"); plot(1:10, 1:10); dev.off() 
</code></p>

<p>Generates Scalable Vector Graphics (SVG) files that can be edited in vector graphics programs, such as InkScape.</p>

<p><code>r
svg("test.svg"); plot(1:10, 1:10); dev.off() 
</code></p>

<h3 id="exercise-2">Exercise 2</h3>

<p>Bar plots</p>

<ul>
  <li><strong>Task 1</strong>: Calculate the mean values for the \Rfunction{Species} components of the first four columns in the \Rfunction{iris} data set. Organize the results in a matrix where the row names are the unique values from the \Rfunction{iris Species} column and the column names are the same as in the first four \Rfunction{iris} columns.</li>
  <li><strong>Task 2</strong>: Generate two bar plots: one with stacked bars and one with horizontally arranged bars.</li>
</ul>

<p>Structure of iris data set:</p>

<p><code>r
class(iris)
</code></p>

<p><code>
## [1] "data.frame"
</code></p>

<p><code>r
iris[1:4,]
</code></p>

<p><code>
##   Sepal.Length Sepal.Width Petal.Length Petal.Width Species
## 1          5.1         3.5          1.4         0.2  setosa
## 2          4.9         3.0          1.4         0.2  setosa
## 3          4.7         3.2          1.3         0.2  setosa
## 4          4.6         3.1          1.5         0.2  setosa
</code></p>

<p><code>r
table(iris$Species)
</code></p>

<p><code>
## 
##     setosa versicolor  virginica 
##         50         50         50
</code></p>

<h1 id="grid-graphics">Grid Graphics</h1>

<ul>
  <li>What is <code>grid</code>?
    <ul>
      <li>Low-level graphics system</li>
      <li>Highly flexible and controllable system</li>
      <li>Does not provide high-level functions</li>
      <li>Intended as development environment for custom plotting functions</li>
      <li>Pre-installed on new R distributions</li>
    </ul>
  </li>
  <li>Documentation and Help
    <ul>
      <li><a href="http://www.stat.auckland.ac.nz/~paul/grid/grid.html">Manual</a></li>
      <li><a href="http://www.stat.auckland.ac.nz/~paul/RGraphics/rgraphics.html">Book</a></li>
    </ul>
  </li>
</ul>

<h1 id="lattice-graphics">lattice Graphics</h1>

<ul>
  <li>What is <code>lattice</code>?
    <ul>
      <li>High-level graphics system</li>
      <li>Developed by Deepayan Sarkar</li>
      <li>Implements Trellis graphics system from S-Plus</li>
      <li>Simplifies high-level plotting tasks: arranging complex graphical features</li>
      <li>Syntax similar to R’s base graphics</li>
    </ul>
  </li>
  <li>Documentation and Help
    <ul>
      <li><a href="http://lmdvr.r-forge.r-project.org">Manual</a></li>
      <li><a href="http://www.his.sunderland.ac.uk/~cs0her/Statistics/UsingLatticeGraphicsInR.htm">Intro</a></li>
      <li><a href="http://www.amazon.com/Lattice-Multivariate-Data-Visualization-Use/dp/0387759689">Book</a></li>
    </ul>
  </li>
</ul>

<p>Open a list of all functions available in the lattice package</p>

<p><code>r
library(help=lattice) 
</code></p>

<p>Accessing and changing global parameters:</p>

<p><code>r
?lattice.options
?trellis.device
</code></p>

<h2 id="scatter-plot-sample">Scatter Plot Sample</h2>

<p><code>r
library(lattice)
p1 &lt;- xyplot(1:8 ~ 1:8 | rep(LETTERS[1:4], each=2), as.table=TRUE) 
plot(p1)
</code></p>

<p><img src="Rgraphics_files/figure-html/scatter_plot_lattice-1.png" alt="" /><!-- --></p>

<h2 id="line-plot-sample">Line Plot Sample</h2>

<p><code>r
library(lattice)
p2 &lt;- parallelplot(~iris[1:4] | Species, iris, horizontal.axis = FALSE, 
              layout = c(1, 3, 1))  
plot(p2)
</code></p>

<p><img src="Rgraphics_files/figure-html/line_plot_lattice-1.png" alt="" /><!-- --></p>

<h1 id="ggplot2-graphics">ggplot2 Graphics</h1>

<ul>
  <li>What is <code>ggplot2</code>?
    <ul>
      <li>High-level graphics system</li>
      <li>Implements grammar of graphics from <a href="http://www.amazon.com/Grammar-Graphics-Leland-Wilkinson/dp/0387987746">Leland Wilkinson</a></li>
      <li>Streamlines many graphics workflows for complex plots</li>
      <li>Syntax centered around main <code>ggplot</code> function</li>
      <li>Simpler <code>qplot</code> function provides many shortcuts</li>
    </ul>
  </li>
  <li>Documentation and Help
    <ul>
      <li><a href="http://had.co.nz/ggplot2/">Manual</a></li>
      <li><a href="http://www.ling.upenn.edu/~joseff/rstudy/summer2010_ggplot2_intro.html">Intro</a></li>
      <li><a href="http://had.co.nz/ggplot2/book/">Book</a></li>
      <li><a href="http://www.cookbook-r.com/Graphs/">Cookbook for R</a></li>
    </ul>
  </li>
</ul>

<h2 id="ggplot2-usage"><code>ggplot2</code> Usage</h2>

<ul>
  <li><code>ggplot</code> function accepts two arguments
    <ul>
      <li>Data set to be plotted</li>
      <li>Aesthetic mappings provided by <code>aes</code> function</li>
    </ul>
  </li>
  <li>Additional parameters such as geometric objects (<em>e.g.</em> points, lines, bars) are passed on by appending them with <code>+</code> as separator.</li>
  <li>List of available <code>geom_*</code> functions see <a href="http://docs.ggplot2.org/current/">here</a></li>
  <li>Settings of plotting theme can be accessed with the command <code>theme_get()</code> and its settings can be changed with <code>theme()</code>.</li>
  <li>Preferred input data object
    <ul>
      <li><code>qplot</code>: <code>data.frame</code> (support for <code>vector</code>, <code>matrix</code>, <code>...</code>)</li>
      <li><code>ggplot</code>: <code>data.frame</code></li>
    </ul>
  </li>
  <li>Packages with convenience utilities to create expected inputs
    <ul>
      <li><code>plyr</code></li>
      <li><code>reshape</code></li>
    </ul>
  </li>
</ul>

<h2 id="qplot-function"><code>qplot</code> Function</h2>

<p>The syntax of <code>qplot</code> is similar as R’s basic <code>plot</code> function</p>

<ul>
  <li>Arguments
    <ul>
      <li><code>x</code>: x-coordinates (<em>e.g.</em> <code>col1</code>)</li>
      <li><code>y</code>: y-coordinates (<em>e.g.</em> <code>col2</code>)</li>
      <li><code>data</code>: data frame with corresponding column names</li>
      <li><code>xlim, ylim</code>: <em>e.g.</em> <code>xlim=c(0,10)</code></li>
      <li><code>log</code>: <em>e.g.</em> <code>log="x"</code> or <code>log="xy"</code></li>
      <li><code>main</code>: main title; see <code>?plotmath</code> for mathematical formula</li>
      <li><code>xlab, ylab</code>: labels for the x- and y-axes</li>
      <li><code>color</code>, <code>shape</code>, <code>size</code></li>
      <li><code>...</code>: many arguments accepted by <code>plot</code> function</li>
    </ul>
  </li>
</ul>

<h2 id="qplot-scatter-plot-basics"><code>qplot</code>: scatter plot basics</h2>

<p>Create sample data</p>

<p><code>r
library(ggplot2)
x &lt;- sample(1:10, 10); y &lt;- sample(1:10, 10); cat &lt;- rep(c("A", "B"), 5)
</code></p>

<p>Simple scatter plot</p>

<p><code>r
qplot(x, y, geom="point")
</code></p>

<p><img src="Rgraphics_files/figure-html/qplot_scatter_plot-1.png" alt="" /><!-- --></p>

<p>Prints dots with different sizes and colors</p>

<p><code>r
qplot(x, y, geom="point", size=x, color=cat, 
      main="Dot Size and Color Relative to Some Values")
</code></p>

<p><img src="Rgraphics_files/figure-html/qplot_scatter_plot_dot_param-1.png" alt="" /><!-- --></p>

<p>Drops legend</p>

<p><code>r
qplot(x, y, geom="point", size=x, color=cat) + 
      theme(legend.position = "none")
</code></p>

<p><img src="Rgraphics_files/figure-html/qplot_scatter_plot_no_legend-1.png" alt="" /><!-- --></p>

<p>Plot different shapes</p>

<p><code>r
qplot(x, y, geom="point", size=5, shape=cat)
</code></p>

<p><img src="Rgraphics_files/figure-html/qplot_scatter_plot_shapes-1.png" alt="" /><!-- --></p>

<h3 id="colored-groups">Colored groups</h3>

<p><code>r
p &lt;- qplot(x, y, geom="point", size=x, color=cat, 
            main="Dot Size and Color Relative to Some Values") + 
     theme(legend.position = "none")
print(p)
</code></p>

<p><img src="Rgraphics_files/figure-html/qplot_scatter_plot_colored_groups-1.png" alt="" /><!-- --></p>

<h3 id="regression-line">Regression line</h3>

<p><code>r
set.seed(1410)
dsmall &lt;- diamonds[sample(nrow(diamonds), 1000), ]
p &lt;- qplot(carat, price, data = dsmall) +
           geom_smooth(method="lm")
print(p)
</code></p>

<p><img src="Rgraphics_files/figure-html/qplot_scatter_regression_line-1.png" alt="" /><!-- --></p>

<h3 id="local-regression-curve-loess">Local regression curve (loess)</h3>

<p><code>r
p &lt;- qplot(carat, price, data=dsmall, geom=c("point", "smooth")) 
print(p) # Setting se=FALSE removes error shade
</code></p>

<p><img src="Rgraphics_files/figure-html/qplot_scatter_regression_loess-1.png" alt="" /><!-- --></p>

<h2 id="ggplot-function"><code>ggplot</code> Function</h2>

<ul>
  <li>More important than <code>qplot</code> to access full functionality of <code>ggplot2</code></li>
  <li>Main arguments
    <ul>
      <li>data set, usually a <code>data.frame</code></li>
      <li>aesthetic mappings provided by <code>aes</code> function</li>
    </ul>
  </li>
  <li>General <code>ggplot</code> syntax
    <ul>
      <li><code>ggplot(data, aes(...)) + geom() + ... + stat() + ...</code></li>
    </ul>
  </li>
  <li>Layer specifications
    <ul>
      <li><code>geom(mapping, data, ..., geom, position)</code></li>
      <li><code>stat(mapping, data, ..., stat, position)</code></li>
    </ul>
  </li>
  <li>Additional components
    <ul>
      <li><code>scales</code></li>
      <li><code>coordinates</code></li>
      <li><code>facet</code></li>
    </ul>
  </li>
  <li><code>aes()</code> mappings can be passed on to all components (<code>ggplot, geom</code>, etc.). Effects are global when passed on to <code>ggplot()</code> and local for other components.
    <ul>
      <li><code>x, y</code></li>
      <li><code>color</code>: grouping vector (factor)</li>
      <li><code>group</code>: grouping vector (factor)</li>
    </ul>
  </li>
</ul>

<h3 id="changing-plotting-themes-in-ggplot">Changing Plotting Themes in <code>ggplot</code></h3>

<ul>
  <li>Theme settings can be accessed with <code>theme_get()</code></li>
  <li>Their settings can be changed with <code>theme()</code></li>
</ul>

<p>Example how to change background color to white</p>

<p><code>r
... + theme(panel.background=element_rect(fill = "white", colour = "black")) 
</code></p>

<h3 id="storing-ggplot-specifications">Storing <code>ggplot</code> Specifications</h3>

<p>Plots and layers can be stored in variables</p>

<p><code>r
p &lt;- ggplot(dsmall, aes(carat, price)) + geom_point() 
p # or print(p)
</code></p>

<p>Returns information about data and aesthetic mappings followed by each layer</p>

<p><code>r
summary(p) 
</code></p>

<p>Print dots with different sizes and colors</p>

<p><code>r
bestfit &lt;- geom_smooth(methodw = "lm", se = F, color = alpha("steelblue", 0.5), size = 2)
p + bestfit # Plot with custom regression line
</code></p>

<p>Syntax to pass on other data sets</p>

<p><code>r
p %+% diamonds[sample(nrow(diamonds), 100),] 
</code></p>

<p>Saves plot stored in variable <code>p</code> to file</p>

<p><code>r
ggsave(p, file="myplot.pdf") 
</code></p>

<h2 id="ggplot-scatter-plots"><code>ggplot</code>: scatter plots</h2>

<h3 id="basic-example">Basic example</h3>

<p><code>r
p &lt;- ggplot(dsmall, aes(carat, price, color=color)) + 
            geom_point(size=4)
print(p) 
</code></p>

<p><img src="Rgraphics_files/figure-html/ggplot_scatter_plot1-1.png" alt="" /><!-- --></p>

<h3 id="regression-line-1">Regression line</h3>

<p><code>r
p &lt;- ggplot(dsmall, aes(carat, price)) + geom_point() + 
            geom_smooth(method="lm", se=FALSE) +
    	    theme(panel.background=element_rect(fill = "white", colour = "black"))
print(p) 
</code></p>

<p><img src="Rgraphics_files/figure-html/ggplot_regression_line-1.png" alt="" /><!-- --></p>

<h3 id="several-regression-lines">Several regression lines</h3>

<p><code>r
p &lt;- ggplot(dsmall, aes(carat, price, group=color)) + 
            geom_point(aes(color=color), size=2) + 
            geom_smooth(aes(color=color), method = "lm", se=FALSE) 
print(p) 
</code></p>

<p><img src="Rgraphics_files/figure-html/ggplot_many_regression_lines-1.png" alt="" /><!-- --></p>

<h3 id="local-regression-curve-loess-1">Local regression curve (loess)</h3>

<p><code>r
p &lt;- ggplot(dsmall, aes(carat, price)) + geom_point() + geom_smooth() 
print(p) # Setting se=FALSE removes error shade
</code></p>

<p><img src="Rgraphics_files/figure-html/ggplot_loess_regression-1.png" alt="" /><!-- --></p>

<h2 id="ggplot-line-plot"><code>ggplot</code>: line plot</h2>

<p><code>r
p &lt;- ggplot(iris, aes(Petal.Length, Petal.Width, group=Species, 
            color=Species)) + geom_line() 
print(p) 
</code></p>

<p><img src="Rgraphics_files/figure-html/ggplot_line_plot-1.png" alt="" /><!-- --></p>

<h2 id="faceting">Faceting</h2>

<p><code>r
p &lt;- ggplot(iris, aes(Sepal.Length, Sepal.Width)) + 
    	    geom_line(aes(color=Species), size=1) + 
            facet_wrap(~Species, ncol=1)
print(p) 
</code></p>

<p><img src="Rgraphics_files/figure-html/ggplot_line_plot_faceting-1.png" alt="" /><!-- --></p>

<h3 id="exercise-3">Exercise 3</h3>

<p>Scatter plots with <code>ggplot2</code></p>

<ul>
  <li><strong>Task 1</strong>: Generate scatter plot for first two columns in \Rfunction{iris} data frame and color dots by its \Rfunction{Species} column.</li>
  <li><strong>Task 2</strong>: Use the \Rfunarg{xlim, ylim} functionss to set limits on the x- and y-axes so that all data points are restricted to the left bottom quadrant of the plot.</li>
  <li><strong>Task 3</strong>: Generate corresponding line plot with faceting show individual data sets in saparate plots.</li>
</ul>

<p>Structure of <code>iris</code> data set</p>

<p><code>r
class(iris)
</code></p>

<p><code>
## [1] "data.frame"
</code></p>

<p><code>r
iris[1:4,]
</code></p>

<p><code>
##   Sepal.Length Sepal.Width Petal.Length Petal.Width Species
## 1          5.1         3.5          1.4         0.2  setosa
## 2          4.9         3.0          1.4         0.2  setosa
## 3          4.7         3.2          1.3         0.2  setosa
## 4          4.6         3.1          1.5         0.2  setosa
</code></p>

<p><code>r
table(iris$Species)
</code></p>

<p><code>
## 
##     setosa versicolor  virginica 
##         50         50         50
</code></p>

<h2 id="bar-plots-1">Bar plots</h2>

<p>Sample Set: the following transforms the <code>iris</code> data set into a ggplot2-friendly format.</p>

<p>Calculate mean values for aggregates given by <code>Species</code> column in <code>iris</code> data set</p>

<p><code>r
iris_mean &lt;- aggregate(iris[,1:4], by=list(Species=iris$Species), FUN=mean) 
</code></p>

<p>Calculate standard deviations for aggregates given by <code>Species</code> column in <code>iris</code> data set</p>

<p><code>r
iris_sd &lt;- aggregate(iris[,1:4], by=list(Species=iris$Species), FUN=sd) 
</code></p>

<p>Reformat <code>iris_mean</code> with <code>melt</code></p>

<p><code>r
library(reshape2) # Defines melt function
df_mean &lt;- melt(iris_mean, id.vars=c("Species"), variable.name = "Samples", value.name="Values")
</code></p>

<p>Reformat <code>iris_sd</code> with <code>melt</code></p>

<p><code>r
df_sd &lt;- melt(iris_sd, id.vars=c("Species"), variable.name = "Samples", value.name="Values")
</code></p>

<p>Define standard deviation limits</p>

<p><code>r
limits &lt;- aes(ymax = df_mean[,"Values"] + df_sd[,"Values"], ymin=df_mean[,"Values"] - df_sd[,"Values"])
</code></p>

<h3 id="verical-orientation">Verical orientation</h3>

<p><code>r
p &lt;- ggplot(df_mean, aes(Samples, Values, fill = Species)) + 
	    geom_bar(position="dodge", stat="identity")
print(p) 
</code></p>

<p><img src="Rgraphics_files/figure-html/iris_mean_bar_plot-1.png" alt="" /><!-- --></p>

<h3 id="horizontal-orientation">Horizontal orientation</h3>

<p><code>r
p &lt;- ggplot(df_mean, aes(Samples, Values, fill = Species)) + 
            geom_bar(position="dodge", stat="identity") + coord_flip() + 
            theme(axis.text.y=element_text(angle=0, hjust=1))
print(p) 
</code></p>

<p><img src="Rgraphics_files/figure-html/iris_mean_bar_plot_sideways-1.png" alt="" /><!-- --></p>

<h3 id="faceting-1">Faceting</h3>

<p><code>r
p &lt;- ggplot(df_mean, aes(Samples, Values)) + geom_bar(aes(fill = Species), stat="identity") + 
            facet_wrap(~Species, ncol=1)
print(p) 
</code></p>

<p><img src="Rgraphics_files/figure-html/iris_mean_bar_plot_facetting-1.png" alt="" /><!-- --></p>

<h3 id="error-bars-1">Error bars</h3>

<p><code>r
p &lt;- ggplot(df_mean, aes(Samples, Values, fill = Species)) + 
	    geom_bar(position="dodge", stat="identity") + geom_errorbar(limits, position="dodge") 
print(p) 
</code></p>

<p><img src="Rgraphics_files/figure-html/iris_mean_bar_plot_error-1.png" alt="" /><!-- --></p>

<h3 id="mirrored">Mirrored</h3>

<p><code>r
df &lt;- data.frame(group = rep(c("Above", "Below"), each=10), x = rep(1:10, 2), y = c(runif(10, 0, 1), runif(10, -1, 0)))
p &lt;- ggplot(df, aes(x=x, y=y, fill=group)) + 
	    geom_bar(stat="identity", position="identity")
print(p) 
</code></p>

<p><img src="Rgraphics_files/figure-html/ggplot2_mirrored_barplot-1.png" alt="" /><!-- --></p>

<h2 id="changing-color-settings">Changing Color Settings</h2>

<p><code>r
library(RColorBrewer)
# display.brewer.all() 
p &lt;- ggplot(df_mean, aes(Samples, Values, fill=Species, color=Species)) +
            geom_bar(position="dodge", stat="identity") + geom_errorbar(limits, position="dodge") + 
            scale_fill_brewer(palette="Blues") + scale_color_brewer(palette = "Greys") 
print(p) 
</code></p>

<p><img src="Rgraphics_files/figure-html/ggplot2_color1-1.png" alt="" /><!-- --></p>

<h3 id="using-standard-colors">Using standard colors</h3>

<p><code>r
p &lt;- ggplot(df_mean, aes(Samples, Values, fill=Species, color=Species)) + 
            geom_bar(position="dodge", stat="identity") + geom_errorbar(limits, position="dodge") + 
            scale_fill_manual(values=c("red", "green3", "blue")) + 
            scale_color_manual(values=c("red", "green3", "blue")) 
print(p) 
</code></p>

<p><img src="Rgraphics_files/figure-html/ggplot2_color2-1.png" alt="" /><!-- --></p>

<h3 id="exercise-4">Exercise 4</h3>

<p>Bar plots</p>

<ul>
  <li><strong>Task 1</strong>: Calculate the mean values for the <code>Species</code> components of the first four columns in the <code>iris</code> data set. Use the <code>melt</code> function from the <code>reshape2</code> package to bring the data into the expected format for <code>ggplot</code>.</li>
  <li><strong>Task 2</strong>: Generate two bar plots: one with stacked bars and one with horizontally arranged bars.</li>
</ul>

<p>Structure of iris data set</p>

<p><code>r
class(iris)
</code></p>

<p><code>
## [1] "data.frame"
</code></p>

<p><code>r
iris[1:4,]
</code></p>

<p><code>
##   Sepal.Length Sepal.Width Petal.Length Petal.Width Species
## 1          5.1         3.5          1.4         0.2  setosa
## 2          4.9         3.0          1.4         0.2  setosa
## 3          4.7         3.2          1.3         0.2  setosa
## 4          4.6         3.1          1.5         0.2  setosa
</code></p>

<p><code>r
table(iris$Species)
</code></p>

<p><code>
## 
##     setosa versicolor  virginica 
##         50         50         50
</code></p>

<h2 id="data-reformatting-example">Data reformatting example</h2>

<p>Here for line plot</p>

<p><code>r
y &lt;- matrix(rnorm(500), 100, 5, dimnames=list(paste("g", 1:100, sep=""), paste("Sample", 1:5, sep="")))
y &lt;- data.frame(Position=1:length(y[,1]), y)
y[1:4, ] # First rows of input format expected by melt()
</code></p>

<p><code>
##    Position    Sample1    Sample2    Sample3    Sample4    Sample5
## g1        1 1.32942477 -1.2084007 -0.1958190 -0.4236177  1.7139697
## g2        2 0.92190035 -0.3471160  3.3238031 -1.2340292 -0.3985408
## g3        3 0.01878173  0.8007423 -0.1884464 -0.7419688 -0.5565102
## g4        4 1.95620993  1.7876584 -0.4402745  0.3671016  0.3966960
</code></p>

<p><code>r
df &lt;- melt(y, id.vars=c("Position"), variable.name = "Samples", value.name="Values")
p &lt;- ggplot(df, aes(Position, Values)) + geom_line(aes(color=Samples)) + facet_wrap(~Samples, ncol=1)
print(p)
</code></p>

<p><img src="Rgraphics_files/figure-html/ggplot_melt_data-1.png" alt="" /><!-- --></p>

<p>Same data can be represented in box plot as follows</p>

<p><code>r
ggplot(df, aes(Samples, Values, fill=Samples)) + geom_boxplot()
</code></p>

<h2 id="jitter-plots">Jitter Plots</h2>

<p><code>r
p &lt;- ggplot(dsmall, aes(color, price/carat)) + 
            geom_jitter(alpha = I(1 / 2), aes(color=color))
print(p) 
</code></p>

<p><img src="Rgraphics_files/figure-html/ggplot_jitter_plot-1.png" alt="" /><!-- --></p>

<h2 id="box-plots">Box plots</h2>

<p><code>r
p &lt;- ggplot(dsmall, aes(color, price/carat, fill=color)) + geom_boxplot()
print(p) 
</code></p>

<p><img src="Rgraphics_files/figure-html/ggplot_box_plot2-1.png" alt="" /><!-- --></p>

<h2 id="density-plots-1">Density plots</h2>

<h3 id="line-coloring">Line coloring</h3>

<p><code>r
p &lt;- ggplot(dsmall, aes(carat)) + geom_density(aes(color = color))
print(p) 
</code></p>

<p><img src="Rgraphics_files/figure-html/ggplot_density_plot_linecol-1.png" alt="" /><!-- --></p>

<h3 id="area-coloring">Area coloring</h3>

<p><code>r
p &lt;- ggplot(dsmall, aes(carat)) + geom_density(aes(fill = color))
print(p) 
</code></p>

<p><img src="Rgraphics_files/figure-html/ggplot_density_plot_areacol-1.png" alt="" /><!-- --></p>

<h2 id="histograms-1">Histograms</h2>

<p><code>r
p &lt;- ggplot(iris, aes(x=Sepal.Width)) + geom_histogram(aes(y = ..density.., 
            fill = ..count..), binwidth=0.2) + geom_density()  
print(p) 
</code></p>

<p><img src="Rgraphics_files/figure-html/ggplot_histogram-1.png" alt="" /><!-- --></p>

<h2 id="pie-chart">Pie Chart</h2>

<p><code>r
df &lt;- data.frame(variable=rep(c("cat", "mouse", "dog", "bird", "fly")), 
                 value=c(1,3,3,4,2)) 
p &lt;- ggplot(df, aes(x = "", y = value, fill = variable)) + 
            geom_bar(width = 1, stat="identity") + 
            coord_polar("y", start=pi / 3) + ggtitle("Pie Chart") 
print(p) 
</code></p>

<p><img src="Rgraphics_files/figure-html/ggplot_pie_chart-1.png" alt="" /><!-- --></p>

<h2 id="wind-rose-pie-chart">Wind Rose Pie Chart</h2>

<p><code>r
p &lt;- ggplot(df, aes(x = variable, y = value, fill = variable)) + 
       geom_bar(width = 1, stat="identity") + coord_polar("y", start=pi / 3) + 
       ggtitle("Pie Chart") 
print(p) 
</code></p>

<p><img src="Rgraphics_files/figure-html/ggplot_windrose_pie_chart-1.png" alt="" /><!-- --></p>

<h2 id="arranging-graphics-on-page">Arranging Graphics on Page</h2>

<p><code>r
library(grid)
a &lt;- ggplot(dsmall, aes(color, price/carat)) + geom_jitter(size=4, alpha = I(1 / 1.5), aes(color=color))
b &lt;- ggplot(dsmall, aes(color, price/carat, color=color)) + geom_boxplot()
c &lt;- ggplot(dsmall, aes(color, price/carat, fill=color)) + geom_boxplot() + theme(legend.position = "none")
grid.newpage() # Open a new page on grid device
pushViewport(viewport(layout = grid.layout(2, 2))) # Assign to device viewport with 2 by 2 grid layout 
print(a, vp = viewport(layout.pos.row = 1, layout.pos.col = 1:2))
print(b, vp = viewport(layout.pos.row = 2, layout.pos.col = 1))
print(c, vp = viewport(layout.pos.row = 2, layout.pos.col = 2, width=0.3, height=0.3, x=0.8, y=0.8))
</code></p>

<p><img src="Rgraphics_files/figure-html/ggplot_arrange_graphics-1.png" alt="" /><!-- --></p>

<h2 id="inserting-graphics-into-plots">Inserting Graphics into Plots</h2>

<p><code>r
library(grid)
print(a)
print(b, vp=viewport(width=0.3, height=0.3, x=0.8, y=0.8))
</code></p>

<p><img src="Rgraphics_files/figure-html/ggplot_insert_graphics-1.png" alt="" /><!-- --></p>

<h1 id="specialty-graphics">Specialty Graphics</h1>

<h2 id="venn-diagrams">Venn Diagrams</h2>

<p><code>r
source("http://faculty.ucr.edu/~tgirke/Documents/R_BioCond/My_R_Scripts/overLapper.R")
setlist5 &lt;- list(A=sample(letters, 18), B=sample(letters, 16), C=sample(letters, 20), D=sample(letters, 22), E=sample(letters, 18))
OLlist5 &lt;- overLapper(setlist=setlist5, sep="_", type="vennsets")
counts &lt;- sapply(OLlist5$Venn_List, length)
vennPlot(counts=counts, ccol=c(rep(1,30),2), lcex=1.5, ccex=c(rep(1.5,5), rep(0.6,25),1.5))
</code></p>

<p><img src="Rgraphics_files/figure-html/specgraph_venn-1.png" alt="" /><!-- --></p>

<h2 id="compound-structures">Compound Structures</h2>

<p>Plots depictions of small molecules with <code>ChemmineR</code> package</p>

<p><code>r
library(ChemmineR)
</code></p>

<p><code>
## 
## Attaching package: 'ChemmineR'
</code></p>

<p><code>
## The following object is masked from 'package:ShortRead':
## 
##     view
</code></p>

<p><code>
## The following object is masked from 'package:S4Vectors':
## 
##     fold
</code></p>

<p><code>r
data(sdfsample)
plot(sdfsample[1], print=FALSE)
</code></p>

<p><img src="Rgraphics_files/figure-html/specgraph_structure-1.png" alt="" /><!-- --></p>

<h1 id="genome-graphics">Genome Graphics</h1>

<h2 id="ggbio"><code>ggbio</code></h2>

<ul>
  <li>What is <code>ggbio</code>?
    <ul>
      <li>A programmable genome browser environment</li>
    </ul>
  </li>
  <li>Genome broswer concepts
    <ul>
      <li>A genome browser is a visulalization tool for plotting different types of genomic data in separate tracks along chromosomes.</li>
      <li>The <code>ggbio</code> package [@Yin2012-jj] facilitates plotting of complex genome data objects, such as read alignments (SAM/BAM), genomic context/annotation information (gff/txdb), variant calls (VCF/BCF), and more. To easily compare these data sets, it extends the faceting facility of <code>ggplot2</code> to genome browser-like tracks.</li>
      <li>Most of the core object types for handling genomic data with R/Bioconductor are supported: <code>GRanges</code>, <code>GAlignments</code>, <code>VCF</code>, etc. For more details, see Table 1.1 of the <code>ggbio</code> vignette <a href="http://www.bioconductor.org/packages/release/bioc/vignettes/ggbio/inst/doc/ggbio.pdf">here</a>.</li>
      <li><code>ggbio</code>’s convenience plotting function is <code>autoplot</code>. For more customizable plots, one can use the generic <code>ggplot</code> function.</li>
      <li>Apart from the standard <code>ggplot2</code> plotting components, <code>ggbio</code> defines serval new components useful for genomic data visualization. A detailed list is given in Table 1.2 of the vignette <a href="http://www.bioconductor.org/packages/release/bioc/vignettes/ggbio/inst/doc/ggbio.pdf">here</a>.</li>
      <li>Useful web sites:
        <ul>
          <li><a href="http://www.tengfei.name/ggbio/docs/">ggbio manual</a></li>
          <li><a href="http://www.tengfei.name/ggbio/docs/man/">ggbio functions</a></li>
          <li><a href="http://www.tengfei.name/ggbio/docs/man/autoplot-method.html">autoplot demo</a></li>
        </ul>
      </li>
    </ul>
  </li>
</ul>

<h3 id="tracks-aligning-plots-along-chromosomes">Tracks: aligning plots along chromosomes</h3>

<p><code>r
library(ggbio)
df1 &lt;- data.frame(time = 1:100, score = sin((1:100)/20)*10)
p1 &lt;- qplot(data = df1, x = time, y = score, geom = "line")
df2 &lt;- data.frame(time = 30:120, score = sin((30:120)/20)*10, value = rnorm(120-30 +1))
p2 &lt;- ggplot(data = df2, aes(x = time, y = score)) + geom_line() + geom_point(size = 2, aes(color = value))
tracks(time1 = p1, time2 = p2) + xlim(1, 40) + theme_tracks_sunset()
</code></p>

<p><img src="Rgraphics_files/figure-html/ggbio_align-1.png" alt="" /><!-- --></p>

<h3 id="plotting-genomic-ranges">Plotting genomic ranges</h3>

<p><code>GRanges</code> objects are essential for storing alignment or annotation ranges in R/Bioconductor. The following creates a sample <code>GRanges</code> object and plots its content.</p>

<p><code>r
library(GenomicRanges)
set.seed(1); N &lt;- 100; gr &lt;- GRanges(seqnames = sample(c("chr1", "chr2", "chr3"), size = N, replace = TRUE), IRanges(start = sample(1:300, size = N, replace = TRUE), width = sample(70:75, size = N,replace = TRUE)), strand = sample(c("+", "-"), size = N, replace = TRUE), value = rnorm(N, 10, 3), score = rnorm(N, 100, 30), sample = sample(c("Normal", "Tumor"), size = N, replace = TRUE), pair = sample(letters, size = N, replace = TRUE))
autoplot(gr, aes(color = strand, fill = strand), facets = strand ~ seqnames)
</code></p>

<p><img src="Rgraphics_files/figure-html/ggbio_granges-1.png" alt="" /><!-- --></p>

<h3 id="plotting-coverage">Plotting coverage</h3>

<p><code>r
autoplot(gr, aes(color = strand, fill = strand), facets = strand ~ seqnames, stat = "coverage")
</code></p>

<p><code>
## Scale for 'x' is already present. Adding another scale for 'x', which will replace the existing
## scale.
</code></p>

<p><img src="Rgraphics_files/figure-html/ggbio_coverage-1.png" alt="" /><!-- --></p>

<h3 id="mirrored-coverage">Mirrored coverage</h3>

<p><code>r
pos &lt;- sapply(coverage(gr[strand(gr)=="+"]), as.numeric)
pos &lt;- data.frame(Chr=rep(names(pos), sapply(pos, length)), Strand=rep("+", length(unlist(pos))), Position=unlist(sapply(pos, function(x) 1:length(x))), Coverage=as.numeric(unlist(pos)))
neg &lt;- sapply(coverage(gr[strand(gr)=="-"]), as.numeric)
neg &lt;- data.frame(Chr=rep(names(neg), sapply(neg, length)), Strand=rep("-", length(unlist(neg))), Position=unlist(sapply(neg, function(x) 1:length(x))), Coverage=-as.numeric(unlist(neg)))
covdf &lt;- rbind(pos, neg)
p &lt;- ggplot(covdf, aes(Position, Coverage, fill=Strand)) + 
	    geom_bar(stat="identity", position="identity") + facet_wrap(~Chr)
p
</code></p>

<p><img src="Rgraphics_files/figure-html/ggbio_mirrored_coverage-1.png" alt="" /><!-- --></p>

<h2 id="circular-genome-plots">Circular genome plots</h2>

<p><code>r
ggplot(gr) + layout_circle(aes(fill = seqnames), geom = "rect")
</code></p>

<p><img src="Rgraphics_files/figure-html/ggbio_circular1-1.png" alt="" /><!-- --></p>

<p>More complex circular example</p>

<p><code>r
seqlengths(gr) &lt;- c(400, 500, 700)
values(gr)$to.gr &lt;- gr[sample(1:length(gr), size = length(gr))]
idx &lt;- sample(1:length(gr), size = 50)
gr &lt;- gr[idx]
ggplot() + layout_circle(gr, geom = "ideo", fill = "gray70", radius = 7, trackWidth = 3) +
  layout_circle(gr, geom = "bar", radius = 10, trackWidth = 4,
                aes(fill = score, y = score)) +
  layout_circle(gr, geom = "point", color = "red", radius = 14,
                trackWidth = 3, grid = TRUE, aes(y = score)) +
  layout_circle(gr, geom = "link", linked.to = "to.gr", radius = 6, trackWidth = 1)
</code></p>

<center><img title="vim-r" src="images/circularplot.png" /></center>

<h3 id="alignments-and-variants">Alignments and variants</h3>

<p>To make the following example work, please download and unpack this <a href="http://biocluster.ucr.edu/~tgirke/HTML_Presentations/Manuals/Workshop_Dec_12_16_2013/Rgraphics/data.zip">data archive</a> containing GFF, BAM and VCF sample files.</p>

<p><code>r
library(rtracklayer); library(GenomicFeatures); library(Rsamtools); library(GenomicAlignments); library(VariantAnnotation)
ga &lt;- readGAlignments("./data/SRR064167.fastq.bam", use.names=TRUE, param=ScanBamParam(which=GRanges("Chr5", IRanges(4000, 8000))))
p1 &lt;- autoplot(ga, geom = "rect")
p2 &lt;- autoplot(ga, geom = "line", stat = "coverage")
vcf &lt;- readVcf(file="data/varianttools_gnsap.vcf", genome="ATH1")
p3 &lt;- autoplot(vcf[seqnames(vcf)=="Chr5"], type = "fixed") + xlim(4000, 8000) + theme(legend.position = "none", axis.text.y = element_blank(), axis.ticks.y=element_blank())
txdb &lt;- makeTxDbFromGFF(file="./data/TAIR10_GFF3_trunc.gff", format="gff3")
p4 &lt;- autoplot(txdb, which=GRanges("Chr5", IRanges(4000, 8000)), names.expr = "gene_id")
tracks(Reads=p1, Coverage=p2, Variant=p3, Transcripts=p4, heights = c(0.3, 0.2, 0.1, 0.35)) + ylab("")
</code></p>

<p><img src="Rgraphics_files/figure-html/ggbio_align_variants-1.png" alt="" /><!-- --></p>

<h2 id="additional-examples">Additional examples</h2>

<p>See <code>autoplot</code> demo <a href="http://www.tengfei.name/ggbio/docs/man/autoplot-method.html">here</a></p>

<h2 id="additional-genome-graphics">Additional genome graphics</h2>

<ul>
  <li><a href="http://www.bioconductor.org/packages/devel/bioc/html/Gviz.html"><code>Gviz</code></a></li>
  <li><a href="http://cran.us.r-project.org/web/packages/RCircos/index.html"><code>RCircos</code></a> [@Zhang2013-zn]</li>
  <li><a href="http://bioconductor.org/packages/release/bioc/html/GenomeGraphs.html"><code>Genome Graphs</code></a></li>
  <li><a href="http://genoplotr.r-forge.r-project.org/"><code>genoPlotR</code></a></li>
</ul>

<h2 id="genome-browser-igv">Genome Browser: IGV</h2>

<p>View genome data in IGV</p>

<ul>
  <li>Download and open <a href="http://www.broadinstitute.org/igv/download">IGV</a></li>
  <li>Select in menu in top left corner <em>A. thaliana</em> (TAIR10)</li>
  <li>Upload the following indexed/sorted Bam files with <code>File -&gt; Load from URL...</code></li>
</ul>

<p><code>
http://faculty.ucr.edu/~tgirke/HTML_Presentations/Manuals/Workshop_Dec_6_10_2012/Rrnaseq/results/SRR064154.fastq.bam
http://faculty.ucr.edu/~tgirke/HTML_Presentations/Manuals/Workshop_Dec_6_10_2012/Rrnaseq/results/SRR064155.fastq.bam
http://faculty.ucr.edu/~tgirke/HTML_Presentations/Manuals/Workshop_Dec_6_10_2012/Rrnaseq/results/SRR064166.fastq.bam
http://faculty.ucr.edu/~tgirke/HTML_Presentations/Manuals/Workshop_Dec_6_10_2012/Rrnaseq/results/SRR064167.fastq.bam
</code></p>

<ul>
  <li>To view area of interest, enter its coordinates <code>Chr1:49,457-51,457</code> in position menu on top.</li>
</ul>

<center><img title="vim-r" src="images/igv_peak_3.png" /></center>

<h3 id="create-symbolic-links">Create symbolic links</h3>

<p>For viewing BAM files in IGV as part of <code>systemPipeR</code> workflows.</p>

<ul>
  <li><code>systemPipeR</code>: utilities for building NGS analysis <a href="https://github.com/tgirke/systemPipeR">pipelines</a></li>
</ul>

<p><code>r
library("systemPipeR")
symLink2bam(sysargs=args, htmldir=c("~/.html/", "somedir/"), 
            urlbase="http://myserver.edu/~username/",
            urlfile="IGVurl.txt")
</code></p>

<h3 id="controlling-igv-from-r">Controlling IGV from R</h3>

<p>Note this may not work on all systems.</p>

<p><code>r
library(SRAdb)
startIGV("lm")
sock &lt;- IGVsocket()
session &lt;- IGVsession(files=myurls, 
                      sessionFile="session.xml", 
                      genome="A. thaliana (TAIR10)")
IGVload(sock, session)
IGVgoto(sock, 'Chr1:45296-47019')
</code></p>

<h1 id="references">References</h1>


<div class="tags">
    
</div>



</div>



<footer>
            <div class="row">
                <div class="col-lg-12 footer">
               &copy;2016 UC Riverside. All rights reserved. <br />
 Site last generated: May 24, 2016 <br />
<p><img src="../common_images/company_logo.png/></p>
                </div>
            </div>
</footer>

            </div><!-- /.row -->

    </div>    <!-- /.container -->

</body>


</html>

