---
title: Student Paper Presentations
last_updated: 30-Apr-16
---

## Overview

Each student has been assigned one paper to present in class. The
structure of the paper presentations is outlined in this [Template Slideshow](https://docs.google.com/presentation/d/1VJzj9L75HdGel-mnAbNLYc9Wmdxg3aQP8YmrWG3AsTU/edit?usp=sharing). 
The schedule for the paper presentations is available on the internal [Course Planning Sheet](https://docs.google.com/spreadsheets/d/1OkmYxBt7SZp4Pba8LlRtMRB0JoTie4OJZmo9X-KtQqo/edit#gid=162605681). 
The following lists the assigned papers organized by course project topics.


## Publications organized by course projects

### ChIP-Seq1 - Motif Prediction    

+ McLeay, Robert C, and Timothy L Bailey. 2010. “Motif Enrichment Analysis: A Unified Framework and an Evaluation on ChIP Data.” BMC Bioinformatics 11: 165. [PubMed](http://www.ncbi.nlm.nih.gov/pubmed/20356413)
+ Frith, Martin C., Yutao Fu, Liqun Yu, Jiang‐fan Chen, Ulla Hansen, and Zhiping Weng. 2004. “Detection of Functional DNA Motifs via Statistical Over‐representation.” Nucleic Acids Research 32 (4): 1372–81. [PubMed](http://www.ncbi.nlm.nih.gov/pubmed/14988425)
+ Tompa, M, N Li, T L Bailey, G M Church, B De Moor, E Eskin, A V Favorov, et al. 2005. “Assessing Computational Tools for the Discovery of Transcription Factor Binding Sites.” Nature Biotechnology 23 (1): 137–44. [PubMed](http://www.ncbi.nlm.nih.gov/pubmed/15637633)


### ChiP-Seq2 - Peak Calling  

+ Landt SG, Marinov GK, Kundaje A, Kheradpour P, Pauli F, Batzoglou S, Bernstein BE, Bickel P, Brown JB, Cayting P, et al (2012) ChIP-seq guidelines and practices of the ENCODE and modENCODE consortia. Genome Res 22: 1813–1831 [PubMed](http://www.ncbi.nlm.nih.gov/pubmed/22955991)
+ Zhang Y, Liu T, Meyer CA, Eeckhoute J, Johnson DS, Bernstein BE, Nussbaum C, Myers RM, Brown M, Li W, et al (2008) Model-based analysis of ChIP-Seq (MACS). Genome Biol. doi: 10.1186/gb-2008-9-9-r137 [PubMed](http://www.ncbi.nlm.nih.gov/pubmed/18798982)
+ Wilbanks EG, Facciotti MT (2010) Evaluation of algorithm performance in ChIP-seq peak detection. PLoS One. doi: 10.1371/journal.pone.0011471 [PubMed](http://www.ncbi.nlm.nih.gov/pubmed/20628599)
+ Lun ATL, Smyth GK (2014) De novo detection of differentially bound regions for ChIP-seq data using peaks and windows: controlling error rates correctly. Nucleic Acids Res 42: e95 [PubMed](http://www.ncbi.nlm.nih.gov/pubmed/24852250)


### RNA-Seq1 - RNA-Seq Aligners 

+ Kim D, Pertea G, Trapnell C, Pimentel H, Kelley R, Salzberg SL (2013) TopHat2: accurate alignment of transcriptomes in the presence of insertions, deletions and gene fusions. Genome Biol. doi: 10.1186/gb-2013-14-4-r36 [PubMed](http://www.ncbi.nlm.nih.gov/pubmed/23618408)
+ Kim D, Langmead B, Salzberg SL (2015) HISAT: a fast spliced aligner with low memory requirements. Nat Methods 12: 357–360 [PubMed](http://www.ncbi.nlm.nih.gov/pubmed/25751142)
+ Liao Y, Smyth GK, Shi W (2013) The Subread aligner: fast, accurate and scalable read mapping by seed-and-vote. Nucleic Acids Res 41: e108 [PubMed](http://www.ncbi.nlm.nih.gov/pubmed/23558742)
+ Bray NL, Pimentel H, Melsted P, Pachter L (2016) Near-optimal probabilistic RNA-seq quantification. Nat Biotechnol. doi: 10.1038/nbt.3519 [PubMed](http://www.ncbi.nlm.nih.gov/pubmed/27043002)


### RNA-Seq2 - DEG Methods  

+ Guo Y, Li C-I, Ye F, Shyr Y (2013) Evaluation of read count based RNAseq analysis methods. BMC Genomics 14 Suppl 8: S2 [PubMed](http://www.ncbi.nlm.nih.gov/pubmed/24564449)
+ Robinson MD, McCarthy DJ, Smyth GK (2010) edgeR: a Bioconductor package for differential expression analysis of digital gene expression data. Bioinformatics 26: 139–140 [PubMed](http://www.ncbi.nlm.nih.gov/pubmed/19910308)
+ Love MI, Huber W, Anders S (2014) Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. Genome Biol 15: 550 [PubMed](http://www.ncbi.nlm.nih.gov/pubmed/25516281)
+ Zhou X, Lindsay H, Robinson MD (2014) Robustly detecting differential expression in RNA sequencing data using observation weights. Nucleic Acids Res 42: e91 [PubMed](http://www.ncbi.nlm.nih.gov/pubmed/24753412)


### VAR-Seq1 - Variant Calling  

+ Kumar RD, Chang L-W, Ellis MJ, Bose R (2013) Prioritizing Potentially Druggable Mutations with dGene: An Annotation Tool for Cancer Genome Sequencing Data. PLoS One 8: e67980 [PubMed](http://www.ncbi.nlm.nih.gov/pubmed/23826350)
+ DePristo MA, Banks E, Poplin R, Garimella KV, Maguire JR, Hartl C, Philippakis AA, del Angel G, Rivas MA, Hanna M, et al (2011) A framework for variation discovery and genotyping using next-generation DNA sequencing data. Nat Genet 43: 491–498 [PubMed](http://www.ncbi.nlm.nih.gov/pubmed/21478889)
+ Li H (2011) A statistical framework for SNP calling, mutation discovery, association mapping and population genetical parameter estimation from sequencing data. Bioinformatics 27: 2987–2993 [PubMed](http://www.ncbi.nlm.nih.gov/pubmed/21903627)
+ McKenna A, Hanna M, Banks E, Sivachenko A, Cibulskis K, Kernytsky A, Garimella K, Altshuler D, Gabriel S, Daly M, et al (2010) The Genome Analysis Toolkit: a MapReduce framework for analyzing next-generation DNA sequencing data. Genome Res 20: 1297–1303 [PubMed](http://www.ncbi.nlm.nih.gov/pubmed/20644199)


### VAR-Seq2 - Functional Prediction  

+ Shihab HA, Rogers MF, Gough J, Mort M, Cooper DN, Day INM, Gaunt TR, Campbell C (2015) An integrative approach to predicting the functional effects of non-coding and coding sequence variation. Bioinformatics 31: 1536–1543 [PubMed](http://www.ncbi.nlm.nih.gov/pubmed/25583119)
+ Kumar P, Henikoff S, Ng PC (2009) Predicting the effects of coding non-synonymous variants on protein function using the SIFT algorithm. Nat Protoc 4: 1073–1081 [PubMed](http://www.ncbi.nlm.nih.gov/pubmed/19561590)
+ Adzhubei I, Jordan DM, Sunyaev SR (2013) Predicting functional effect of human missense mutations using PolyPhen-2. Curr Protoc Hum Genet Chapter 7: Unit7.20 [PubMed](http://www.ncbi.nlm.nih.gov/pubmed/23315928)

