<!DOCTYPE html>
  <head>

 <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="">
<meta name="keywords" content=" ">
<title>Introduction to R  | GEN242</title>
<link rel="stylesheet" type="text/css" href="../css/syntax.css">
<link rel="stylesheet" type="text/css" href="../css/font-awesome.min.css">
<!--<link rel="stylesheet" type="text/css" href="../css/bootstrap.min.css">-->
<link rel="stylesheet" type="text/css" href="../css/modern-business.css">
<link rel="stylesheet" type="text/css" href="../css/lavish-bootstrap.css">
<link rel="stylesheet" type="text/css" href="../css/customstyles.css">
<link rel="stylesheet" type="text/css" href="../css/theme-blue.css">
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-cookie/1.4.1/jquery.cookie.min.js"></script>
<script src="../js/jquery.navgoco.min.js"></script>
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/js/bootstrap.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/anchor-js/2.0.0/anchor.min.js"></script>
<script src="../js/toc.js"></script>
<script src="../js/customscripts.js"></script>
<link rel="shortcut icon" href="../common_images/favicon.ico" type="image/x-icon">
<!-- HTML5 Shim and Respond.js IE8 support of HTML5 elements and media queries -->
<!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
<script src="https://oss.maxcdn.com/libs/respond.js/1.4.2/respond.min.js"></script>
<![endif]-->







 

<script>
  $(function () {
      $('[data-toggle="tooltip"]').tooltip()
  })
</script>



  </head>

<body>

   <!-- Navigation -->
<nav class="navbar navbar-inverse navbar-fixed-top" role="navigation">
    <div class="container topnavlinks">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <a class="fa fa-home fa-lg navbar-brand" href="home.html">&nbsp;<span class="projectTitle"> Data Analysis in Genome Biology</span></a>

        </div>
        <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
            <ul class="nav navbar-nav navbar-right">
                <!-- entries without drop-downs appear here -->
                <!-- conditional logic to control which topnav appears for the audience defined in the configuration file.-->
                



















                
                
                
                
                <li><a href="https://piazza.com/class/ilbslqb3j2ovy" target="_blank">Piazza</a></li>
                
                
                
                
                
                <li><a href="https://github.com/tgirke/GEN242" target="_blank">Github Repo</a></li>
                
                
                
                


                <!-- entries with drop-downs appear here -->
                <!-- conditional logic to control which topnav appears for the audience defined in the configuration file.-->

                <li class="dropdown">
                    
                    
                    
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Resources<b class="caret"></b></a>
                    <ul class="dropdown-menu">
                        
                        
                        
                        <li><a href="https://cran.r-project.org/" target="_blank">CRAN</a></li>
                        
                        
                        
                        
                        
                        <li><a href="http://bioconductor.org/" target="_blank">Bioconductor</a></li>
                        
                        
                        
                        
                        
                        <li><a href="http://manuals.bioinformatics.ucr.edu/home/R_BioCondManual" target="_blank">R & Bioconductor Manual</a></li>
                        
                        
                        
                        
                        
                        <li><a href="https://guides.github.com/activities/hello-world/" target="_blank">GitHub</a></li>
                        
                        
                        
                        
                        
                        <li><a href="http://jekyllrb.com/" target="_blank">Jekyll</a></li>
                        
                        
                        
                        
                        
                        <li><a href="http://girke.bioinformatics.ucr.edu/" target="_blank">Girke Lab</a></li>
                        
                        
                        
                        
                        
                        <li><a href="http://idratherbewriting.com/documentation-theme-jekyll" target="_blank">Jekyll Doc Theme</a></li>
                        
                        
                        
                        

                    </ul>
                </li>
                
                


                <!-- special insertion -->

               
                
                
                <!-- Send feedback function -->
<script>
function SendLinkByMail(href) {
var subject= "GEN242 feedback";
var body = "I have some feedback about the Introduction to R page: ";
body += window.location.href;
body += "";
var uri = "mailto:?subject=";
uri += encodeURIComponent(subject);
uri += "&body=";
uri += encodeURIComponent(body);
window.location.href = uri;
}
</script>

<li><a href="javascript:(function()%7BSendLinkByMail()%3B%7D)()%3B" target="_blank"><i class="fa fa-envelope-o"></i> Feedback</a></li>


                <!--uncomment this block if you want simple search instead of algolia-->
                <li>
                     <!--start search-->
                    <div id="search-demo-container">
                        <input type="text" id="search-input" placeholder="search...">
                        <ul id="results-container"></ul>
                    </div>
                    <script src="../js/jekyll-search.js" type="text/javascript"></script>
                    <script type="text/javascript">
                        SimpleJekyllSearch.init({
                            searchInput: document.getElementById('search-input'),
                            resultsContainer: document.getElementById('results-container'),
                            dataSource: '../search.json',
                            searchResultTemplate: '<li><a href="{url}" title="Introduction to R">{title}</a></li>',
                        noResultsText: 'No results found.',
                                limit: 10,
                                fuzzy: true,
                        })
                    </script>
                     <!--end search-->
                </li>

            
        </div>
        <!-- /.container -->
</nav>


    <!-- Page Content -->
    <div class="container">
        <div class="col-lg-12">&nbsp;</div>


<!-- Content Row -->
<div class="row">
    <!-- Sidebar Column -->
    <div class="col-md-3">

        <script>

            $(document).ready(function() {
                // Initialize navgoco with default options
                $("#mysidebar").navgoco({
                    caretHtml: '',
                    accordion: true,
                    openClass: 'active', // open
                    save: false, // leave false or nav highlighting doesn't work right
                    cookie: {
                        name: 'navgoco',
                        expires: false,
                        path: '/'
                    },
                    slide: {
                        duration: 400,
                        easing: 'swing'
                    }
                });

                $("#collapseAll").click(function(e) {
                    e.preventDefault();
                    $("#mysidebar").navgoco('toggle', false);
                });

                $("#expandAll").click(function(e) {
                    e.preventDefault();
                    $("#mysidebar").navgoco('toggle', true);
                });

            });

        </script>


        



















        <ul id="mysidebar" class="nav">

            <span class="siteTagline">GEN242</span>
            <span class="versionTagline">Spring 2016</span>

            
            
            
                
            
            <li><a href="#">Home</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/home.html">Introduction</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_syllabus.html">Syllabus</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Internal</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_internal.html">Internal Resources</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Schedule</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_schedule.html">Course Schedule</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Slides</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_01.html">Course Introduction</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_02.html">Molbiol for Bioinformatics</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_03.html">Databases and Software</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_04.html">DNA Sequencing</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_05.html">Introduction to R</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_06.html">Sequence Alignments</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_07.html">Programming in R</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_08.html">Multiple Alignments</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_09.html">Short Read Alignments</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_10.html">NGS Analysis Basics</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_slides_11.html">Gene Expression Analysis</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Homework</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_homework_01.html">HW1 - Online Tools</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_homework_02.html">HW2 - Biocluster/Linux</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_homework_03.html">HW3 - Introduction to R</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_homework_04.html">HW4 - Pairwise Alignments</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_09.html">HW5 - Programming in R</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_08.html">HW6 - Sequence Analysis</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">GitHub</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_tutorial_01.html">GitHub</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Biocluster</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_tutorial_02.html">Biocluster</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Introduction to R</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_01.html">1. Overview</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_02.html">2. R Package Repositories</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_03.html">3. Installation of R and Add-on Packages</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_04.html">4. Getting Around</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_05.html">5. Basic Syntax</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_06.html">6. Data Types</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_07.html">7. Data objects</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_08.html">8. Important Utilities</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_09.html">9. Operators and Calculations</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_10.html">10. Reading and Writing External Data</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_11.html">11. Useful R Functions</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_12.html">12. Graphics in R</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_13.html">13. Analysis Routine</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_14.html">14. Session Info</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rbasics_15.html">15. References</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Programming in R</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_01.html">1. Overview</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_02.html">2. Control Structures</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_03.html">3. Loops</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_04.html">4. Functions</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_05.html">5. Useful Utilities</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_06.html">6. Running R Scripts</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_07.html">7. Building R Packages</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_08.html">8. Programming Exercises</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_09.html">9. Homework 5</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_10.html">10. Session Info</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Programming_in_R_11.html">11. References</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">NGS Analysis Basics</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_01.html">1. Overview</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_02.html">2. Package Requirements</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_03.html">3. Strings in R Base</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_04.html">4. Sequences in Bioconductor</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_05.html">5. NGS Sequences</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_06.html">6. Range Operations</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_07.html">7. Transcript Ranges</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_08.html">8. Homework 6</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_09.html">9. Session Info</a></li>
                    

                    

                    
                    
                    
                    
                    <li><a href="../mydoc/mydoc_Rsequences_10.html">10. References</a></li>
                    

                    

                    
                    
                </ul>
                
                
            
            <li><a href="#">Projects</a>
                <ul>
                    
                    
                    
                    <li><a href="../mydoc/mydoc_project_01.html">Project 1</a></li>
                    

                    

                    
                    
                </ul>
                
                
                


                <!-- if you aren't using the accordion, uncomment this block:

                     <p class="external">
                         <a href="#" id="collapseAll">Collapse All</a> | <a href="#" id="expandAll">Expand All</a>
                     </p>
                 -->

    </div>
    <!-- this highlights the active parent class in the navgoco sidebar. this is critical so that the parent expands when you're viewing a page. This must appear below the sidebar code above. Otherwise, if placed inside customscripts.js, the script runs before the sidebar code runs and the class never gets inserted.-->
    <script>$("li.active").parents('li').toggleClass("active");</script>


            <!-- Content Column -->
            <div class="col-md-9">
                
                <div class="post-header">
   <h1 class="post-title-main">Introduction to R</h1>
</div>

<div class="post-content">

   

<!-- this handles the automatic toc. use ## for subheads to auto-generate the on-page minitoc. if you use html tags, you must supply an ID for the heading element in order for it to appear in the minitoc. -->
<script>
$( document ).ready(function() {
  // Handler for .ready() called.

$('#toc').toc({ minimumHeaders: 0, listType: 'ul', showSpeed: 0, headers: 'h2,h3,h4' });

/* this offset helps account for the space taken up by the floating toolbar. */
$('#toc').on('click', 'a', function() {
  var target = $(this.getAttribute('href'))
    , scroll_target = target.offset().top

  $(window).scrollTop(scroll_target - 10);
  return false
})
  
});
</script>


<div id="toc"></div>


<!--
    
    <a target="_blank" href="https://github.com/tgirke/manuals/tree/master/mydoc//vignettes/08_Rbasics/Rbasics.Rmd" class="btn btn-default githubEditButton" role="button"><i class="fa fa-github fa-lg"></i> Edit me</a>
    
-->
<!--
%% \VignetteEngine{knitr::rmarkdown}
%\VignetteIndexEntry{Overview Vignette}
%% \VignetteDepends{methods}
%% \VignetteKeywords{compute cluster, pipeline, reports}
%% \VignettePackage{longevityTools}
-->

<!---
- Compile from command-line
echo "rmarkdown::render('Rbasics.Rmd', clean=F)" | R -slave; R CMD Stangle Rbasics.Rmd; Rscript ../md2jekyll.R Rbasics.knit.md 8

- Commit to github
git commit -am "some edits"; git push -u origin master

- To customize font size and other style features, add this line to output section in preamble:  
    css: style.css
-->

```{r style, echo = FALSE, results = 'asis'}
BiocStyle::markdown()
options(width=100, max.print=1000)
knitr::opts_chunk$set(
    eval=as.logical(Sys.getenv("KNITR_EVAL", "TRUE")),
    cache=as.logical(Sys.getenv("KNITR_CACHE", "TRUE")))
```

```{r setup, echo=FALSE, messages=FALSE, warnings=FALSE}
suppressPackageStartupMessages({
    library(limma) 
    library(ggplot2) }) 
```

# Overview

## What is R?

[R](http://cran.at.r-project.org) is a powerful statistical environment and
programming language for the analysis and visualization of data.  The
associated [Bioconductor](http://bioconductor.org/) and CRAN package
repositories provide many additional R packages for statistical data analysis
for a wide array of research areas. The R software is free and runs on all
common operating systems. 

## Why Using R?
* Complete statistical environment and programming language
* Efficient functions and data structures for data analysis
* Powerful graphics
* Access to fast growing number of analysis packages
* Most widely used language in bioinformatics
* Is standard for data mining and biostatistical analysis
* Technical advantages: free, open-source, available for all OSs

## Books and Documentation
* simpleR - Using R for Introductory Statistics (John Verzani, 2004) - [URL](http://cran.r-project.org/doc/contrib/Verzani-SimpleR.pdf)
* Bioinformatics and Computational Biology Solutions Using R and Bioconductor (Gentleman et al., 2005) - [URL](http://www.bioconductor.org/help/publications/books/bioinformatics-and-computational-biology-solutions/)
* More on this see "Finding Help" section in UCR Manual - [URL](http://manuals.bioinformatics.ucr.edu/home/R\_BioCondManual\#TOC-Finding-Help)

## R Working Environments

## Working environments (IDEs) for R
<center><img title="R_Interfaces" src="images/rinterface.png"/></center>
<center> R Projects and Interfaces</center>

Some R working environments with support for syntax highlighting and utilities to send code 
to the R console: 

* [RStudio](https://www.rstudio.com/products/rstudio/features): excellent choice for beginners ([Cheat Sheet](http://www.rstudio.com/wp-content/uploads/2016/01/rstudio-IDE-cheatsheet.pdf)) 
* Basic R code editors provided by Rguis 
* [gedit](https://wiki.gnome.org/Apps/Gedit), [Rgedit](http://rgedit.sourceforge.net/), [RKWard](https://rkward.kde.org/), [Eclipse](http://www.walware.de/goto/statet), [Tinn-R](http://www.sciviews.org/Tinn-R/), [Notepad++](https://notepad-plus-plus.org/), [NppToR](http://sourceforge.net/projects/npptor/)
* [Vim-R-Tmux](http://manuals.bioinformatics.ucr.edu/home/programming-in-r/vim-r): R working environment based on vim and tmux 
* [Emacs](http://www.xemacs.org/Download/index.html) ([ESS add-on package](http://ess.r-project.org/))
	
### Example: RStudio 

New integrated development environment (IDE) for [R](http://www.rstudio.com/ide/download/). Highly functional for both beginners and 
advanced.

<center><img title="RStudio" src="images/rstudio.png"/></center>
<center> RStudio IDE</center>

Some userful shortcuts: `Ctrl+Enter` (send code), `Ctrl+Shift+C` (comment/uncomment), `Ctrl+1/2` (switch window focus)

### Example: Vim-R-Tmux

Terminal-based Working Environment for R: [Vim-R-Tmux](http://manuals.bioinformatics.ucr.edu/home/programming-in-r/vim-r)

<center><img title="Vim-R-Tmux" src="images/screenshot.png" ></center>
<center>Vim-R-Tmux IDE for R</center>

# R Package Repositories

* CRAN (>8,000 packages) general data analysis - [URL](http://cran.at.r-project.org/)
* Bioconductor (>1,100 packages) bioscience data analysis - [URL](http://www.bioconductor.org/)
* Omegahat (>90 packages) programming interfaces - [URL](https://github.com/omegahat?tab=repositories)

# Installation of R and Add-on Packages

(1.) Install R for your operating system from [CRAN](http://cran.at.r-project.org/).

(2.) Install RStudio from [RStudio](http://www.rstudio.com/ide/download).

(3.) Install CRAN Packages from R console like this:

```{r install_cran, eval=FALSE}
install.packages(c("pkg1", "pkg2")) 
install.packages("pkg.zip", repos=NULL)
```

(4.) Install Bioconductor packages as follows:

```{r install_bioc, eval=FALSE}
source("http://www.bioconductor.org/biocLite.R")
library(BiocInstaller)
BiocVersion()
biocLite()
biocLite(c("pkg1", "pkg2"))
```

(5.) For more details consult the [Bioc Install page](http://www.bioconductor.org/install/)
and [BiocInstaller](http://www.bioconductor.org/packages/release/bioc/html/BiocInstaller.html) package.

# Getting Around

## Startup and Closing Behavior

* __Starting R__:
    The R GUI versions, including RStudio, under Windows and Mac OS X can be
    opened by double-clicking their icons. Alternatively, one can start it by
    typing `R` in a terminal (default under Linux). 

* __Startup/Closing Behavior__:
    The R environment is controlled by hidden files in the startup directory:
    `.RData`, `.Rhistory` and `.Rprofile` (optional). 
	
    
* __Closing R__:

```{r closing_r, eval=FALSE}
q()  
```
```
Save workspace image? [y/n/c]:
```
        
* __Note__:
    When responding with `y`, then the entire R workspace will be written to
    the `.RData` file which can become very large. Often it is sufficient to just
    save an analysis protocol in an R source file. This way one can quickly
    regenerate all data sets and objects. 


## Navigating directories

Create an object with the assignment operator `<-` or `=`
```{r r_assignment, eval=FALSE}
object <- ...
```

List objects in current R session
```{r r_ls, eval=FALSE}
ls()
```

Return content of current working directory
```{r r_dirshow, eval=FALSE}
dir()
```

Return path of current working directory
```{r r_dirpath, eval=FALSE}
getwd()
```

Change current working directory
```{r r_setwd, eval=FALSE}
setwd("/home/user")
```

# Basic Syntax

General R command syntax

```{r r_syntax, eval=FALSE}
object <- function_name(arguments) 
object <- object[arguments] 
```

Finding help

```{r r_find_help, eval=FALSE}
?function_name
```

Load a library/package

```{r r_package_load, eval=FALSE}
library("my_library") 
```

List functions defined by a library

```{r r_package_functions, eval=FALSE}
library(help="my_library")
```

Load library manual (PDF or HTML file)

```{r r_load_vignette, eval=FALSE}
vignette("my_library") 
```

Execute an R script from within R

```{r r_execute_script, eval=FALSE}
source("my_script.R")
```

Execute an R script from command-line (the first of the three options is preferred)

```{sh sh_execute_script, eval=FALSE}
$ Rscript my_script.R
$ R CMD BATCH my_script.R 
$ R --slave < my_script.R 
```

# Data Types 

## Numeric data

Example: `1, 2, 3, ...`

```{r r_numeric_data, eval=TRUE}

x <- c(1, 2, 3)
x
is.numeric(x)
as.character(x)
```

## Character data

Example: `"a", "b", "c", ...`

```{r r_character_data, eval=TRUE}
x <- c("1", "2", "3")
x
is.character(x)
as.numeric(x)
```

## Complex data

Example: mix of both

```{r r_complex_data, eval=TRUE}
c(1, "b", 3)
```

## Logical data

Example: `TRUE` of `FALSE`

```{r r_logical_data, eval=TRUE}
x <- 1:10 < 5
x  
!x
which(x) # Returns index for the 'TRUE' values in logical vector
```

# Data objects

## Object types

## Vectors (1D)

Definition: `numeric` or `character`

```{r r_vector_object, eval=TRUE}
myVec <- 1:10; names(myVec) <- letters[1:10]  
myVec[1:5]
myVec[c(2,4,6,8)]
myVec[c("b", "d", "f")]
```

## Factors (1D)

Definition: vectors with grouping information

```{r r_factor_object, eval=TRUE}
factor(c("dog", "cat", "mouse", "dog", "dog", "cat"))
```

## Matrices (2D)

Definition: two dimensional structures with data of same type

```{r r_matrix_object, eval=TRUE}
myMA <- matrix(1:30, 3, 10, byrow = TRUE) 
class(myMA)
myMA[1:2,]
myMA[1, , drop=FALSE]
```

## Data Frames (2D)

Definition: two dimensional objects with data of variable types

```{r r_dataframe_object, eval=TRUE}
myDF <- data.frame(Col1=1:10, Col2=10:1) 
myDF[1:2, ]
```

## Arrays

Definition: data structure with one, two or more dimensions


## Lists

Definition: containers for any object type

```{r r_list_object, eval=TRUE}
myL <- list(name="Fred", wife="Mary", no.children=3, child.ages=c(4,7,9)) 
myL
myL[[4]][1:2] 
```

## Functions

Definition: piece of code

```{r r_function_object, eval=FALSE}
myfct <- function(arg1, arg2, ...) { 
	function_body 
}
```

## Subsetting of data objects

__(1.) Subsetting by positive or negative index/position numbers__

```{r r_subset_by_index, eval=TRUE}
myVec <- 1:26; names(myVec) <- LETTERS 
myVec[1:4]
```

__(2.) Subsetting by same length logical vectors__

```{r r_subset_by_logical, eval=TRUE}
myLog <- myVec > 10
myVec[myLog] 
```	

__(3.) Subsetting by field names__

```{r r_subset_by_names, eval=TRUE}
myVec[c("B", "K", "M")]
```

__(4.) Subset with `$` sign__: references a single column or list component by its name 

```{r r_subset_by_dollar, eval=TRUE}
iris$Species[1:8]
```

# Important Utilities
	
## Combining Objects

The `c` function combines vectors and lists

```{r r_combine_vectors, eval=TRUE}
c(1, 2, 3)
x <- 1:3; y <- 101:103
c(x, y)
iris$Species[1:8]
```

The `cbind` and `rbind` functions can be used to append columns and rows, respecively.
```{r r_cbind_rbind, eval=TRUE}
ma <- cbind(x, y)
ma
rbind(ma, ma)
```

## Accessing Dimensions of Objects

Length and dimension information of objects

```{r r_length_dim, eval=TRUE}
length(iris$Species)
dim(iris)
```

## Accessing Name Slots of Objects

Accessing row and column names of 2D objects
```{r col_row_names, eval=TRUE}
rownames(iris)[1:8]
colnames(iris)
```

Return name field of vectors and lists
```{r name_slots, eval=TRUE}
names(myVec)
names(myL)
```

## Sorting Objects

The function `sort` returns a vector in ascending or descending order
```{r sort_objects, eval=TRUE}
sort(10:1)
```

The function `order` returns a sorting index for sorting an object
```{r order_objects, eval=TRUE}
sortindex <- order(iris[,1], decreasing = FALSE)
sortindex[1:12]
iris[sortindex,][1:2,]
sortindex <- order(-iris[,1]) # Same as decreasing=TRUE
```
Sorting multiple columns
```{r order_columns, eval=TRUE}
iris[order(iris$Sepal.Length, iris$Sepal.Width),][1:2,]
```

# Operators and Calculations

## Comparison Operators

Comparison operators: `==`, `!=`, `<`, `>`, `<=`, `>=`
```{r comparison_operators, eval=TRUE}
1==1
```
Logical operators: AND: `&`, OR: `|`, NOT: `!`
```{r logical_operators, eval=TRUE}
x <- 1:10; y <- 10:1
x > y & x > 5
```

## Basic Calculations

To look up math functions, see Function Index [here](http://cran.at.r-project.org/doc/manuals/R-intro.html#Function-and-variable-index)
```{r logical_calculations, eval=TRUE}
x + y
sum(x)
mean(x)
apply(iris[1:6,1:3], 1, mean) 
```

# Reading and Writing External Data
## Import data from tabular files into R

```{r read_delim, eval=FALSE}
myDF <- read.delim("myData.xls", sep="\t")
```

## Export data from R to tabular files
```{r write_table, eval=FALSE}
write.table(myDF, file="myfile.xls", sep="\t", quote=FALSE, col.names=NA)
```

## Copy and paste into R

On Windows/Linux systems
```{r paste_windows, eval=FALSE}
read.delim("clipboard") 
```
On Mac OS X systems
```{r paste_osx, eval=FALSE}
read.delim(pipe("pbpaste")) 
```

## Copy and paste from R 

On Windows/Linux systems
```{r copy_windows, eval=FALSE}
write.table(iris, "clipboard", sep="\t", col.names=NA, quote=F) 
```

On Mac OS X systems
```{r copy_osx, eval=FALSE}
zz <- pipe('pbcopy', 'w')
write.table(iris, zz, sep="\t", col.names=NA, quote=F)
close(zz) 
```

## Homework 3A 

Homework 3A: [Object Subsetting Routines and Import/Export](http://girke.bioinformatics.ucr.edu/GEN242/mydoc/mydoc_homework_03.html)

# Useful R Functions

## Unique entries

Make vector entries unique with `unique`

```{r unique, eval=TRUE}
length(iris$Sepal.Length)
length(unique(iris$Sepal.Length))
```

## Count occurrences

Count occurrences of entries with `table`
```{r table, eval=TRUE}
table(iris$Species)
```

## Aggregate data

Compute aggregate statistics with `aggregate`
```{r aggregate, eval=TRUE}
aggregate(iris[,1:4], by=list(iris$Species), FUN=mean, na.rm=TRUE)
```

## Intersect data

Compute intersect between two vectors with `%in%`
```{r intersect, eval=TRUE}
month.name %in% c("May", "July")
```

## Merge data frames

Join two data frames by common field entries with `merge` (here row names `by.x=0`). To obtain only the common rows, change `all=TRUE` to `all=FALSE`. To merge on specific columns, refer to them by their position numbers or their column names.
```{r merge, eval=TRUE}
frame1 <- iris[sample(1:length(iris[,1]), 30), ]
frame1[1:2,]
dim(frame1)
my_result <- merge(frame1, iris, by.x = 0, by.y = 0, all = TRUE)
dim(my_result)
```

# Graphics in R

## Advantages

- Powerful environment for visualizing scientific data
- Integrated graphics and statistics infrastructure
- Publication quality graphics
- Fully programmable 
- Highly reproducible
- Full [LaTeX](http://www.latex-project.org/) and Markdown support via `knitr` and `R markdown`
- Vast number of R packages with graphics utilities

## Documentation for R Graphics

__General__

- Graphics Task Page - [URL](http://cran.r-project.org/web/views/Graphics.html)
- R Graph Gallery - [URL](http://addictedtor.free.fr/graphiques/allgraph.php)
- R Graphical Manual - [URL](http://cged.genes.nig.ac.jp/RGM2/index.php)
- Paul Murrell's book R (Grid) Graphics - [URL](http://www.stat.auckland.ac.nz/~paul/RGraphics/rgraphics.html)

__Interactive graphics__

- rggobi` (GGobi) - [URL](http://www.ggobi.org/)
- `iplots` - [URL](http://www.rosuda.org/iplots/)
- Open GL (`rgl`) - [URL](http://rgl.neoscientists.org/gallery.shtml)

## Graphics Environments

__Viewing and saving graphics in R__

- On-screen graphics
- postscript, pdf, svg
- jpeg, png, wmf, tiff, ...

__Four major graphic environments__

(a) Low-level infrastructure

- R Base Graphics (low- and high-level)
- `grid`: [Manual](http://www.stat.auckland.ac.nz/~paul/grid/grid.html)
        
(b) High-level infrastructure
        \begin{itemize}
- `lattice`: [Manual](http://lmdvr.r-forge.r-project.org), [Intro](http://www.his.sunderland.ac.uk/~cs0her/Statistics/UsingLatticeGraphicsInR.htm), [Book](http://www.amazon.com/Lattice-Multivariate-Data-Visualization-Use/dp/0387759689)
- `ggplot2`: [Manual](http://had.co.nz/ggplot2/), [Intro](http://www.ling.upenn.edu/~joseff/rstudy/summer2010_ggplot2_intro.html), [Book](http://had.co.nz/ggplot2/book/)

## Base Graphics: Overview

__Important high-level plotting functions__

- `plot`: generic x-y plotting
- `barplot`: bar plots
- `boxplot`: box-and-whisker plot
- `hist`: histograms
- `pie`: pie charts
- `dotchart`: cleveland dot plots
- `image, heatmap, contour, persp`: functions to generate image-like plots
- `qqnorm, qqline, qqplot`: distribution comparison plots
- `pairs, coplot`: display of multivariant data

__Help on graphics functions__

- `?myfct`
- `?plot`
- `?par`

### Preferred Object Types

- Matrices and data frames
- Vectors
- Named vectors

## Scatter Plots

### Basic Scatter Plot

Sample data set for subsequent plots

```{r sample_data, eval=TRUE}
set.seed(1410)
y <- matrix(runif(30), ncol=3, dimnames=list(letters[1:10], LETTERS[1:3]))
```

Plot data
```{r basic_scatter_plot, eval=TRUE}
plot(y[,1], y[,2]) 
```

### All pairs

```{r pairs_scatter_plot, eval=TRUE}
pairs(y) 
```

### With labels

```{r labels_scatter_plot, eval=TRUE}
plot(y[,1], y[,2], pch=20, col="red", main="Symbols and Labels")
text(y[,1]+0.03, y[,2], rownames(y))
```

## More examples

__Print instead of symbols the row names__

```{r row_scatter_plot, eval=TRUE}
plot(y[,1], y[,2], type="n", main="Plot of Labels")
text(y[,1], y[,2], rownames(y)) 
```

__Usage of important plotting parameters__

```{r plot_usage, eval=FALSE}
grid(5, 5, lwd = 2) 
op <- par(mar=c(8,8,8,8), bg="lightblue")
plot(y[,1], y[,2], type="p", col="red", cex.lab=1.2, cex.axis=1.2, 
     cex.main=1.2, cex.sub=1, lwd=4, pch=20, xlab="x label", 
     ylab="y label", main="My Main", sub="My Sub")
par(op)
```
__Important arguments_

- `mar`: specifies the margin sizes around the plotting area in order: `c(bottom, left, top, right)` 
- `col`: color of symbols
- `pch`: type of symbols, samples: `example(points)`
- `lwd`: size of symbols
- `cex.*`: control font sizes
- For details see `?par`


### Add regression line 

```{r plot_regression, eval=TRUE}
plot(y[,1], y[,2])
myline <- lm(y[,2]~y[,1]); abline(myline, lwd=2) 
summary(myline) 
```

### Log scale

Same plot as above, but on log scale

```{r plot_regression_log, eval=TRUE}
plot(y[,1], y[,2], log="xy") 
```

### Add a mathematical expression

```{r plot_regression_math, eval=TRUE}
plot(y[,1], y[,2]); text(y[1,1], y[1,2], expression(sum(frac(1,sqrt(x^2*pi)))), cex=1.3) 
```

## Homework 3B 

Homework 3B: [Scatter Plots](http://girke.bioinformatics.ucr.edu/GEN242/mydoc/mydoc_homework_03.html)


## Line Plots

### Single data set

```{r plot_line_single, eval=TRUE}
plot(y[,1], type="l", lwd=2, col="blue") 
```

### Many Data Sets

Plots line graph for all columns in data frame `y`. The `split.screen` function is used in this example in a for loop to overlay several line graphs in the same plot. 

```{r plot_line_many, eval=TRUE}
split.screen(c(1,1)) 
plot(y[,1], ylim=c(0,1), xlab="Measurement", ylab="Intensity", type="l", lwd=2, col=1)
for(i in 2:length(y[1,])) { 
	screen(1, new=FALSE)
	plot(y[,i], ylim=c(0,1), type="l", lwd=2, col=i, xaxt="n", yaxt="n", ylab="", xlab="", main="", bty="n") 
}
close.screen(all=TRUE) 
```

## Bar Plots 

### Basics

```{r plot_bar_simple, eval=TRUE}
barplot(y[1:4,], ylim=c(0, max(y[1:4,])+0.3), beside=TRUE, legend=letters[1:4]) 
text(labels=round(as.vector(as.matrix(y[1:4,])),2), x=seq(1.5, 13, by=1) + sort(rep(c(0,1,2), 4)), y=as.vector(as.matrix(y[1:4,]))+0.04) 
```
    
### Error Bars

```{r plot_bar_error, eval=TRUE}
bar <- barplot(m <- rowMeans(y) * 10, ylim=c(0, 10))
stdev <- sd(t(y))
arrows(bar, m, bar, m + stdev, length=0.15, angle = 90)
```

## Histograms

```{r plot_hist, eval=TRUE}
hist(y, freq=TRUE, breaks=10)
```

## Density Plots

```{r plot_dens, eval=TRUE}
plot(density(y), col="red")
```

## Pie Charts

```{r plot_pie, eval=TRUE}
pie(y[,1], col=rainbow(length(y[,1]), start=0.1, end=0.8), clockwise=TRUE)
legend("topright", legend=row.names(y), cex=1.3, bty="n", pch=15, pt.cex=1.8, 
col=rainbow(length(y[,1]), start=0.1, end=0.8), ncol=1) 
```

## Color Selection Utilities

Default color palette and how to change it

```{r color_palette, eval=TRUE}
palette()
palette(rainbow(5, start=0.1, end=0.2))
palette()
palette("default")
```

The `gray` function allows to select any type of gray shades by providing values from 0 to 1
```{r color_grey, eval=TRUE}
gray(seq(0.1, 1, by= 0.2))
```

Color gradients with `colorpanel` function from `gplots` library`
```{r color_gradient, eval=TRUE}
library(gplots)
colorpanel(5, "darkblue", "yellow", "white")
```
Much more on colors in R see Earl Glynn's color chart [here](http://research.stowers-institute.org/efg/R/Color/Chart/)


## Saving Graphics to File

After the `pdf()` command all graphs are redirected to file `test.pdf`. Works for all common formats similarly: jpeg, png, ps, tiff, ...
```{r save_graphics, eval=FALSE}
pdf("test.pdf")
plot(1:10, 1:10)
dev.off() 
```

Generates Scalable Vector Graphics (SVG) files that can be edited in vector graphics programs, such as InkScape.

```{r save_graphics_svg, eval=FALSE}
library("RSvgDevice")
devSVG("test.svg")
plot(1:10, 1:10)
dev.off() 
```

## Homework 3C

Homework 3C: [Bar Plots](http://girke.bioinformatics.ucr.edu/GEN242/mydoc/mydoc_homework_03.html)

# Analysis Routine

## Overview

The following exercise introduces a variety of useful data analysis utilities in R. 

## Analysis Routine: Data Import

- __Step 1__: To get started with this exercise, direct your R session to a dedicated workshop directory and download into this directory the following sample tables. Then import the files into Excel and save them as tab delimited text files.

    - [MolecularWeight_tair7.xls](http://faculty.ucr.edu/~tgirke/Documents/R_BioCond/Samples/MolecularWeight_tair7.xls)
    - [TargetP_analysis_tair7.xls](http://faculty.ucr.edu/~tgirke/Documents/R_BioCond/Samples/TargetP_analysis_tair7.xls)

__Import the tables into R__

Import molecular weight table

```{r import_data1, eval=FALSE}
my_mw <- read.delim(file="MolecularWeight_tair7.xls", header=T, sep="\t") 
my_mw[1:2,]
``` 

Import subcelluar targeting table
```{r import_data2, eval=FALSE}
my_target <- read.delim(file="TargetP_analysis_tair7.xls", header=T, sep="\t") 
my_target[1:2,]
```

Online import of molecular weight table
```{r import_data1b, eval=TRUE}
my_mw <- read.delim(file="http://faculty.ucr.edu/~tgirke/Documents/R_BioCond/Samples/MolecularWeight_tair7.xls", header=T, sep="\t") 
my_mw[1:2,]
``` 

Online import of subcelluar targeting table
```{r import_data2b, eval=TRUE}
my_target <- read.delim(file="http://faculty.ucr.edu/~tgirke/Documents/R_BioCond/Samples/TargetP_analysis_tair7.xls", header=T, sep="\t") 
my_target[1:2,]
```

## Merging Data Frames

- __Step 2__: Assign uniform gene ID column titles

```{r col_names_uni, eval=TRUE}
colnames(my_target)[1] <- "ID"
colnames(my_mw)[1] <- "ID" 
```

- __Step 3__: Merge the two tables based on common ID field

```{r merge_tables, eval=TRUE}
my_mw_target <- merge(my_mw, my_target, by.x="ID", by.y="ID", all.x=T)
```

- __Step 4__: Shorten one table before the merge and then remove the non-matching rows (NAs) in the merged file

```{r merge_tables_shorten, eval=TRUE}
my_mw_target2a <- merge(my_mw, my_target[1:40,], by.x="ID", by.y="ID", all.x=T)  # To remove non-matching rows, use the argument setting 'all=F'.
my_mw_target2 <- na.omit(my_mw_target2a) # Removes rows containing "NAs" (non-matching rows).
```

- __Homework 3D__: How can the merge function in the previous step be executed so that only the common rows among the two data frames are returned? Prove that both methods - the two step version with `na.omit` and your method - return identical results. 
- __Homework 3E__: Replace all `NAs` in the data frame `my_mw_target2a` with zeros.

```{r homework3D_solutions, eval=FALSE, echo=FALSE}
my_mw_target_tmp <- merge(my_mw, my_target[1:40,], by.x="ID", by.y="ID", all=FALSE) 
all(my_mw_target2 == my_mw_target_tmp)
my_mw_target2a <- as.matrix(my_mw_target2a)
my_mw_target2a[is.na(my_mw_target2a)] <- 0
my_mw_target2a <- as.data.frame(my_mw_target2a)
```

## Filtering Data

- __Step 5__: Retrieve all records with a value of greater than 100,000 in 'MW' column and 'C' value in 'Loc' column (targeted to chloroplast).

```{r filter_tables1, eval=TRUE}
query <- my_mw_target[my_mw_target[, 2] > 100000 & my_mw_target[, 4] == "C", ] 
query[1:4, ]
dim(query)
```

- __Homework 3F__: How many protein entries in the `my`_mw`_target` data frame have a MW of greater then 4,000 and less then 5,000. Subset the data frame accordingly and sort it by MW to check that your result is correct.


## String Substitutions

- __Step 6__: Use a regular expression in a substitute function to generate a separate ID column that lacks the gene model extensions.
<<label=Exercise 4.7, eval=TRUE, echo=TRUE, keep.source=TRUE>>=

```{r string_sub, eval=TRUE}
my_mw_target3 <- data.frame(loci=gsub("\\..*", "", as.character(my_mw_target[,1]), perl = TRUE), my_mw_target)
my_mw_target3[1:3,1:8]
```

- __Homework 3G__: Retrieve those rows in `my_mw_target3` where the second column contains the following identifiers: `c("AT5G52930.1", "AT4G18950.1", "AT1G15385.1", "AT4G36500.1", "AT1G67530.1")`. Use the `%in%` function for this query. As an alternative approach, assign the second column to the row index of the data frame and then perform the same query again using the row index. Explain the difference of the two methods.

## Calculations on Data Frames

- __Step 7__: Count the number of duplicates in the loci column with the `table` function and append the result to the data frame with the `cbind` function.

```{r calcul_1, eval=TRUE}
mycounts <- table(my_mw_target3[,1])[my_mw_target3[,1]]
my_mw_target4 <- cbind(my_mw_target3, Freq=mycounts[as.character(my_mw_target3[,1])]) 
```

- __Step 8__: Perform a vectorized devision of columns 3 and 4 (average AA weight per protein)

```{r calcul_2, eval=TRUE}
data.frame(my_mw_target4, avg_AA_WT=(my_mw_target4[,3] / my_mw_target4[,4]))[1:2,5:11] 
```

- __Step 9__: Calculate for each row the mean and standard deviation across several columns

```{r calcul_3, eval=TRUE}
mymean <- apply(my_mw_target4[,6:9], 1, mean)
mystdev <- apply(my_mw_target4[,6:9], 1, sd, na.rm=TRUE)
data.frame(my_mw_target4, mean=mymean, stdev=mystdev)[1:2,5:12] 
```

## Plotting Example

- __Step 10__: Generate scatter plot columns: 'MW' and 'Residues' 

```{r plot_example, eval=TRUE}
plot(my_mw_target4[1:500,3:4], col="red")
```

## Export Results and Run Entire Exercise as Script

- __Step 11__: Write the data frame `my_mw_target4` into a tab-delimited text file and inspect it in Excel.

```{r export_example, eval=TRUE}
write.table(my_mw_target4, file="my_file.xls", quote=F, sep="\t", col.names = NA) 
```

- __Homework 3H__: Write all commands from this exercise into an R script named `exerciseRbasics.R`, or download it from [here](http://faculty.ucr.edu/~tgirke/Documents/R_BioCond/My_R_Scripts/exerciseRbasics.R). Then execute the script with the `source` function like this: `source("exerciseRbasics.R")`. This will run all commands of this exercise and generate the corresponding output files in the current working directory.

```{r source_example, eval=FALSE}
source("exerciseRbasics.R")
```


# Session Info

```{r sessionInfo}
sessionInfo()
```

# References



<div class="tags">
    
</div>



</div>



<footer>
            <div class="row">
                <div class="col-lg-12 footer">
               &copy;2016 UC Riverside. All rights reserved. <br />
 Site last generated: Apr 26, 2016 <br />
<p><img src="../common_images/company_logo.png/></p>
                </div>
            </div>
</footer>

            </div><!-- /.row -->

    </div>    <!-- /.container -->

</body>


</html>

